---
title: "Metatranscriptome data analysis"
author: "Sarah Hu"
date: "9/1/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Set up working R environment
```{r}
library(multidplyr)
library(tidyverse)
library(dbplyr)
library(RSQLite)
library(tictoc)
library(data.table)
library(compositions)
```

# Import metatranscriptome R data objects

## Input all TPM results

Below R objects sourced by first isolating only the eukaryotic transcripts, averaging across replicates, and placing into long format.

```{r}
load(file = "/scratch/user/skhu/SPOT-ALOHA/Avg_tpm_ALL_acrossreps_annot.RData", verbose = T)
head(tpm_ALL_mean_reps_annot)
dim(tpm_ALL_mean_reps_annot)
```

Use ```multidplyr``` to perform tidyverse functions in parallel.
```{r}
# cluster
tic()
all_shard <- tpm_ALL_mean_reps_annot %>% 
  group_by(SAMPLENAME) %>% 
  partition(cluster)
toc() #95.245 sec elapsed
```

```{r}
# Supergroup
supergroup_all <- all_shard %>%
  group_by(Domain, Supergroup, PACIFIC_REGION, SAMPLENAME, SAMPLE_ID_PARSED, DEPTH_CATEGORY) %>% 
  summarize(SUM_TPM_supergroup = sum(MEAN_TPM)) %>% 
  collect()

supergroup_all %>% 
  filter(Domain == "Eukaryota") %>% 
  ggplot(aes(x = SAMPLE_ID_PARSED, y = SUM_TPM_supergroup, fill = Supergroup)) +
  geom_bar(stat = "identity", position = "fill", color = "black")+
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
```


```{r}
# Phylum
phylum_all <- all_shard %>%
  group_by(Domain, Supergroup, Phylum, PACIFIC_REGION, SAMPLENAME, SAMPLE_ID_PARSED, DEPTH_CATEGORY) %>% 
  summarize(SUM_TPM_supergroup = sum(MEAN_TPM)) %>% 
  collect() %>% 
  unite(Supergroup_phylum, Supergroup, Phylum, sep = " ")

phylum_all %>% 
  filter(Domain == "Eukaryota") %>% 
  ggplot(aes(x = SAMPLE_ID_PARSED, y = SUM_TPM_supergroup, fill = Supergroup_phylum)) +
  geom_bar(stat = "identity", position = "fill", color = "black")+
  theme_linedraw() +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))
```



# Taxonomic composition & distribution

Evaluate taxonomic diversity and composition across all sites.

## Taxonomic level curation and estimate relative abundance

Taxa curation
3) Average across replicates (or how should replicates be considered)
4) Sum to each sample

## Taxonomy bar plot
5) bar plot of relative TPM? CPM? for major taxnomic groups
Keep ALOHA July and April separate, ALOHA on the left side and on the right side, all CA sites.
Also place "sun" emoji for samples that are within the euphotic zone



# Whole community functional profile

Prepare data ahead of time. Use normalization to the whole community.
Include functional modules that were curated for this work.

> What percentage of the data are represented when we only look at the curated data?

Change input data to all numeric for downstream transformations, etc.

1) all data, change zeros to 1 (or to below the lowest value)

2) all data, remove NAs and zeros. Only represent proteins that appear in all samples

3) Subset to only those protein IDs of interest (get stats on what this is)
Make sure to retain module information in definiation. 
3.1) change zeroes to 1 (or lowest)
3.2) remove NAs

## PCoA with whole community functional profile

```{r}
head(tpm_all_euk_clr)
```
```{r}
pca_lr <- prcomp(as.data.frame(tpm_all_euk_clr))

variance_lr <- (pca_lr$sdev^2)/sum(pca_lr$sdev^2)
## View bar plot
barplot(variance_lr, main = "Log-Ratio PCA Screeplot", xlab = "PC Axis", ylab = "% Variance", 
    cex.names = 1.5, cex.axis = 1.5, cex.lab = 1.5, cex.main = 1.5)
```


### Does PCoA change when only taxonomic information is input?
(supplementary)

## Define each site by module
(k-means?)

# Transcriptional profiles for taxonomic groups
(supplementary?)
Normalized by individual taxonomic groups.

## PCoA with curated functional profiles and known taxonomic IDs.
(subset this way to increase confidence?)

### Heterotrophy

Isolate and sum to pathways associated with heterotrophy. Gene level is likely too much. Determine for each 

> How varied are the transcriptional signatures related to heterotrophy among protists?

Use flag/half square heatmap, where top half shows the percent number of genes and bottom half shows the TPM summary. 

### N & P metabolism

Isolate at individual gene level for N & P pathways. Show TPM across sites and core taxa

> For ecologically-important phytoplankton at ALOHA and coastal CA, what is the relative importance of nutrient profiles? nutrient utilization?

Plot heat map for each taxa. Show all N and P relevant genes on the x axis and y axis should have all samples. 

### Determine statistical significance
Estimate statistical significance for N&P genes, pairwise comparisons. Then add a column for each gene ID that lists the number of samples that are statistically different. 



# Functional redundancy

Input data is all normalized. Then perform pairwise Jaccard dissimilarity with taxa and then with KEGG/pfam IDs - plot as x/y

# Session info
```{r}
sessionInfo()
```

