<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Analysis of metatranscriptome data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./metat-analysis.html">Data analysis</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./env-metadata.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Water column profiles</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./process-eukrhythmic-output.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">eukrhythmic output</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metat-analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Data analysis</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#premise" id="toc-premise" class="nav-link active" data-scroll-target="#premise">Premise</a></li>
  <li><a href="#set-up-working-r-environment" id="toc-set-up-working-r-environment" class="nav-link" data-scroll-target="#set-up-working-r-environment">Set up working R environment</a></li>
  <li><a href="#whole-community-analysis" id="toc-whole-community-analysis" class="nav-link" data-scroll-target="#whole-community-analysis">Whole community analysis</a>
  <ul class="collapse">
  <li><a href="#input-all-tpm-results" id="toc-input-all-tpm-results" class="nav-link" data-scroll-target="#input-all-tpm-results">Input all TPM results</a></li>
  <li><a href="#taxonomic-composition-distribution" id="toc-taxonomic-composition-distribution" class="nav-link" data-scroll-target="#taxonomic-composition-distribution">Taxonomic composition &amp; distribution</a>
  <ul class="collapse">
  <li><a href="#plot-domain-level" id="toc-plot-domain-level" class="nav-link" data-scroll-target="#plot-domain-level">Plot domain level</a></li>
  <li><a href="#plot-to-supergroup" id="toc-plot-to-supergroup" class="nav-link" data-scroll-target="#plot-to-supergroup">Plot to supergroup</a></li>
  </ul></li>
  <li><a href="#pcoa-with-whole-community-functional-profile" id="toc-pcoa-with-whole-community-functional-profile" class="nav-link" data-scroll-target="#pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</a></li>
  <li><a href="#untargeted---de-transcripts" id="toc-untargeted---de-transcripts" class="nav-link" data-scroll-target="#untargeted---de-transcripts">Untargeted - DE transcripts</a></li>
  </ul></li>
  <li><a href="#prep-to-address-key-questions" id="toc-prep-to-address-key-questions" class="nav-link" data-scroll-target="#prep-to-address-key-questions">Prep to address key questions</a>
  <ul class="collapse">
  <li><a href="#import-kegg-ids" id="toc-import-kegg-ids" class="nav-link" data-scroll-target="#import-kegg-ids">Import KEGG IDs</a></li>
  </ul></li>
  <li><a href="#prepare-data-frames-for-targeted-approach" id="toc-prepare-data-frames-for-targeted-approach" class="nav-link" data-scroll-target="#prepare-data-frames-for-targeted-approach">Prepare data frames for targeted approach</a></li>
  <li><a href="#question-1" id="toc-question-1" class="nav-link" data-scroll-target="#question-1">Question 1</a>
  <ul class="collapse">
  <li><a href="#untargeted" id="toc-untargeted" class="nav-link" data-scroll-target="#untargeted">Untargeted</a>
  <ul class="collapse">
  <li><a href="#stats-on-de-transcripts" id="toc-stats-on-de-transcripts" class="nav-link" data-scroll-target="#stats-on-de-transcripts">Stats on DE transcripts</a></li>
  <li><a href="#local---untargeted-mining" id="toc-local---untargeted-mining" class="nav-link" data-scroll-target="#local---untargeted-mining">Local - untargeted mining</a></li>
  </ul></li>
  <li><a href="#targeted" id="toc-targeted" class="nav-link" data-scroll-target="#targeted">Targeted</a></li>
  </ul></li>
  <li><a href="#question-2" id="toc-question-2" class="nav-link" data-scroll-target="#question-2">Question 2</a>
  <ul class="collapse">
  <li><a href="#untargeted-1" id="toc-untargeted-1" class="nav-link" data-scroll-target="#untargeted-1">Untargeted</a>
  <ul class="collapse">
  <li><a href="#shared-vs-unique" id="toc-shared-vs-unique" class="nav-link" data-scroll-target="#shared-vs-unique">Shared vs unique</a></li>
  </ul></li>
  <li><a href="#targeted-1" id="toc-targeted-1" class="nav-link" data-scroll-target="#targeted-1">Targeted</a></li>
  </ul></li>
  <li><a href="#supplementary" id="toc-supplementary" class="nav-link" data-scroll-target="#supplementary">Supplementary</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analysis of metatranscriptome data</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="premise" class="level1">
<h1>Premise</h1>
<p>We have taken outputs from <em>eukrhythmic</em> and performed transcript-level normalization so we can do downstream analysis. There are two core questions for this metatranscriptome study that can be subset into untargeted and targeted approaches. Questions stem from previous studies at both the NPSG and sites off of coastal California.</p>
<table class="table">
<colgroup>
<col style="width: 3%">
<col style="width: 45%">
<col style="width: 50%">
</colgroup>
<tbody>
<tr class="odd">
<td></td>
<td><strong>(1) How does nutrient utilization among euphotic zone phytoplankton vary between coastal California and the NPSG?</strong></td>
<td><strong>(2) Do we see a similar euphotic vs.&nbsp;sub-euphotic shift in the metabolic potential of taxa observed at coastal California and the NPSG?</strong></td>
</tr>
<tr class="even">
<td>Untargeted</td>
<td><ol type="1">
<li><p>Subset whole dataset to taxa of interest (haptophytes, dinos, diatoms, archaeplastida), and to transcripts that were annotated.</p></li>
<li><p>Run in DEseq2 to determine differentially expressed genes between: SPOT surface vs.&nbsp;all other samples in the euphotic zone.</p></li>
<li><p>Tally what appears similar and different across sample comparisons</p></li>
<li><p>Then tally what appears different between taxa..</p></li>
</ol></td>
<td><ol type="1">
<li><p>Subset whole dataset to taxa of interest, dinos, ciliates, rhizaria, haptophytes.</p></li>
<li><p>Run in DESeq2. Compare all euphotic to all subeuphotic. Then NPSG euphotic to subeuphotic. Then SPOT euphotic to subeuphotic.</p></li>
<li><p>Tally what appears similar and different across sample comparisons</p></li>
<li><p>Then tally what appears different between taxa.</p></li>
</ol></td>
</tr>
<tr class="odd">
<td>Targeted</td>
<td>SUBSET only those 4 groups from above. Run the count normalization, then isolate the key genes of interest. What is happening with critical N&amp;P pathways?</td>
<td>SUBSET only those 4 groups from above. Run the count normalization, then isolate the all the key genes of interest. Paying particular attention to phagotrophy-based genes.</td>
</tr>
</tbody>
</table>
</section>
<section id="set-up-working-r-environment" class="level1">
<h1>Set up working R environment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="whole-community-analysis" class="level1">
<h1>Whole community analysis</h1>
<p>Below, we will import the averaged (across replicates) and scaled transcript per million (TPM) values so we can look at the overall protistan community structure.</p>
<section id="input-all-tpm-results" class="level2">
<h2 class="anchored" data-anchor-id="input-all-tpm-results">Input all TPM results</h2>
<p>Below R objects sourced by first isolating only the eukaryotic transcripts, averaging across replicates, and placing into long format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"Avg_scaled_tpm_08222023.RData"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>taxfxn <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"TaxonomicAndFunctionalAnnotations.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Import metadata as well.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># metadata_tmp &lt;- read.delim("input-data/complete-sample-list.txt", header = TRUE, sep = ",")</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_metadata &lt;- read.csv("input-data/sample_table.csv") %&gt;%</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># unite(SAMPLE, Sample, Replicate, sep = "_", remove = FALSE) %&gt;% </span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select(RUN = SRR, everything()) %&gt;% </span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># right_join(metadata_tmp)</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(sample_metadata, file = "input-data/sample-metadata.csv")</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>sample_metadata_1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"input-data/sample-metadata.csv"</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> sample_metadata_1 <span class="sc">%&gt;%</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE_MATCH =</span> <span class="fu">sub</span>(<span class="st">"_[^_]+$"</span>, <span class="st">""</span>, SAMPLE))</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>sample_metadata</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>    X         RUN         SAMN                  SAMPLE       Sample   Replicate
1   1  SRR6048900 SAMN07647713  July_1000m_Rep1andRep2   July_1000m Rep1andRep2
2   2  SRR6048899 SAMN07647714   July_150m_Rep1andRep2    July_150m Rep1andRep2
3   3  SRR6048898 SAMN07647715            July_5m_Rep1      July_5m        Rep1
4   4  SRR6048897 SAMN07647716            July_5m_Rep2      July_5m        Rep2
5   5  SRR6048896 SAMN07647717           July_DCM_Rep1     July_DCM        Rep1
6   6  SRR6048895 SAMN07647718           July_DCM_Rep2     July_DCM        Rep2
7   7  SRR6048894 SAMN07647719 March_1000m_Rep1andRep2  March_1000m Rep1andRep2
8   8  SRR6048893 SAMN07647720  March_150m_Rep1andRep2   March_150m Rep1andRep2
9   9  SRR6048892 SAMN07647721           March_5m_Rep1     March_5m        Rep1
10 10  SRR6048891 SAMN07647722           March_5m_Rep2     March_5m        Rep2
11 11  SRR6048902 SAMN07647723          March_DCM_Rep1    March_DCM        Rep1
12 12  SRR6048901 SAMN07647724          March_DCM_Rep2    March_DCM        Rep2
13 13 SRR11178183 SAMN14206057             Catalina_19     Catalina          19
14 14 SRR11178182 SAMN14206058             Catalina_20     Catalina          20
15 15 SRR11178173 SAMN14206059             Catalina_21     Catalina          21
16 16 SRR11178172 SAMN14206060             Catalina_22     Catalina          22
17 17 SRR11178171 SAMN14206061             Catalina_23     Catalina          23
18 18 SRR11178170 SAMN14206062             Catalina_24     Catalina          24
19 19 SRR11178169 SAMN14206063              PortofLA_1     PortofLA           1
20 20 SRR11178168 SAMN14206064              PortofLA_2     PortofLA           2
21 21 SRR11178167 SAMN14206065              PortofLA_3     PortofLA           3
22 22 SRR11178166 SAMN14206066              PortofLA_4     PortofLA           4
23 23 SRR11178181 SAMN14206067              PortofLA_5     PortofLA           5
24 24 SRR11178180 SAMN14206068              PortofLA_6     PortofLA           6
25 25  SRR5799332 SAMN07269832      SPOT_150m_Rep1and2    SPOT_150m    Rep1and2
26 26  SRR5799333 SAMN07269833      SPOT_150m_Rep3and4    SPOT_150m    Rep3and4
27 27  SRR5799340 SAMN07269834      SPOT_150m_Rep5and6    SPOT_150m    Rep5and6
28 28  SRR5799341 SAMN07269835      SPOT_890m_Rep1and2    SPOT_890m    Rep1and2
29 29  SRR5799343 SAMN07269836      SPOT_890m_Rep3and4    SPOT_890m    Rep3and4
30 30  SRR5799344 SAMN07269837      SPOT_890m_Rep5and6    SPOT_890m    Rep5and6
31 31  SRR5799342 SAMN07269838          SPOT_890m_Rep7    SPOT_890m        Rep7
32 32  SRR5799338 SAMN07269826          SPOT_surface_7 SPOT_surface           7
33 33  SRR5799339 SAMN07269827          SPOT_surface_8 SPOT_surface           8
34 34  SRR5799336 SAMN07269828          SPOT_surface_9 SPOT_surface           9
35 35  SRR5799337 SAMN07269829         SPOT_surface_10 SPOT_surface          10
36 36  SRR5799334 SAMN07269830         SPOT_surface_11 SPOT_surface          11
37 37  SRR5799335 SAMN07269831         SPOT_surface_12 SPOT_surface          12
38 38 SRR11178179 SAMN14206069         SPOT_surface_13 SPOT_surface          13
39 39 SRR11178178 SAMN14206070         SPOT_surface_14 SPOT_surface          14
40 40 SRR11178177 SAMN14206071         SPOT_surface_!5 SPOT_surface          !5
41 41 SRR11178176 SAMN14206072         SPOT_surface_16 SPOT_surface          16
42 42 SRR11178175 SAMN14206073         SPOT_surface_17 SPOT_surface          17
43 43 SRR11178174 SAMN14206074         SPOT_surface_18 SPOT_surface          18
         BASES  BIOPROJECT    BIOSAMPLE       DATE EXPERIMENT
1  22179808750 PRJNA406025 SAMN07647713 2015-07-30 SRX3195831
2  22251034000 PRJNA406025 SAMN07647714 2015-07-30 SRX3195832
3  36858334750 PRJNA406025 SAMN07647715 2015-07-30 SRX3195833
4  28427020000 PRJNA406025 SAMN07647716 2015-07-30 SRX3195834
5  33698315000 PRJNA406025 SAMN07647717 2015-07-30 SRX3195835
6  30540287750 PRJNA406025 SAMN07647718 2015-07-30 SRX3195836
7  10727941750 PRJNA406025 SAMN07647719 2016-03-27 SRX3195837
8  28071351250 PRJNA406025 SAMN07647720 2016-03-27 SRX3195838
9  35787763750 PRJNA406025 SAMN07647721 2016-03-27 SRX3195839
10 26198209000 PRJNA406025 SAMN07647722 2016-03-27 SRX3195840
11 36850615000 PRJNA406025 SAMN07647723 2016-03-27 SRX3195829
12 25119184500 PRJNA406025 SAMN07647724 2016-03-27 SRX3195830
13  8956532750 PRJNA608423 SAMN14206057 2015-05-21 SRX7798845
14  9890647250 PRJNA608423 SAMN14206058 2015-05-21 SRX7798846
15  9213733750 PRJNA608423 SAMN14206059 2015-05-21 SRX7798855
16  5346436500 PRJNA608423 SAMN14206060 2015-05-21 SRX7798856
17  7289444750 PRJNA608423 SAMN14206061 2015-05-21 SRX7798857
18 11865681500 PRJNA608423 SAMN14206062 2015-05-21 SRX7798858
19  7728633500 PRJNA608423 SAMN14206063 2015-05-20 SRX7798859
20  9733064500 PRJNA608423 SAMN14206064 2015-05-20 SRX7798860
21  4491843250 PRJNA608423 SAMN14206065 2015-05-20 SRX7798861
22  6380099500 PRJNA608423 SAMN14206066 2015-05-20 SRX7798862
23  7907104250 PRJNA608423 SAMN14206067 2015-05-20 SRX7798847
24  9629580250 PRJNA608423 SAMN14206068 2015-05-20 SRX7798848
25 21179947000 PRJNA391503 SAMN07269832 2015-05-20 SRX2978825
26 13838597000 PRJNA391503 SAMN07269833 2015-05-20 SRX2978824
27 24355501250 PRJNA391503 SAMN07269834 2015-05-20 SRX2978817
28  4599264250 PRJNA391503 SAMN07269835 2015-05-20 SRX2978816
29 20218093250 PRJNA391503 SAMN07269836 2015-05-20 SRX2978814
30 15436881250 PRJNA391503 SAMN07269837 2015-05-20 SRX2978813
31 16907100750 PRJNA391503 SAMN07269838 2015-05-20 SRX2978815
32  6490449000 PRJNA391503 SAMN07269826 2015-05-20 SRX2978819
33  3741887000 PRJNA391503 SAMN07269827 2015-05-20 SRX2978818
34  8523152000 PRJNA391503 SAMN07269828 2015-05-20 SRX2978821
35 10675178500 PRJNA391503 SAMN07269829 2015-05-20 SRX2978820
36  8449256000 PRJNA391503 SAMN07269830 2015-05-20 SRX2978823
37  7203720250 PRJNA391503 SAMN07269831 2015-05-20 SRX2978822
38 12565907750 PRJNA608423 SAMN14206069 2015-05-20 SRX7798849
39  7464807500 PRJNA608423 SAMN14206070 2015-05-20 SRX7798850
40  8583553750 PRJNA608423 SAMN14206071 2015-05-20 SRX7798851
41  8126200000 PRJNA608423 SAMN14206072 2015-05-20 SRX7798852
42  8329988250 PRJNA608423 SAMN14206073 2015-05-20 SRX7798853
43  6430857250 PRJNA608423 SAMN14206074 2015-05-20 SRX7798854
                  LAT_LON                  LIBRARY            SAMPLENAME DEPTH
1  24.54987 N 156.33178 W               July_1000m  NPSG_ALOHA_1000_July  1000
2  24.54987 N 156.33178 W                July_150m   NPSG_ALOHA_150_July   150
3  24.54987 N 156.33178 W        July_surface_Rep1     NPSG_ALOHA_5_July     5
4  24.54987 N 156.33178 W        July_surface_Rep2     NPSG_ALOHA_5_July     5
5  24.54987 N 156.33178 W            July_DCM_Rep1   NPSG_ALOHA_119_July   119
6  24.54987 N 156.33178 W            July_DCM_Rep2   NPSG_ALOHA_119_July   119
7  24.54987 N 156.33178 W              March_1000m NPSG_ALOHA_1000_March  1000
8  24.54987 N 156.33178 W               March_150m  NPSG_ALOHA_150_March   150
9  24.54987 N 156.33178 W       March_surface_Rep1    NPSG_ALOHA_5_March     5
10 24.54987 N 156.33178 W       March_surface_Rep2    NPSG_ALOHA_5_March     5
11 24.54987 N 156.33178 W           March_DCM_Rep1  NPSG_ALOHA_120_March   120
12 24.54987 N 156.33178 W           March_DCM_Rep2  NPSG_ALOHA_120_March   120
13       33.44 N 118.47 W     Catalina_19_S22_L004     CA_Catalina_5_May     5
14       33.44 N 118.47 W     Catalina_20_S23_L004     CA_Catalina_5_May     5
15       33.44 N 118.47 W     Catalina_21_S24_L004     CA_Catalina_5_May     5
16       33.44 N 118.47 W     Catalina_22_S25_L004     CA_Catalina_5_May     5
17       33.44 N 118.47 W     Catalina_23_S26_L004     CA_Catalina_5_May     5
18       33.44 N 118.47 W     Catalina_24_S27_L004     CA_Catalina_5_May     5
19       33.71 N 118.26 W     Port_of_LA_1_S1_L001     CA_PortofLA_5_May     5
20       33.71 N 118.26 W     Port_of_LA_2_S2_L001     CA_PortofLA_5_May     5
21       33.71 N 118.26 W     Port_of_LA_3_S3_L001     CA_PortofLA_5_May     5
22       33.71 N 118.26 W     Port_of_LA_4_S4_L001     CA_PortofLA_5_May     5
23       33.71 N 118.26 W     Port_of_LA_5_S5_L001     CA_PortofLA_5_May     5
24       33.71 N 118.26 W     Port_of_LA_6_S6_L001     CA_PortofLA_5_May     5
25       33.55 N 118.40 W             SPOT_150m_25       CA_SPOT_150_May   150
26       33.55 N 118.40 W             SPOT_150m_26       CA_SPOT_150_May   150
27       33.55 N 118.40 W             SPOT_150m_27       CA_SPOT_150_May   150
28       33.55 N 118.40 W             SPOT_890m_28       CA_SPOT_890_May   890
29       33.55 N 118.40 W             SPOT_890m_29       CA_SPOT_890_May   890
30       33.55 N 118.40 W             SPOT_890m_30       CA_SPOT_890_May   890
31       33.55 N 118.40 W             SPOT_890m_31       CA_SPOT_890_May   890
32       33.55 N 118.40 W           SPOT_surface_7         CA_SPOT_5_May     5
33       33.55 N 118.40 W           SPOT_surface_8         CA_SPOT_5_May     5
34       33.55 N 118.40 W           SPOT_surface_9         CA_SPOT_5_May     5
35       33.55 N 118.40 W          SPOT_surface_10         CA_SPOT_5_May     5
36       33.55 N 118.40 W          SPOT_surface_11         CA_SPOT_5_May     5
37       33.55 N 118.40 W          SPOT_surface_12         CA_SPOT_5_May     5
38       33.44 N 118.47 W SPOT_surface_13_S15_L003         CA_SPOT_5_May     5
39       33.44 N 118.47 W SPOT_surface_14_S16_L003         CA_SPOT_5_May     5
40       33.44 N 118.47 W SPOT_surface_15_S17_L003         CA_SPOT_5_May     5
41       33.44 N 118.47 W SPOT_surface_16_S18_L003         CA_SPOT_5_May     5
42       33.44 N 118.47 W SPOT_surface_17_S19_L003         CA_SPOT_5_May     5
43       33.44 N 118.47 W SPOT_surface_18_S20_L003         CA_SPOT_5_May     5
      SAMPLE_ID_PARSED DEPTH_CATEGORY   REGION PACIFIC_REGION MONTH
1    July ALOHA 1000 m         1000 m    ALOHA           NPSG  July
2     July ALOHA 150 m          150 m    ALOHA           NPSG  July
3   July ALOHA surface        surface    ALOHA           NPSG  July
4   July ALOHA surface        surface    ALOHA           NPSG  July
5       July ALOHA DCM            DCM    ALOHA           NPSG  July
6       July ALOHA DCM            DCM    ALOHA           NPSG  July
7   March ALOHA 1000 m         1000 m    ALOHA           NPSG March
8    March ALOHA 150 m          150 m    ALOHA           NPSG March
9  March ALOHA surface        surface    ALOHA           NPSG March
10 March ALOHA surface        surface    ALOHA           NPSG March
11     March ALOHA DCM            DCM    ALOHA           NPSG March
12     March ALOHA DCM            DCM    ALOHA           NPSG March
13    Catalina surface        surface Catalina             CA   May
14    Catalina surface        surface Catalina             CA   May
15    Catalina surface        surface Catalina             CA   May
16    Catalina surface        surface Catalina             CA   May
17    Catalina surface        surface Catalina             CA   May
18    Catalina surface        surface Catalina             CA   May
19  Port of LA surface        surface PortofLA             CA   May
20  Port of LA surface        surface PortofLA             CA   May
21  Port of LA surface        surface PortofLA             CA   May
22  Port of LA surface        surface PortofLA             CA   May
23  Port of LA surface        surface PortofLA             CA   May
24  Port of LA surface        surface PortofLA             CA   May
25          SPOT 150 m          150 m     SPOT             CA   May
26          SPOT 150 m          150 m     SPOT             CA   May
27          SPOT 150 m          150 m     SPOT             CA   May
28          SPOT 890 m          890 m     SPOT             CA   May
29          SPOT 890 m          890 m     SPOT             CA   May
30          SPOT 890 m          890 m     SPOT             CA   May
31          SPOT 890 m          890 m     SPOT             CA   May
32        SPOT surface        surface     SPOT             CA   May
33        SPOT surface        surface     SPOT             CA   May
34        SPOT surface        surface     SPOT             CA   May
35        SPOT surface        surface     SPOT             CA   May
36        SPOT surface        surface     SPOT             CA   May
37        SPOT surface        surface     SPOT             CA   May
38        SPOT surface        surface     SPOT             CA   May
39        SPOT surface        surface     SPOT             CA   May
40        SPOT surface        surface     SPOT             CA   May
41        SPOT surface        surface     SPOT             CA   May
42        SPOT surface        surface     SPOT             CA   May
43        SPOT surface        surface     SPOT             CA   May
          LIGHT SAMPLE_MATCH
1  Sub-euphotic   July_1000m
2      Euphotic    July_150m
3      Euphotic      July_5m
4      Euphotic      July_5m
5      Euphotic     July_DCM
6      Euphotic     July_DCM
7  Sub-euphotic  March_1000m
8      Euphotic   March_150m
9      Euphotic     March_5m
10     Euphotic     March_5m
11     Euphotic    March_DCM
12     Euphotic    March_DCM
13     Euphotic     Catalina
14     Euphotic     Catalina
15     Euphotic     Catalina
16     Euphotic     Catalina
17     Euphotic     Catalina
18     Euphotic     Catalina
19     Euphotic     PortofLA
20     Euphotic     PortofLA
21     Euphotic     PortofLA
22     Euphotic     PortofLA
23     Euphotic     PortofLA
24     Euphotic     PortofLA
25 Sub-euphotic    SPOT_150m
26 Sub-euphotic    SPOT_150m
27 Sub-euphotic    SPOT_150m
28 Sub-euphotic    SPOT_890m
29 Sub-euphotic    SPOT_890m
30 Sub-euphotic    SPOT_890m
31 Sub-euphotic    SPOT_890m
32     Euphotic SPOT_surface
33     Euphotic SPOT_surface
34     Euphotic SPOT_surface
35     Euphotic SPOT_surface
36     Euphotic SPOT_surface
37     Euphotic SPOT_surface
38     Euphotic SPOT_surface
39     Euphotic SPOT_surface
40     Euphotic SPOT_surface
41     Euphotic SPOT_surface
42     Euphotic SPOT_surface
43     Euphotic SPOT_surface</code></pre>
</div>
</div>
</section>
<section id="taxonomic-composition-distribution" class="level2">
<h2 class="anchored" data-anchor-id="taxonomic-composition-distribution">Taxonomic composition &amp; distribution</h2>
<p>Evaluate taxonomic diversity and composition across all sites.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(taxfxn<span class="sc">$</span>SequenceID))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 14840771</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">rownames</span>(mean_counts_df)))</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 14737135</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>annotated_mean <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(mean_counts_df <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">mutate</span>(<span class="at">SequenceID =</span> <span class="fu">rownames</span>(mean_counts_df)),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>                                 taxfxn,</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">by =</span> <span class="st">"SequenceID"</span>)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 14737135-14737693</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="do">## 558 were duplicated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Remove the duplicated annotations and make a base taxonomic dataframe for downstream work.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>tax_plot <span class="ot">&lt;-</span> annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Domain<span class="sc">:</span>Species, <span class="fu">starts_with</span>(<span class="st">"mean"</span>), SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(tax_plot) #14737624\</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/tax_plot_08222023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>tax_only <span class="ot">&lt;-</span> tax_plot <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"mean"</span>), <span class="at">names_to =</span> <span class="st">"SAMPLE"</span>, <span class="at">values_to =</span> <span class="st">"scaledTPM"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">SUM_scaledTPM =</span> <span class="fu">sum</span>(scaledTPM)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/tax_only_08242023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  tax_only</code></pre>
</div>
</div>
<section id="plot-domain-level" class="level3">
<h3 class="anchored" data-anchor-id="plot-domain-level">Plot domain level</h3>
<p>Creating plot to see total number and relative abundance of non-eukaryotic hits in the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>(tax_only <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> DOMAIN)) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))) <span class="sc">/</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>(tax_only <span class="sc">%&gt;%</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> DOMAIN)) <span class="sc">+</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<p>Stats on what % of the data are eukaryotic vs.&nbsp;not.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>SUM_TOTAL_TPM <span class="ot">&lt;-</span> <span class="fu">sum</span>(tax_only<span class="sc">$</span>SUM_scaledTPM)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>tax_only <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SUM_scaledTPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span>  <span class="co"># Remove zeroes</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">Sum_of_TPM =</span> <span class="fu">sum</span>(SUM_scaledTPM),</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">Perc_of_total =</span> <span class="dv">100</span><span class="sc">*</span>(Sum_of_TPM<span class="sc">/</span>SUM_TOTAL_TPM),</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups =</span> <span class="st">"rowwise"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 3
# Rowwise:  DOMAIN
  DOMAIN       Sum_of_TPM Perc_of_total
  &lt;chr&gt;             &lt;dbl&gt;         &lt;dbl&gt;
1 Archaea         305556.        0.159 
2 Bacteria       5438767.        2.84  
3 Eukaryota    185858173.       96.9   
4 Unclassified    174159.        0.0908</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Count = count(DOMAIN))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-to-supergroup" class="level3">
<h3 class="anchored" data-anchor-id="plot-to-supergroup">Plot to supergroup</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tax_only)</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>tax_only <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SUPERGROUP =</span> <span class="fu">str_squish</span>(Supergroup),</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">PHYLUM =</span> <span class="fu">str_squish</span>(Phylum),</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DOMAIN <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, Supergroup) <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> Supergroup)) <span class="sc">+</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Tally composition of the supergroup.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tax_only)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "SAMPLE"        "Domain"        "Supergroup"    "Phylum"       
 [5] "Class"         "Order"         "Family"        "Genus"        
 [9] "Species"       "SUM_scaledTPM"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(tax_only$Phylum)</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> tax_only <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SUPERGROUP =</span> <span class="fu">str_squish</span>(Supergroup),</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">PHYLUM =</span> <span class="fu">str_squish</span>(Phylum),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DOMAIN <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(DOMAIN, SUPERGROUP, Class<span class="sc">:</span>Species) <span class="sc">%&gt;%</span> </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `SAMPLE`, `Domain`, `Supergroup`, `Phylum`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_names<span class="sc">$</span>Phylum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] " Apicomplexa"      " Ciliophora"       " Dinoflagellata"  
 [4] " Perkinsea"        "Unclassified"      " Conosa"          
 [7] " Discosea"         " Lobosa"           " Chlorophyta"     
[10] " Glaucophyta"      " Rhodophyta"       " Discoba"         
[13] " Cryptophyta"      " Hacrobia_X"       " Haptophyta"      
[16] " Choanoflagellida" " Fungi"            " Cercozoa"        
[19] " Foraminifera"     " Ochrophyta"       " Opalozoa"        
[22] " Sagenista"        " Stramenopiles_X" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_names<span class="sc">$</span>Supergroup)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] " Alveolata"      " Amoebozoa"      " Archaeplastida" " Excavata"      
[5] " Hacrobia"       " Opisthokonta"   " Rhizaria"       " Stramenopiles" 
[9] "Unclassified"   </code></pre>
</div>
</div>
<p>Now, reformat taxonomic information to visualize.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_only<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "mean.Catalina"     "mean.July_1000m"   "mean.July_150m"   
 [4] "mean.July_5m"      "mean.July_DCM"     "mean.March_1000m" 
 [7] "mean.March_150m"   "mean.March_5m"     "mean.March_DCM"   
[10] "mean.PortofLA"     "mean.SPOT_150m"    "mean.SPOT_890m"   
[13] "mean.SPOT_surface"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(tax_only)</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="co"># head(sample_metadata)</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>tax_formatted <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax_only) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove preceding spaces in taxonomic names</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species), <span class="sc">~</span> <span class="fu">as.character</span>(<span class="fu">str_squish</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> SUM_scaledTPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE_MATCH =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"SAMPLE_MATCH"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Taxonomic group curation</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">case_when</span>(</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Alveolata"</span> <span class="sc">~</span> <span class="fu">paste</span>(Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>    (Class <span class="sc">==</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span> Supergroup <span class="sc">==</span> <span class="st">"Stramenopiles"</span>) <span class="sc">~</span> <span class="fu">paste</span>(Supergroup, <span class="st">"Diatom"</span>, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>    (Class <span class="sc">!=</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span> Supergroup <span class="sc">==</span> <span class="st">"Stramenopiles"</span>) <span class="sc">~</span> <span class="fu">paste</span>(Supergroup, <span class="st">"Other"</span>, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Archaeplastida"</span> <span class="sc">~</span> <span class="st">"Chlorophyte"</span>,</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Supergroup</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., sample_metadata, by = "SAMPLE_MATCH"): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 13 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tax_formatted)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assign levels for taxonomic groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>tax_order_color<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"#612741"</span>,<span class="st">"#b74a70"</span>,<span class="st">"#b7757c"</span>,<span class="st">"#eecfbf"</span>,<span class="st">"#92462f"</span>,<span class="st">"#bb603c"</span>,<span class="st">"#dfa837"</span>,<span class="st">"#ccc050"</span>,<span class="st">"#33431e"</span>,<span class="st">"#93b778"</span>,<span class="st">"#61ac86"</span>,<span class="st">"#657abb"</span>,<span class="st">"#1c1949"</span>,<span class="st">"#8a8d84"</span>)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>supergroup_order <span class="ot">&lt;-</span> (<span class="fu">as.character</span>(<span class="fu">unique</span>(tax_formatted<span class="sc">$</span>SUPERGROUP)))</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tax_order_color) <span class="ot">&lt;-</span> (supergroup_order)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figure-1" class="level4">
<h4 class="anchored" data-anchor-id="figure-1">Figure 1</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>tax_formatted <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">factor</span>(tax_formatted<span class="sc">$</span>SUPERGROUP, <span class="at">levels =</span> supergroup_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE_ID_PARSED, PACIFIC_REGION, LIGHT, DEPTH, SUPERGROUP) <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ID_PARSED, <span class="at">y =</span> SUM, <span class="at">fill =</span> SUPERGROUP)) <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(PACIFIC_REGION), <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">space =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> tax_order_color) <span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE_ID_PARSED', 'PACIFIC_REGION',
'LIGHT', 'DEPTH'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="additional-taxonomic-resolution" class="level4">
<h4 class="anchored" data-anchor-id="additional-taxonomic-resolution">Additional taxonomic resolution</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tax_formatted)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       SAMPLE.x    Domain Supergroup      Phylum         Class        Order
1 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
2 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
3 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
4 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
5 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
6 mean.Catalina Eukaryota  Alveolata Apicomplexa Colpodellidea Colpodellida
        Family    Genus        Species SUM_scaledTPM SAMPLE_MATCH  X
1 Chromeraceae Chromera Chromera velia      224.8899     Catalina 13
2 Chromeraceae Chromera Chromera velia      224.8899     Catalina 14
3 Chromeraceae Chromera Chromera velia      224.8899     Catalina 15
4 Chromeraceae Chromera Chromera velia      224.8899     Catalina 16
5 Chromeraceae Chromera Chromera velia      224.8899     Catalina 17
6 Chromeraceae Chromera Chromera velia      224.8899     Catalina 18
          RUN         SAMN    SAMPLE.y   Sample Replicate       BASES
1 SRR11178183 SAMN14206057 Catalina_19 Catalina        19  8956532750
2 SRR11178182 SAMN14206058 Catalina_20 Catalina        20  9890647250
3 SRR11178173 SAMN14206059 Catalina_21 Catalina        21  9213733750
4 SRR11178172 SAMN14206060 Catalina_22 Catalina        22  5346436500
5 SRR11178171 SAMN14206061 Catalina_23 Catalina        23  7289444750
6 SRR11178170 SAMN14206062 Catalina_24 Catalina        24 11865681500
   BIOPROJECT    BIOSAMPLE       DATE EXPERIMENT          LAT_LON
1 PRJNA608423 SAMN14206057 2015-05-21 SRX7798845 33.44 N 118.47 W
2 PRJNA608423 SAMN14206058 2015-05-21 SRX7798846 33.44 N 118.47 W
3 PRJNA608423 SAMN14206059 2015-05-21 SRX7798855 33.44 N 118.47 W
4 PRJNA608423 SAMN14206060 2015-05-21 SRX7798856 33.44 N 118.47 W
5 PRJNA608423 SAMN14206061 2015-05-21 SRX7798857 33.44 N 118.47 W
6 PRJNA608423 SAMN14206062 2015-05-21 SRX7798858 33.44 N 118.47 W
               LIBRARY        SAMPLENAME DEPTH SAMPLE_ID_PARSED DEPTH_CATEGORY
1 Catalina_19_S22_L004 CA_Catalina_5_May     5 Catalina surface        surface
2 Catalina_20_S23_L004 CA_Catalina_5_May     5 Catalina surface        surface
3 Catalina_21_S24_L004 CA_Catalina_5_May     5 Catalina surface        surface
4 Catalina_22_S25_L004 CA_Catalina_5_May     5 Catalina surface        surface
5 Catalina_23_S26_L004 CA_Catalina_5_May     5 Catalina surface        surface
6 Catalina_24_S27_L004 CA_Catalina_5_May     5 Catalina surface        surface
    REGION PACIFIC_REGION MONTH    LIGHT            SUPERGROUP
1 Catalina             CA   May Euphotic Alveolata-Apicomplexa
2 Catalina             CA   May Euphotic Alveolata-Apicomplexa
3 Catalina             CA   May Euphotic Alveolata-Apicomplexa
4 Catalina             CA   May Euphotic Alveolata-Apicomplexa
5 Catalina             CA   May Euphotic Alveolata-Apicomplexa
6 Catalina             CA   May Euphotic Alveolata-Apicomplexa</code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_formatted<span class="sc">$</span>SUPERGROUP)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Alveolata-Apicomplexa"    "Alveolata-Ciliophora"    
 [3] "Alveolata-Dinoflagellata" "Alveolata-Perkinsea"     
 [5] "Alveolata-Unclassified"   "Amoebozoa"               
 [7] "Chlorophyte"              "Excavata"                
 [9] "Hacrobia"                 "Opisthokonta"            
[11] "Rhizaria"                 "Stramenopiles-Diatom"    
[13] "Stramenopiles-Other"      "Unclassified"            </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>tax_interest <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Alveolata-Ciliophora"</span>, <span class="st">"Alveolata-Dinoflagellata"</span>, <span class="st">"Chlorophyte"</span>, <span class="st">"Hacrobia"</span>, <span class="st">"Rhizaria"</span>,<span class="st">"Stramenopiles-Diatom"</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>pink_qual <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#fff7f3"</span>, <span class="st">"#fcc5c0"</span>, <span class="st">"#fa9fb5"</span>,<span class="st">"#f768a1"</span>,<span class="st">"#dd3497"</span>, <span class="st">"#ae017e"</span>, <span class="st">"#7a0177"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>tax_formatted <span class="sc">%&gt;%</span> </span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">factor</span>(tax_formatted<span class="sc">$</span>SUPERGROUP, <span class="at">levels =</span> supergroup_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SUPERGROUP <span class="sc">%in%</span> tax_interest) <span class="sc">%&gt;%</span> </span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE_ID_PARSED, PACIFIC_REGION, LIGHT, DEPTH, SUPERGROUP, Class, Order) <span class="sc">%&gt;%</span> </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ID_PARSED, <span class="at">y =</span> Order)) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"white"</span>, <span class="fu">aes</span>(<span class="at">fill =</span> <span class="fu">log</span>(SUM))) <span class="sc">+</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(PACIFIC_REGION), <span class="at">rows =</span> <span class="fu">vars</span>(SUPERGROUP), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>      <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> pink_qual) <span class="sc">+</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># coord_flip() +</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Order"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE_ID_PARSED', 'PACIFIC_REGION',
'LIGHT', 'DEPTH', 'SUPERGROUP', 'Class'. You can override using the `.groups`
argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="480"></p>
</div>
</div>
</section>
<section id="table-1" class="level4">
<h4 class="anchored" data-anchor-id="table-1">Table 1</h4>
<p>Get stats on what transcripts were annotated to various levels, etc.</p>
<p>This represents a toy dataset to troubleshoot code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/toy_1000.Rdata"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Change to original names, so code below can be written</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>annotated_mean <span class="ot">&lt;-</span> annotated_mean_10000</span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>df_ctr_norm <span class="ot">&lt;-</span> ctr_norm_10000</span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annotated_mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>How many transcripts are annotated?</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>             <span class="fu">count</span>(<span class="at">output =</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"KEGGs"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"PFAMs"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> GOs <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"GOs"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> (GOs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>), </span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"All"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(output <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>output) <span class="sc">%&gt;%</span> </span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">values_to =</span> <span class="st">"Number annotated"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent_annotated =</span> <span class="dv">100</span><span class="sc">*</span>(<span class="st">`</span><span class="at">Number annotated</span><span class="st">`</span><span class="sc">/</span><span class="fu">length</span>(annotated_mean<span class="sc">$</span>SequenceID)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<thead>
<tr class="header">
<th>DB</th>
<th>Number of seqs annotated</th>
<th>Percent annotated</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KEGGs</td>
<td>9970784</td>
<td>67.7</td>
<td></td>
</tr>
<tr class="even">
<td>PFAMs</td>
<td>13595566</td>
<td>92.3</td>
<td></td>
</tr>
<tr class="odd">
<td>GOs</td>
<td>5442486</td>
<td>36.9</td>
<td></td>
</tr>
<tr class="even">
<td>All</td>
<td>4963542</td>
<td>33.7</td>
<td></td>
</tr>
</tbody>
</table>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a>summary_annotated <span class="ot">&lt;-</span> annotated_mean <span class="sc">%&gt;%</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUM =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric)))) <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SUM, GOs, PFAMs, KEGG_ko, Domain, Supergroup)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(summary_annotated, file = "summary_of_annotated_totalTPM.csv")</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sum(summary_annotated$SUM) # TPM amount that is annotated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>How many are annotated and have taxonomic IDs?</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>noneuk <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Unclassified"</span>, <span class="st">"Bacteria"</span>, <span class="st">"Archaea"</span>)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">as.data.frame</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>             <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk), </span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"IDed as Eukaryote"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Supergroup <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Supergroup level"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Class <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Class level"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Order <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Order level"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>            <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Family <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Family level"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>            <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Genus <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Genus level"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a>              <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> <span class="sc">!</span>(Domain <span class="sc">%in%</span> noneuk) <span class="sc">&amp;</span> Species <span class="sc">!=</span> <span class="st">"Unclassified"</span>, </span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"Species level"</span>))) <span class="sc">%&gt;%</span>  </span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(output <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>output) <span class="sc">%&gt;%</span> </span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">values_to =</span> <span class="st">"Number annotated"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent_annotated =</span> <span class="dv">100</span><span class="sc">*</span>(<span class="st">`</span><span class="at">Number annotated</span><span class="st">`</span><span class="sc">/</span><span class="fu">length</span>(annotated_mean<span class="sc">$</span>SequenceID)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Need to run the above on the HPC.</p>
</blockquote>
</section>
</section>
</section>
<section id="pcoa-with-whole-community-functional-profile" class="level2">
<h2 class="anchored" data-anchor-id="pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</h2>
<p>Ahead of time, normalized community composition data was center scaled. Below, we can import this dataset and perform principle component analysis.</p>
<p>Import normalized center data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run on HPC</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"normed_center_df_08222023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run on HPC</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>pca_lr <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">as.data.frame</span>(<span class="fu">t</span>(df_ctr_norm)))</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>variance_lr <span class="ot">&lt;-</span> (pca_lr<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span><span class="fu">sum</span>(pca_lr<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Import PCA objects from running R on the HPC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/pca_objects.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  pca_df
  variance_lr</code></pre>
</div>
</div>
<p>Determine how appropriate the PCoA is in 2 vs 3 dimensions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(variance_lr, <span class="at">main =</span> <span class="st">"Log-Ratio PCA Screeplot"</span>, <span class="at">xlab =</span> <span class="st">"PC Axis"</span>, <span class="at">ylab =</span> <span class="st">"% Variance"</span>, <span class="at">cex.names =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>pc1 <span class="ot">&lt;-</span> <span class="fu">round</span>(variance_lr[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>pc2 <span class="ot">&lt;-</span> <span class="fu">round</span>(variance_lr[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(sample_metadata)</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="co"># svg("Rplot_metat_pca_wholecommunity.svg", width = 7, height = 7)</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>pca_df <span class="sc">%&gt;%</span> </span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"SAMPLE_MATCH"</span> <span class="ot">=</span> <span class="st">"SAMPLE"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGION_ORDER =</span> <span class="fu">case_when</span>(REGION <span class="sc">==</span> <span class="st">"ALOHA"</span> <span class="sc">~</span> <span class="fu">paste</span>(<span class="st">"ALOHA"</span>, MONTH), <span class="cn">TRUE</span> <span class="sc">~</span> REGION)) <span class="sc">%&gt;%</span> </span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2)) <span class="sc">+</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_jitter</span>(<span class="at">stroke =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> REGION_ORDER, <span class="at">fill =</span> DEPTH_CATEGORY, <span class="at">color =</span> LIGHT)) <span class="sc">+</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_shape_manual(values = c(21, 22, 23, 24)) +</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">24</span>, <span class="dv">25</span>, <span class="dv">23</span>, <span class="dv">22</span>, <span class="dv">21</span>)) <span class="sc">+</span></span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#6a51a3"</span>, <span class="st">"#f768a1"</span>, <span class="st">"#8c96c6"</span>, <span class="st">"#fd8d3c"</span>, <span class="st">"#238b45"</span>)) <span class="sc">+</span></span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#cccccc"</span>, <span class="st">"#252525"</span>)) <span class="sc">+</span></span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="fu">c</span>(<span class="dv">22</span>)))) <span class="sc">+</span></span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">13</span>)) <span class="sc">+</span></span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span>  <span class="fu">paste</span>(pc1, <span class="st">"%"</span>), <span class="at">y =</span> <span class="fu">paste</span>(pc2, <span class="st">"%"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="untargeted---de-transcripts" class="level2">
<h2 class="anchored" data-anchor-id="untargeted---de-transcripts">Untargeted - DE transcripts</h2>
</section>
</section>
<section id="prep-to-address-key-questions" class="level1">
<h1>Prep to address key questions</h1>
<section id="import-kegg-ids" class="level2">
<h2 class="anchored" data-anchor-id="import-kegg-ids">Import KEGG IDs</h2>
<p>These have been curated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../KEGG_DB/combined_kegg.csv"</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co"># head(kegg$KO_number)</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>curated_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../KEGG_DB/reformat-kegg-pfam-skh.csv"</span>)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>key_geneid <span class="ot">&lt;-</span> curated_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>X) <span class="sc">%&gt;%</span> </span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(kegg <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">KEGG =</span> KO_number, <span class="fu">everything</span>(), <span class="sc">-</span>X)) <span class="sc">%&gt;%</span> </span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"KeggOrthology_"</span>), Category01, Category02, FullName, GeneID, Gene_identification, KEGG, PFAM, Descriptions, <span class="at">REF =</span> REFs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in right_join(., kegg %&gt;% select(KEGG = KO_number, everything(), : Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 38 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
</div>
<p>Subset key_geneid dataframe to select what we want from it.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(key_geneid)</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(key_geneid$KeggOrthology_B)</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>kegg_ortho_based <span class="ot">&lt;-</span> key_geneid <span class="sc">%&gt;%</span> </span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(KeggOrthology_B, KEGG, GeneID, Gene_identification) <span class="sc">%&gt;%</span> </span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(KeggOrthology_B)) <span class="sc">%&gt;%</span> </span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(key_geneid)</span></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(kegg_ortho_based)</span></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>kegg_curated <span class="ot">&lt;-</span> key_geneid <span class="sc">%&gt;%</span> </span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Category01, Category02, KEGG, GeneID, Gene_identification) <span class="sc">%&gt;%</span> </span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(Category01)) <span class="sc">%&gt;%</span> </span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb55-17"><a href="#cb55-17" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(kegg_curated)</span></span>
<span id="cb55-18"><a href="#cb55-18" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(kegg_curated$Category01)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Create ordered list for genes related to N and P metabolism.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(curated_kegg)</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>geneid_order_nuts <span class="ot">&lt;-</span> <span class="fu">as.character</span>(curated_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"N and P metabolism"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ORDER"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ORDER, GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(<span class="fu">data.frame</span>(<span class="at">ORDER =</span> <span class="dv">0</span>, <span class="at">GeneID =</span> <span class="st">"Mean"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">as.numeric</span>(ORDER)) <span class="sc">%&gt;%</span> </span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(GeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"GeneID"</span>]])</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>geneid_order_nuts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Mean"                  "NRT, narK, nrtP, nasA" "amt, AMT, MEP"        
 [4] "AAPJ"                  "NR"                    "nirA"                 
 [7] "HCP"                   "NPD"                   "CYNS"                 
[10] "glnA, GLUL"            "GLT1"                  "GLT(ferredoxin)"      
[13] "GLTD"                  "gdhA"                  "gdh2"                 
[16] "CPS1"                  "OTC, argF, argI"       "argG, ASS1"           
[19] "argH, ASL"             "rocF, arg"             "URE"                  
[22] "ureG"                  "ureD, ureH"            "UT"                   
[25] "DUR3"                  "PHO87_91"              "PTA"                  
[28] "NPT"                   "ARSA"                  "5-nucleotidase"       
[31] "phoD"                  "GDP"                   "NTD"                  
[34] "VTC"                   "arsC"                 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>geneid_order_hets <span class="ot">&lt;-</span> <span class="fu">as.character</span>(curated_kegg <span class="sc">%&gt;%</span> </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"Phagotrophy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"ORDER"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(ORDER, Category02) <span class="sc">%&gt;%</span> </span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbind</span>(<span class="fu">data.frame</span>(<span class="at">ORDER =</span> <span class="dv">0</span>, <span class="at">Category02 =</span> <span class="st">"Mean"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">as.numeric</span>(ORDER)) <span class="sc">%&gt;%</span>  </span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Category02) <span class="sc">%&gt;%</span> <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"Category02"</span>]]) </span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>geneid_order_hets</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Mean"                            "Actin polymerization"           
 [3] "Chitinase"                       "Endocytosis"                    
 [5] "Fatty acid biosynthesis"         "Fatty acid breakdown"           
 [7] "Lysosome binding and processing" "Motility and prey recognition"  
 [9] "Phagosome maturation"            "SNARE complex"                  
[11] "V-type ATPase"                  </code></pre>
</div>
</div>
</section>
</section>
<section id="prepare-data-frames-for-targeted-approach" class="level1">
<h1>Prepare data frames for targeted approach</h1>
<p>Data for targeted sections is imported from <code>process-eukrhythmic-output</code> code titled <em>Curated genes only</em>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/Avg_scaled_tpm_curated_08252023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  counts_curated
  sample_merged</code></pre>
</div>
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(counts_curated)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To address question 1, subset to a curated set of genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>counts_curated_annot <span class="ot">&lt;-</span> counts_curated <span class="sc">%&gt;%</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG_ko, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg_curated)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., kegg_curated): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 5 of `x` matches multiple rows in `y`.
ℹ Row 277 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>counts_curated_kegg <span class="ot">&lt;-</span> counts_curated <span class="sc">%&gt;%</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG_ko, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg_ortho_based)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., kegg_ortho_based): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 614 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(ctr_norm_curated)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/normed_center_df_curated_08252023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  ctr_norm_curated
  ds_tpm_curated</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ctr_norm_curated)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                      SequenceID
1  megahit_ALOHA_July_5m_k51_1000052_flag=0_multi=3.5000_len=279_ALOHA_July_5m.filtered.fasta.p1
2  megahit_ALOHA_July_5m_k51_1000134_flag=1_multi=8.0000_len=201_ALOHA_July_5m.filtered.fasta.p1
3 megahit_ALOHA_July_5m_k51_1000527_flag=1_multi=13.0000_len=230_ALOHA_July_5m.filtered.fasta.p1
4 megahit_ALOHA_July_5m_k51_1000562_flag=1_multi=13.9730_len=236_ALOHA_July_5m.filtered.fasta.p1
5 megahit_ALOHA_July_5m_k51_1000680_flag=1_multi=14.0000_len=771_ALOHA_July_5m.filtered.fasta.p1
6 megahit_ALOHA_July_5m_k51_1000685_flag=0_multi=16.2025_len=372_ALOHA_July_5m.filtered.fasta.p1
  SPOT_150m_Rep1and2 SPOT_150m_Rep3and4 SPOT_surface_11 SPOT_surface_12
1           -0.66862           -0.66862        -0.66862        -0.66862
2           -0.66862           -0.66862        -0.66862        -0.66862
3           -0.66862           -0.66862        -0.66862        -0.66862
4           -0.66862           -0.66862        -0.66862        -0.66862
5           -0.66862           -0.66862        -0.66862         2.08898
6           -0.66862           -0.66862        -0.66862        -0.66862
  SPOT_surface_9 SPOT_surface_10 SPOT_surface_7 SPOT_surface_8
1       -0.66862        -0.66862       -0.66862       -0.66862
2       -0.66862        -0.66862       -0.66862       -0.66862
3       -0.66862        -0.66862       -0.66862       -0.66862
4       -0.66862        -0.66862       -0.66862       -0.66862
5       -0.66862        -0.66862       -0.66862       -0.66862
6       -0.66862        -0.66862       -0.66862       -0.66862
  SPOT_150m_Rep5and6 SPOT_890m_Rep1and2 SPOT_890m_Rep7 SPOT_890m_Rep3and4
1           -0.66862           -0.66862       -0.66862           -0.66862
2           -0.66862           -0.66862       -0.66862           -0.66862
3           -0.66862           -0.66862       -0.66862           -0.66862
4           -0.66862           -0.66862       -0.66862           -0.66862
5           -0.66862           -0.66862       -0.66862           -0.66862
6           -0.66862           -0.66862       -0.66862           -0.66862
  SPOT_890m_Rep5and6 March_5m_Rep1 March_150m_Rep1andRep2 July_DCM_Rep2
1           -0.66862      -0.66862               -0.66862      -0.66862
2           -0.66862      -0.66862               -0.66862      -0.66862
3           -0.66862      -0.66862               -0.66862      -0.66862
4           -0.66862      -0.66862               -0.66862      -0.66862
5           -0.66862      -0.66862               -0.66862      -0.66862
6           -0.66862      -0.66862               -0.66862      -0.66862
  July_DCM_Rep1 July_5m_Rep1 July_1000m_Rep1andRep2 March_DCM_Rep1
1      -0.66862     1.769635               -0.66862       -0.66862
2      -0.66862     2.881847               -0.66862       -0.66862
3      -0.66862    -0.668620               -0.66862       -0.66862
4      -0.66862     3.889006               -0.66862       -0.66862
5      -0.66862     5.229919               -0.66862       -0.66862
6      -0.66862     5.464715               -0.66862       -0.66862
  March_5m_Rep2 March_1000m_Rep1andRep2 July_5m_Rep2 July_150m_Rep1andRep2
1      -0.66862                -0.66862    -0.668620              -0.66862
2      -0.66862                -0.66862    -0.668620              -0.66862
3      -0.66862                -0.66862     4.179680              -0.66862
4      -0.66862                -0.66862     5.305635              -0.66862
5      -0.66862                -0.66862     3.719440              -0.66862
6      -0.66862                -0.66862     3.034110              -0.66862
  March_DCM_Rep2 PortofLA_4 PortofLA_3 PortofLA_2 PortofLA_1 Catalina_24
1       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862   -0.668620
2       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862   -0.668620
3       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862   -0.668620
4       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862    0.719916
5       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862   -0.668620
6       -0.66862   -0.66862   -0.66862   -0.66862   -0.66862   -0.668620
  Catalina_23 Catalina_22 Catalina_21 SPOT_surface_18 SPOT_surface_17
1    -0.66862   -0.668620    -0.66862        -0.66862        -0.66862
2    -0.66862   -0.668620    -0.66862        -0.66862        -0.66862
3    -0.66862   -0.668620    -0.66862        -0.66862        -0.66862
4    -0.66862   -0.668620    -0.66862        -0.66862        -0.66862
5    -0.66862   -0.668620    -0.66862        -0.66862        -0.66862
6    -0.66862    2.162123    -0.66862        -0.66862        -0.66862
  SPOT_surface_16 SPOT_surface_!5 SPOT_surface_14 SPOT_surface_13 PortofLA_6
1        -0.66862      -0.6686200        -0.66862        1.068627   -0.66862
2        -0.66862      -0.6686200        -0.66862       -0.668620   -0.66862
3        -0.66862      -0.6686200        -0.66862       -0.668620   -0.66862
4         0.76628       0.7393821        -0.66862        1.330602   -0.66862
5        -0.66862      -0.6686200        -0.66862       -0.668620   -0.66862
6        -0.66862      -0.6686200        -0.66862       -0.668620   -0.66862
  PortofLA_5 Catalina_20 Catalina_19  X rownames   ShortSeqID    Domain
1   -0.66862    -0.66862    -0.66862  4        3 Seq_46717283 Eukaryota
2   -0.66862    -0.66862    -0.66862 13       12  Seq_5362361 Eukaryota
3   -0.66862    -0.66862    -0.66862 47       46  Seq_5968972 Eukaryota
4   -0.66862    -0.66862    -0.66862 52       51 Seq_38019630 Eukaryota
5   -0.66862    -0.66862    -0.66862 57       56 Seq_42359935 Eukaryota
6   -0.66862    -0.66862    -0.66862 59       58 Seq_22294350 Eukaryota
     Supergroup         Phylum            Class                Order
1      Hacrobia     Haptophyta Prymnesiophyceae         Prymnesiales
2     Alveolata Dinoflagellata      Dinophyceae       Prorocentrales
3     Alveolata Dinoflagellata      Dinophyceae           Suessiales
4      Rhizaria   Foraminifera    Globothalamea      Globothalamea_X
5 Stramenopiles     Ochrophyta  Bacillariophyta Chaetocerotophycidae
6     Alveolata     Ciliophora     Spirotrichea           Tintinnida
                Family            Genus                  Species
1 Chrysochromulinaceae Chrysochromulina Chrysochromulina ericina
2      Prorocentraceae     Prorocentrum             Unclassified
3          Suessiaceae     Unclassified             Unclassified
4     Globothalamea_XX     Unclassified             Unclassified
5      Chaetocerotales      Chaetoceros             Unclassified
6         Unclassified     Unclassified             Unclassified
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      GOs
1 GO:0002376,GO:0003674,GO:0003824,GO:0004611,GO:0004612,GO:0005575,GO:0005622,GO:0005623,GO:0005634,GO:0005730,GO:0005737,GO:0005829,GO:0005975,GO:0005996,GO:0006006,GO:0006082,GO:0006083,GO:0006094,GO:0006950,GO:0006952,GO:0006955,GO:0007154,GO:0008150,GO:0008152,GO:0009058,GO:0009267,GO:0009605,GO:0009607,GO:0009620,GO:0009814,GO:0009817,GO:0009987,GO:0009991,GO:0010035,GO:0010038,GO:0015976,GO:0016020,GO:0016036,GO:0016051,GO:0016829,GO:0016830,GO:0016831,GO:0016999,GO:0017144,GO:0019318,GO:0019319,GO:0019752,GO:0031667,GO:0031668,GO:0031669,GO:0031974,GO:0031981,GO:0032787,GO:0033554,GO:0042221,GO:0042594,GO:0043207,GO:0043226,GO:0043227,GO:0043228,GO:0043229,GO:0043231,GO:0043232,GO:0043233,GO:0043436,GO:0044237,GO:0044238,GO:0044281,GO:0044283,GO:0044422,GO:0044424,GO:0044428,GO:0044444,GO:0044446,GO:0044464,GO:0045087,GO:0046364,GO:0046686,GO:0050832,GO:0050896,GO:0051704,GO:0051707,GO:0051716,GO:0070013,GO:0071496,GO:0071704,GO:0098542,GO:1901576
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       -
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       -
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       -
5                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       -
6                                                                                                                                                                                                                                                                                                                                                                                                                                                         GO:0000323,GO:0003674,GO:0003824,GO:0004175,GO:0004197,GO:0005575,GO:0005576,GO:0005615,GO:0005622,GO:0005623,GO:0005737,GO:0005764,GO:0005773,GO:0006508,GO:0006807,GO:0008150,GO:0008152,GO:0008233,GO:0008234,GO:0009056,GO:0009057,GO:0009987,GO:0016787,GO:0019538,GO:0030163,GO:0043170,GO:0043226,GO:0043227,GO:0043229,GO:0043231,GO:0044237,GO:0044238,GO:0044248,GO:0044257,GO:0044260,GO:0044265,GO:0044267,GO:0044421,GO:0044424,GO:0044444,GO:0044464,GO:0051603,GO:0070011,GO:0071704,GO:0140096,GO:1901564,GO:1901565,GO:1901575
                       PFAMs   KEGG_ko
1                  PEPCK_ATP ko:K01610
2                        Mqo ko:K00116
3    Acyl_transf_1,Ank,Ank_2 ko:K00645
4                      Actin ko:K05692
5               Calreticulin ko:K08057
6 Inhibitor_I29,Peptidase_C1 ko:K01365</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename column headers</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>names_orig <span class="ot">&lt;-</span> <span class="fu">colnames</span>(ctr_norm_curated)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>names_new <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_[^_]+$"</span>, <span class="st">""</span>, names_orig)</span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="co"># names_new</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(ctr_norm_curated) <span class="ot">&lt;-</span> names_new</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean across columns that have the same name - which are replicates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(ctr_norm_curated)</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a>mean_ctr_df <span class="ot">&lt;-</span> ctr_norm_curated <span class="sc">%&gt;%</span></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(<span class="fu">as.list</span>(.) <span class="sc">%&gt;%</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Filter</span>(is.numeric, .) <span class="sc">%&gt;%</span></span>
<span id="cb75-5"><a href="#cb75-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(<span class="fu">names</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb75-6"><a href="#cb75-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(as.data.frame) <span class="sc">%&gt;%</span></span>
<span id="cb75-7"><a href="#cb75-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(rowMeans) <span class="sc">%&gt;%</span></span>
<span id="cb75-8"><a href="#cb75-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setNames</span>(<span class="fu">paste0</span>(<span class="st">"mean."</span>, <span class="fu">names</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb75-9"><a href="#cb75-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean"</span>), SequenceID, ShortSeqID, Domain<span class="sc">:</span>Species, GOs, PFAMs, KEGG) <span class="sc">%&gt;%</span> </span>
<span id="cb75-10"><a href="#cb75-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>mean.rownames, <span class="sc">-</span>mean.X)</span>
<span id="cb75-11"><a href="#cb75-11" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(mean_ctr_df)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Join with KEGG information that is curated</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat with ctr_norm_curated</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>ctr_curated_annot <span class="ot">&lt;-</span> mean_ctr_df <span class="sc">%&gt;%</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg_curated)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., kegg_curated): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 5 of `x` matches multiple rows in `y`.
ℹ Row 277 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>ctr_curated_kegg <span class="ot">&lt;-</span> mean_ctr_df <span class="sc">%&gt;%</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(kegg_ortho_based)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., kegg_ortho_based): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 614 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
</div>
<p>Use ‘mean_ctr_df’ for downstream targeted analyses.</p>
</section>
<section id="question-1" class="level1">
<h1>Question 1</h1>
<blockquote class="blockquote">
<p>How does nutrient utilization among euphotic zone phytoplankton vary between coastal California and the NPSG?</p>
</blockquote>
<section id="untargeted" class="level2">
<h2 class="anchored" data-anchor-id="untargeted">Untargeted</h2>
<p>Complete code: <code>/scripts/get_euphotic_DE_bytaxa.R</code></p>
<p>Use output files below for further analysis below.</p>
<section id="stats-on-de-transcripts" class="level3">
<h3 class="anchored" data-anchor-id="stats-on-de-transcripts">Stats on DE transcripts</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>de_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"input-data/bytaxa_euphoticzone_DE.csv"</span>)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>de_all</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Example code for compiling DE transcripts by individual taxa (from script):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"euphotic_by_taxa.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>res_hapto <span class="ot">&lt;-</span> <span class="fu">results</span>(de_hapto, <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>plot_hapto <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res_hapto) <span class="sc">%&gt;%</span> </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGULATION =</span> <span class="fu">case_when</span>(</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in CA"</span>,</span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in NPSG"</span></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">SIGNIFICANT =</span> <span class="fu">case_when</span>(</span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>    pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Significantly"</span>,</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Not significantly"</span></span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> baseMean, <span class="at">y =</span> log2FoldChange, <span class="at">color =</span> SIGNIFICANT)) <span class="sc">+</span></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#878787"</span>, <span class="st">"#d73027"</span>)) <span class="sc">+</span></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">mcols</span>(res_hapto)<span class="sc">$</span>description[<span class="dv">2</span>])</span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>hapto_transcripts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res_hapto) <span class="sc">%&gt;%</span> </span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGULATION =</span> <span class="fu">case_when</span>(</span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in CA"</span>,</span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in NPSG"</span></span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb84-24"><a href="#cb84-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">SIGNIFICANT =</span> <span class="fu">case_when</span>(</span>
<span id="cb84-25"><a href="#cb84-25" aria-hidden="true" tabindex="-1"></a>    pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Significantly"</span>,</span>
<span id="cb84-26"><a href="#cb84-26" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Not significantly"</span></span>
<span id="cb84-27"><a href="#cb84-27" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb84-28"><a href="#cb84-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SIGNIFICANT <span class="sc">==</span> <span class="st">"Significantly"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb84-29"><a href="#cb84-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SequenceID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb84-30"><a href="#cb84-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TAXA =</span> <span class="st">"Haptophyte"</span>)</span>
<span id="cb84-31"><a href="#cb84-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-32"><a href="#cb84-32" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(hapto_transcripts, <span class="at">file =</span> <span class="st">"hapto_sig_de_euphotic.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>See script: <code>euphotic_region_bytax.R</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># List of DE transcripts - what is upregulated vs downregulated by taxa</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>de_all <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"bytaxa_euphoticzone_DE.csv"</span>)</span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>de_chloro <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"chloro_sig_de_euphotic.csv"</span>)</span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>de_dino <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"dinos_sig_de_euphotic.csv"</span>)</span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>de_diatom <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"diatom_sig_de_euphotic.csv"</span>)</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>de_hapto <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"hapto_sig_de_euphotic.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Sequence IDs are all unique, because they correspond to taxa. Therefore, we need to include gene-level IDs so we can compare what is DE across CA vs.&nbsp;NPSG.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import KEGG information</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>taxfxn <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"TaxonomicAndFunctionalAnnotations.csv"</span>)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>alltax <span class="ot">&lt;-</span> de_chloro <span class="sc">%&gt;%</span> </span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(de_diatom) <span class="sc">%&gt;%</span> </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(de_dino) <span class="sc">%&gt;%</span> </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(de_hapto) <span class="sc">%&gt;%</span> </span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SequenceID, log2FoldChange, pvalue, REGULATION, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This <code>alltax</code> Robj is helpful for coming back to getting more taxonomic resolution.</p>
<p>What is shared versus unique?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>alltax_compareDE <span class="ot">&lt;-</span> alltax <span class="sc">%&gt;%</span> </span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TAXA, <span class="at">values_from =</span> n)</span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(alltax_compareDE, alltax, <span class="at">file =</span> <span class="st">"taxa-untarget-q1.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="local---untargeted-mining" class="level3">
<h3 class="anchored" data-anchor-id="local---untargeted-mining">Local - untargeted mining</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/taxa-untarget-q1.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggupset)</span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>alltax <span class="sc">%&gt;%</span> </span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb89-8"><a href="#cb89-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb89-9"><a href="#cb89-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENEID_ko_pfam, REGULATION) <span class="sc">%&gt;%</span> </span>
<span id="cb89-10"><a href="#cb89-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">TAXA =</span> <span class="fu">list</span>(TAXA)) <span class="sc">%&gt;%</span> </span>
<span id="cb89-11"><a href="#cb89-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> TAXA)) <span class="sc">+</span></span>
<span id="cb89-12"><a href="#cb89-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> REGULATION)) <span class="sc">+</span></span>
<span id="cb89-13"><a href="#cb89-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#74add1"</span>, <span class="st">"#fdae61"</span>)) <span class="sc">+</span></span>
<span id="cb89-14"><a href="#cb89-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_upset</span>(<span class="at">order_by =</span> <span class="st">"freq"</span>) <span class="sc">+</span></span>
<span id="cb89-15"><a href="#cb89-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb89-16"><a href="#cb89-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb89-17"><a href="#cb89-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'GENEID_ko_pfam'. You can override using
the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-44-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Based on this plot, it looks like these taxa primarily have their own strategies in the coastal California environment versus the NPSG environment.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(alltax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                                      SequenceID
1   megahit_ALOHA_July_5m_k51_100623_flag=1_multi=8.4034_len=574_ALOHA_July_5m.filtered.fasta.p1
2 megahit_ALOHA_July_5m_k51_1008920_flag=0_multi=17.7732_len=342_ALOHA_July_5m.filtered.fasta.p1
3 megahit_ALOHA_July_5m_k51_1052293_flag=1_multi=40.0969_len=505_ALOHA_July_5m.filtered.fasta.p1
4  megahit_ALOHA_July_5m_k51_108347_flag=0_multi=37.3628_len=820_ALOHA_July_5m.filtered.fasta.p1
5 megahit_ALOHA_July_5m_k51_112715_flag=1_multi=108.5316_len=557_ALOHA_July_5m.filtered.fasta.p1
6 megahit_ALOHA_July_5m_k51_1130367_flag=1_multi=22.0000_len=906_ALOHA_July_5m.filtered.fasta.p2
  log2FoldChange       pvalue        REGULATION        TAXA     X rownames
1       5.463782 2.766358e-02 upregulated in CA Chlorophyte   690      689
2       3.609570 2.497255e-02 upregulated in CA Chlorophyte  1005     1004
3       7.575613 7.550389e-03 upregulated in CA Chlorophyte  5811     5810
4       5.558651 3.243688e-03 upregulated in CA Chlorophyte  9239     9238
5      25.186544 6.509685e-15 upregulated in CA Chlorophyte 14151    14150
6       2.703206 4.347091e-03 upregulated in CA Chlorophyte 14507    14506
    ShortSeqID    Domain     Supergroup      Phylum                Class
1 Seq_48897575 Eukaryota Archaeplastida Chlorophyta        Chlorophyceae
2 Seq_13728792 Eukaryota Archaeplastida Chlorophyta         Unclassified
3 Seq_13623672 Eukaryota Archaeplastida Chlorophyta  Chlorodendrophyceae
4 Seq_45857102 Eukaryota Archaeplastida Chlorophyta      Pyramimonadales
5  Seq_5315521 Eukaryota Archaeplastida Chlorophyta      Pyramimonadales
6 Seq_16589800 Eukaryota Archaeplastida Chlorophyta Nephroselmidophyceae
              Order              Family        Genus      Species
1 Chlamydomonadales Chlamydomonadales_X Unclassified Unclassified
2      Unclassified        Unclassified Unclassified Unclassified
3   Chlorodendrales    Chlorodendraceae  Tetraselmis Unclassified
4 Pyramimonadales_X        Unclassified Unclassified Unclassified
5 Pyramimonadales_X  Pyramimonadales_XX Unclassified Unclassified
6      Unclassified        Unclassified Unclassified Unclassified
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       GOs
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  GO:0005575,GO:0005623,GO:0005929,GO:0031514,GO:0042995,GO:0043226,GO:0044464,GO:0120025
2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        -
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        -
4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        -
5 GO:0000028,GO:0000054,GO:0000056,GO:0002181,GO:0003674,GO:0003735,GO:0005198,GO:0005575,GO:0005622,GO:0005623,GO:0005737,GO:0005829,GO:0005840,GO:0006403,GO:0006405,GO:0006407,GO:0006412,GO:0006518,GO:0006611,GO:0006807,GO:0006810,GO:0006886,GO:0006913,GO:0006996,GO:0008104,GO:0008150,GO:0008152,GO:0009058,GO:0009059,GO:0009987,GO:0010467,GO:0015031,GO:0015833,GO:0015931,GO:0015935,GO:0016043,GO:0019538,GO:0022607,GO:0022613,GO:0022618,GO:0022626,GO:0022627,GO:0031503,GO:0032991,GO:0033036,GO:0033750,GO:0034613,GO:0034622,GO:0034641,GO:0034645,GO:0042254,GO:0042255,GO:0042274,GO:0042886,GO:0043043,GO:0043170,GO:0043226,GO:0043228,GO:0043229,GO:0043232,GO:0043603,GO:0043604,GO:0043933,GO:0044085,GO:0044237,GO:0044238,GO:0044249,GO:0044260,GO:0044267,GO:0044271,GO:0044391,GO:0044422,GO:0044424,GO:0044444,GO:0044445,GO:0044446,GO:0044464,GO:0045184,GO:0046907,GO:0050657,GO:0050658,GO:0051029,GO:0051168,GO:0051169,GO:0051179,GO:0051234,GO:0051236,GO:0051640,GO:0051641,GO:0051649,GO:0051656,GO:0065003,GO:0070727,GO:0070925,GO:0071166,GO:0071426,GO:0071428,GO:0071702,GO:0071704,GO:0071705,GO:0071826,GO:0071840,GO:0097064,GO:1901564,GO:1901566,GO:1901576,GO:1990904
6                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        -
                                                        PFAMs
1 AAA_5,AAA_6,AAA_7,AAA_8,AAA_9,DHC_N1,DHC_N2,Dynein_heavy,MT
2                                               Ribosomal_60s
3                                               Ribosomal_L14
4                                                         NMO
5                                               Ribosomal_S19
6                          GGDEF,HATPase_c,HisKA,cNMP_binding
              KEGG_ko
1           ko:K10408
2           ko:K02942
3           ko:K02894
4                   -
5           ko:K02958
6 ko:K01420,ko:K10914</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(alltax_compareDE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  REGULATION        GENEID_ko_pfam Dinoflagellate Chlorophyte Diatom Haptophyte
  &lt;chr&gt;             &lt;chr&gt;                   &lt;int&gt;       &lt;int&gt;  &lt;int&gt;      &lt;int&gt;
1 upregulated in CA -;ko:K00241                 3          NA     NA         NA
2 upregulated in CA -;ko:K00542                 2          NA     NA         NA
3 upregulated in CA -;ko:K01137                 4          NA     NA         NA
4 upregulated in CA -;ko:K01805                 7          NA     NA         NA
5 upregulated in CA -;ko:K07374                 2          NA     NA         NA
6 upregulated in CA -;ko:K08906                NA           5     NA         NA</code></pre>
</div>
</div>
<section id="what-transcripts-are-shared-across-all-taxa-vs.-unique" class="level4">
<h4 class="anchored" data-anchor-id="what-transcripts-are-shared-across-all-taxa-vs.-unique">What transcripts are shared across all taxa vs.&nbsp;unique?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a>tax_indiv <span class="ot">&lt;-</span> alltax_compareDE <span class="sc">%&gt;%</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CATEGORY =</span> <span class="fu">case_when</span>(</span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>                     (<span class="fu">is.na</span>(Chlorophyte) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Diatom) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Dinoflagellate only"</span>,</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Diatom) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Chlorophyte only"</span>,</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Chlorophyte) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Diatom only"</span>,</span>
<span id="cb95-6"><a href="#cb95-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Chlorophyte) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Diatom)) <span class="sc">~</span> <span class="st">"Haptophyte only"</span>,</span>
<span id="cb95-7"><a href="#cb95-7" aria-hidden="true" tabindex="-1"></a>                   <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb95-8"><a href="#cb95-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(CATEGORY))</span>
<span id="cb95-9"><a href="#cb95-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb95-10"><a href="#cb95-10" aria-hidden="true" tabindex="-1"></a>shared_euphotic <span class="ot">&lt;-</span> alltax_compareDE <span class="sc">%&gt;%</span> </span>
<span id="cb95-11"><a href="#cb95-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb95-12"><a href="#cb95-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY =</span> <span class="st">"Shared across all taxa"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-13"><a href="#cb95-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(tax_indiv) <span class="sc">%&gt;%</span> </span>
<span id="cb95-14"><a href="#cb95-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(GENEID_ko_pfam, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"PFAMs"</span>, <span class="st">"KEGG_ko"</span>), <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb95-15"><a href="#cb95-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG_ko, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb95-16"><a href="#cb95-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(key_geneid) <span class="sc">%&gt;%</span> </span>
<span id="cb95-17"><a href="#cb95-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">COUNT =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., key_geneid): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 3188 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(shared_euphotic)</span></span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(alltax_compareDE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(shared_euphotic<span class="sc">$</span>KeggOrthology_A)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Environmental Information Processing"
[2] "Cellular Processes"                  
[3] "Genetic Information Processing"      
[4] NA                                    
[5] "Metabolism"                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(shared_euphotic<span class="sc">$</span>Category01)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] NA                           "Phagotrophy"               
[3] "Energy Acquisition"         "N and P metabolism"        
[5] "Nutrient processing"        "Carbon fixation"           
[7] "Gluconeogenesis-glycolysis" "Glyoxylate cycle"          
[9] "Photosynthesis"            </code></pre>
</div>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a>paths_ORDER <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Photosynthesis"</span>, <span class="st">"Energy Acquisition"</span>, <span class="st">"Carbon fixation"</span>, <span class="st">"Nutrient processing"</span>, <span class="st">"Gluconeogenesis-glycolysis"</span>, <span class="st">"N and P metabolism"</span>, <span class="st">"Glyoxylate cycle"</span>, <span class="st">"Phagotrophy"</span>, </span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Environmental Information Processing"</span>, <span class="st">"Cellular Processes"</span>,</span>
<span id="cb103-3"><a href="#cb103-3" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Genetic Information Processing"</span>, <span class="st">"Metabolism"</span>,</span>
<span id="cb103-4"><a href="#cb103-4" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Other"</span>)</span>
<span id="cb103-5"><a href="#cb103-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-6"><a href="#cb103-6" aria-hidden="true" tabindex="-1"></a>paths_COLOR <span class="ot">&lt;-</span><span class="fu">c</span> (<span class="st">"#f7fcb9"</span>, <span class="st">"#fdae6b"</span>, <span class="st">"#fc9272"</span>, <span class="st">"#7fcdbb"</span>, <span class="st">"#d9f0a3"</span>, <span class="st">"#78c679"</span>, <span class="st">"#bfd3e6"</span>, <span class="st">"#fde0dd"</span>,</span>
<span id="cb103-7"><a href="#cb103-7" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#fd8d3c"</span>, <span class="st">"#225ea8"</span>, </span>
<span id="cb103-8"><a href="#cb103-8" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#cb181d"</span>, <span class="st">"#41ab5d"</span>,</span>
<span id="cb103-9"><a href="#cb103-9" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#969696"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a>shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Category01 =</span> <span class="fu">str_replace_na</span>(Category01, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(CATEGORY, REGULATION, Category01) <span class="sc">%&gt;%</span> </span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">Category01_SUM =</span> <span class="fu">sum</span>(COUNT)) <span class="sc">%&gt;%</span> </span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY_TYPE =</span> <span class="st">"KEGG_cat01"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(CATEGORY, REGULATION, CATEGORY_TYPE, <span class="at">TYPE_NAME =</span> Category01, <span class="at">SUM =</span> Category01_SUM) <span class="sc">%&gt;%</span></span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">#</span></span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>    shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KeggOrthology_A =</span> <span class="fu">str_replace_na</span>(KeggOrthology_A, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(CATEGORY, REGULATION, KeggOrthology_A) <span class="sc">%&gt;%</span> </span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">KeggOrthology_A_SUM =</span> <span class="fu">sum</span>(COUNT)) <span class="sc">%&gt;%</span> </span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY_TYPE =</span> <span class="st">"KEGG_ORTHO_A"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(CATEGORY, REGULATION, CATEGORY_TYPE, <span class="at">TYPE_NAME =</span> KeggOrthology_A, <span class="at">SUM =</span> KeggOrthology_A_SUM)</span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb104-18"><a href="#cb104-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CATEGORY_TYPE_ORDER =</span> <span class="fu">factor</span>(TYPE_NAME, <span class="at">levels =</span> paths_ORDER)) <span class="sc">%&gt;%</span> </span>
<span id="cb104-19"><a href="#cb104-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> CATEGORY_TYPE, <span class="at">fill =</span> CATEGORY_TYPE_ORDER, <span class="at">y =</span> SUM)) <span class="sc">+</span></span>
<span id="cb104-20"><a href="#cb104-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">stroke =</span> <span class="fl">0.4</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(CATEGORY, REGULATION)) <span class="sc">+</span></span>
<span id="cb104-21"><a href="#cb104-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb104-22"><a href="#cb104-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb104-23"><a href="#cb104-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">h =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb104-24"><a href="#cb104-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb104-25"><a href="#cb104-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb104-26"><a href="#cb104-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> paths_COLOR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'CATEGORY', 'REGULATION'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'CATEGORY', 'REGULATION'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_bar(stat = "identity", position = "fill", color = "black", :
Ignoring unknown parameters: `stroke`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<p><strong>Take home messages</strong>: Most of the transcripts are upregulated in California. More dynamic environment compared to the NPSG surface. For transcripts that were shared across all taxa, they were on the overall low end. Among KEGG categories, a lot were assocaited with genetic information processing. For those that were curated for this study, many were related to phagotrophy. Emphasizing just how different the strategies are among all of these taxa to succeed in the euphotic zone.</p>
<p><em>NEXT STEPS:</em> &gt; Questions to follow up on: 1. For the dinoflagellates - what are some of the top phagotrophy genes only in CA? and the energy acquisition genes? 2. For haptophyte and diatom only, why the gluconeogenesis-glycolysis genes? 3. to round out the above, what about the chlorophytes? 4. For the shared - only upregulated in CA, and primarily phagotrophy - what are these? AND under the “Genetic Information Processing” KEGG category.</p>
<p>Note, that for these figures above, we are looking at the TOTAL NUMBER of significantly upregulated transcripts depending on taxonomic group and sample. We are not including the log FC.</p>
</section>
</section>
</section>
<section id="targeted" class="level2">
<h2 class="anchored" data-anchor-id="targeted">Targeted</h2>
<p>First we will focus on transcripts among phytoplankton that were annotated under the N and P metabolism category.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(counts_curated_annot<span class="sc">$</span>Category01)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Gluconeogenesis-glycolysis" "Energy Acquisition"        
[3] "Phagotrophy"                "Nutrient processing"       
[5] "Glyoxylate cycle"           "N and P metabolism"        
[7] "Carbon fixation"            "Photosynthesis"            
[9] "Entner-Doudoroff pathway"  </code></pre>
</div>
</div>
<p>Planning to subset genes from the N and P metabolism category. Modify input table for downstream visualization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="co"># min(as.numeric(counts_curated_annot))</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>n_p_TPM_0 <span class="ot">&lt;-</span> counts_curated_annot <span class="sc">%&gt;%</span> </span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"N and P metabolism"</span> <span class="sc">&amp;</span> Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean."</span>), Supergroup<span class="sc">:</span>Species, Category02, GeneID, Gene_identification, PFAMs) <span class="sc">%&gt;%</span> </span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"mean."</span>), <span class="at">names_to =</span> <span class="st">"SAMPLE_tmp"</span>, <span class="at">values_to =</span> <span class="st">"TPM"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE_tmp, <span class="st">"mean."</span>))</span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a><span class="co"># head(n_p_TPM_0)</span></span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a>min_TPM <span class="ot">&lt;-</span> <span class="fu">min</span>((n_p_TPM_0 <span class="sc">%&gt;%</span> <span class="fu">filter</span>(TPM<span class="sc">&gt;</span><span class="dv">0</span>))<span class="sc">$</span>TPM)</span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a><span class="co"># log(min_TPM)</span></span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>n_p_TPM_1 <span class="ot">&lt;-</span> n_p_TPM_0 <span class="sc">%&gt;%</span></span>
<span id="cb109-13"><a href="#cb109-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TAXA =</span> <span class="fu">case_when</span>(</span>
<span id="cb109-14"><a href="#cb109-14" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Dinoflagellata"</span> <span class="sc">~</span> <span class="st">"Dinoflagellate"</span>,</span>
<span id="cb109-15"><a href="#cb109-15" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Ciliophora"</span> <span class="sc">~</span> <span class="st">"Ciliate"</span>,</span>
<span id="cb109-16"><a href="#cb109-16" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Haptophyta"</span> <span class="sc">~</span> <span class="st">"Haptophyte"</span>,</span>
<span id="cb109-17"><a href="#cb109-17" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Chlorophyta"</span> <span class="sc">~</span> <span class="st">"Chlorophyte"</span>,</span>
<span id="cb109-18"><a href="#cb109-18" aria-hidden="true" tabindex="-1"></a>    Class <span class="sc">==</span> <span class="st">"Bacillariophyta"</span> <span class="sc">~</span> <span class="st">"Diatom"</span>,</span>
<span id="cb109-19"><a href="#cb109-19" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Rhizaria"</span> <span class="sc">~</span> <span class="st">"Rhizaria"</span>,</span>
<span id="cb109-20"><a href="#cb109-20" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Else"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb109-21"><a href="#cb109-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(TAXA, SAMPLE, Category02, GeneID, Gene_identification, PFAMs, TPM) <span class="sc">%&gt;%</span> </span>
<span id="cb109-22"><a href="#cb109-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">!=</span> <span class="st">"Else"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb109-23"><a href="#cb109-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TPM_mod =</span> <span class="fu">case_when</span>(</span>
<span id="cb109-24"><a href="#cb109-24" aria-hidden="true" tabindex="-1"></a>    TPM <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> min_TPM,</span>
<span id="cb109-25"><a href="#cb109-25" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> TPM))</span>
<span id="cb109-26"><a href="#cb109-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-27"><a href="#cb109-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Changed 0 for TPM to be the minimum</span></span>
<span id="cb109-28"><a href="#cb109-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-29"><a href="#cb109-29" aria-hidden="true" tabindex="-1"></a><span class="co"># length(unique(n_p_TPM_1$GeneID)) # 24 total genes</span></span>
<span id="cb109-30"><a href="#cb109-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-31"><a href="#cb109-31" aria-hidden="true" tabindex="-1"></a>n_p_TPM_meanSAMPLE <span class="ot">&lt;-</span> n_p_TPM_1 <span class="sc">%&gt;%</span> </span>
<span id="cb109-32"><a href="#cb109-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAXA, Category02, GeneID, Gene_identification, PFAMs) <span class="sc">%&gt;%</span> </span>
<span id="cb109-33"><a href="#cb109-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MEAN_TPM =</span> <span class="fu">mean</span>(TPM_mod)) <span class="sc">%&gt;%</span> </span>
<span id="cb109-34"><a href="#cb109-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">SAMPLE =</span> <span class="st">"Mean"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAXA', 'Category02', 'GeneID',
'Gene_identification'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_meanGeneID <span class="ot">&lt;-</span> n_p_TPM_1 <span class="sc">%&gt;%</span> </span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAXA, Category02, SAMPLE, Gene_identification, PFAMs) <span class="sc">%&gt;%</span> </span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MEAN_TPM =</span> <span class="fu">mean</span>(TPM_mod)) <span class="sc">%&gt;%</span> </span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">GeneID =</span> <span class="st">"Mean"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAXA', 'Category02', 'SAMPLE',
'Gene_identification'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="ot">&lt;-</span> n_p_TPM_1 <span class="sc">%&gt;%</span> </span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(TAXA, Category02, SAMPLE, GeneID, Gene_identification, PFAMs) <span class="sc">%&gt;%</span> </span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">MEAN_TPM =</span> <span class="fu">mean</span>(TPM_mod)) <span class="sc">%&gt;%</span> </span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(n_p_TPM_meanSAMPLE) <span class="sc">%&gt;%</span> </span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(n_p_TPM_meanGeneID) <span class="sc">%&gt;%</span> </span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">FACET_SAMPLE =</span> <span class="fu">case_when</span>(</span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>    SAMPLE <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">~</span> <span class="st">"AVG"</span>,</span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"SAMPLE"</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">FACET_GENE =</span> <span class="fu">case_when</span>(</span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>    GeneID <span class="sc">==</span> <span class="st">"Mean"</span> <span class="sc">~</span> <span class="st">"AVG"</span>,</span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"GENE"</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a>  )) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'TAXA', 'Category02', 'SAMPLE', 'GeneID',
'Gene_identification'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(n_p_TPM_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 9
# Groups:   TAXA, Category02, SAMPLE, GeneID, Gene_identification [4]
  TAXA  Category02 SAMPLE GeneID Gene_identification PFAMs MEAN_TPM FACET_SAMPLE
  &lt;chr&gt; &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;               &lt;chr&gt;    &lt;dbl&gt; &lt;chr&gt;       
1 Chlo… GS/GOGAT   Catal… GLT(f… E1.4.7.1; glutamat… GATa…  4.64    SAMPLE      
2 Chlo… GS/GOGAT   Catal… GLT1   GLT1; glutamate sy… Fer4…  1.97    SAMPLE      
3 Chlo… GS/GOGAT   Catal… GLT1   GLT1; glutamate sy… Fer4…  1.68    SAMPLE      
4 Chlo… GS/GOGAT   Catal… GLT1   GLT1; glutamate sy… GATa…  0.00543 SAMPLE      
5 Chlo… GS/GOGAT   Catal… GLTD   gltD; glutamate sy… Fer4…  0.582   SAMPLE      
6 Chlo… GS/GOGAT   Catal… gdh2   GDH2; glutamate de… Bac_…  0.567   SAMPLE      
# ℹ 1 more variable: FACET_GENE &lt;chr&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(n_p_TPM_df<span class="sc">$</span>TAXA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Chlorophyte"    "Ciliate"        "Diatom"         "Dinoflagellate"
[5] "Haptophyte"     "Rhizaria"      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(n_p_TPM_df<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Catalina"     "July_1000m"   "July_150m"    "July_5m"      "July_DCM"    
 [6] "March_1000m"  "March_150m"   "March_5m"     "March_DCM"    "PortofLA"    
[11] "SPOT_150m"    "SPOT_890m"    "SPOT_surface" "Mean"        </code></pre>
</div>
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a>sample_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Catalina"</span>, <span class="st">"PortofLA"</span>, <span class="st">"SPOT_surface"</span>, <span class="st">"SPOT_150m"</span>, <span class="st">"SPOT_890m"</span>,</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"July_5m"</span>, <span class="st">"March_5m"</span>, <span class="st">"July_DCM"</span>,<span class="st">"March_DCM"</span>, <span class="st">"July_150m"</span>, <span class="st">"March_150m"</span>, <span class="st">"July_1000m"</span>, <span class="st">"March_1000m"</span>, <span class="st">"Mean"</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>sample_label <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Catalina"</span>, <span class="st">"Port of LA"</span>, <span class="st">"SPOT surface"</span>, <span class="st">"SPOT 150m"</span>, <span class="st">"SPOT 890m"</span>,</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ALOHA surface (July)"</span>, <span class="st">"ALOHA surface (March)"</span>, <span class="st">"ALOHA DCM (July)"</span>,<span class="st">"ALOHA DCM (March)"</span>, <span class="st">"ALOHA 150m (July)"</span>, <span class="st">"ALOHA 150m (March)"</span>, <span class="st">"ALOHA 1000m (July)"</span>, <span class="st">"ALOHA 1000m (March)"</span>, <span class="st">"Mean"</span>)</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>yl_red <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#ffffcc"</span>, <span class="st">"#fed976"</span>, <span class="st">"#feb24c"</span>, <span class="st">"#fc4e2a"</span>, <span class="st">"#e31a1c"</span>, <span class="st">"#b10026"</span>)</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>reds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#f7f7f7"</span>,<span class="st">"#fee5d9"</span>, <span class="st">"#fcae91"</span>, <span class="st">"#fb6a4a"</span>, <span class="st">"#a50f15"</span>)</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df<span class="sc">$</span>SAMPLE_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(n_p_TPM_df<span class="sc">$</span>SAMPLE, <span class="at">levels =</span> <span class="fu">rev</span>(sample_order), <span class="at">labels =</span> <span class="fu">rev</span>(sample_label))</span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df<span class="sc">$</span>GeneID_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(n_p_TPM_df<span class="sc">$</span>GeneID, <span class="at">levels =</span> geneid_order_nuts)</span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>phyto <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Diatom"</span>, <span class="st">"Chlorophyte"</span>, <span class="st">"Haptophyte"</span>, <span class="st">"Dinoflagellate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="sc">%&gt;%</span></span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">%in%</span> phyto) <span class="sc">%&gt;%</span> </span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ORDER, <span class="at">y =</span> GeneID_ORDER, <span class="at">fill =</span> <span class="fu">log</span>(MEAN_TPM))) <span class="sc">+</span></span>
<span id="cb122-4"><a href="#cb122-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb122-5"><a href="#cb122-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb122-6"><a href="#cb122-6" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb122-7"><a href="#cb122-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb122-8"><a href="#cb122-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb122-9"><a href="#cb122-9" aria-hidden="true" tabindex="-1"></a>      <span class="co"># colors = reds) +</span></span>
<span id="cb122-10"><a href="#cb122-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> yl_red) <span class="sc">+</span></span>
<span id="cb122-11"><a href="#cb122-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb122-12"><a href="#cb122-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb122-13"><a href="#cb122-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(FACET_GENE), <span class="at">rows =</span> <span class="fu">vars</span>(TAXA, <span class="fu">rev</span>(FACET_SAMPLE)), </span>
<span id="cb122-14"><a href="#cb122-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">switch =</span> <span class="st">"y"</span>) <span class="sc">+</span></span>
<span id="cb122-15"><a href="#cb122-15" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_y_discrete(position = "right") +</span></span>
<span id="cb122-16"><a href="#cb122-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb122-17"><a href="#cb122-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb122-18"><a href="#cb122-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb122-19"><a href="#cb122-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb122-20"><a href="#cb122-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb122-21"><a href="#cb122-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb122-22"><a href="#cb122-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb122-23"><a href="#cb122-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb122-24"><a href="#cb122-24" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb122-25"><a href="#cb122-25" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb122-26"><a href="#cb122-26" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid" width="768"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="sc">%&gt;%</span> </span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">==</span> <span class="st">"Chlorophyte"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ORDER, <span class="at">y =</span> GeneID, <span class="at">fill =</span> <span class="fu">log</span>(MEAN_TPM))) <span class="sc">+</span></span>
<span id="cb123-4"><a href="#cb123-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb123-5"><a href="#cb123-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb123-6"><a href="#cb123-6" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb123-7"><a href="#cb123-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb123-8"><a href="#cb123-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb123-9"><a href="#cb123-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> reds) <span class="sc">+</span></span>
<span id="cb123-10"><a href="#cb123-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb123-11"><a href="#cb123-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb123-12"><a href="#cb123-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(FACET_GENE), <span class="at">rows =</span> <span class="fu">vars</span>(<span class="fu">rev</span>(FACET_SAMPLE)), <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb123-13"><a href="#cb123-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_y_discrete(position = "right") +</span></span>
<span id="cb123-14"><a href="#cb123-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb123-15"><a href="#cb123-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb123-16"><a href="#cb123-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb123-17"><a href="#cb123-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb123-18"><a href="#cb123-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb123-19"><a href="#cb123-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb123-20"><a href="#cb123-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb123-21"><a href="#cb123-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb123-22"><a href="#cb123-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb123-23"><a href="#cb123-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="sc">%&gt;%</span> </span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">==</span> <span class="st">"Haptophyte"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ORDER, <span class="at">y =</span> GeneID, <span class="at">fill =</span> <span class="fu">log</span>(MEAN_TPM))) <span class="sc">+</span></span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> reds) <span class="sc">+</span></span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(FACET_GENE), <span class="at">rows =</span> <span class="fu">vars</span>(<span class="fu">rev</span>(FACET_SAMPLE)), <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb124-13"><a href="#cb124-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_y_discrete(position = "right") +</span></span>
<span id="cb124-14"><a href="#cb124-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb124-15"><a href="#cb124-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb124-16"><a href="#cb124-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb124-17"><a href="#cb124-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb124-18"><a href="#cb124-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb124-19"><a href="#cb124-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb124-20"><a href="#cb124-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb124-21"><a href="#cb124-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb124-22"><a href="#cb124-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb124-23"><a href="#cb124-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-53-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="sc">%&gt;%</span> </span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">==</span> <span class="st">"Dinoflagellate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ORDER, <span class="at">y =</span> GeneID, <span class="at">fill =</span> <span class="fu">log</span>(MEAN_TPM))) <span class="sc">+</span></span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> reds) <span class="sc">+</span></span>
<span id="cb125-10"><a href="#cb125-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb125-11"><a href="#cb125-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb125-12"><a href="#cb125-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(FACET_GENE), <span class="at">rows =</span> <span class="fu">vars</span>(<span class="fu">rev</span>(FACET_SAMPLE)), <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb125-13"><a href="#cb125-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_y_discrete(position = "right") +</span></span>
<span id="cb125-14"><a href="#cb125-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb125-15"><a href="#cb125-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb125-16"><a href="#cb125-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb125-17"><a href="#cb125-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb125-18"><a href="#cb125-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-19"><a href="#cb125-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-20"><a href="#cb125-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-21"><a href="#cb125-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-22"><a href="#cb125-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb125-23"><a href="#cb125-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb126"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb126-1"><a href="#cb126-1" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df <span class="sc">%&gt;%</span> </span>
<span id="cb126-2"><a href="#cb126-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">==</span> <span class="st">"Diatom"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb126-3"><a href="#cb126-3" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ORDER, <span class="at">y =</span> GeneID, <span class="at">fill =</span> <span class="fu">log</span>(MEAN_TPM))) <span class="sc">+</span></span>
<span id="cb126-4"><a href="#cb126-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb126-5"><a href="#cb126-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_stepsn</span>(</span>
<span id="cb126-6"><a href="#cb126-6" aria-hidden="true" tabindex="-1"></a>      <span class="co"># limits = c(-5,5),</span></span>
<span id="cb126-7"><a href="#cb126-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># breaks = c(0.0, 3.5, 7.0, 10.5),</span></span>
<span id="cb126-8"><a href="#cb126-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">show.limits =</span> <span class="cn">TRUE</span>,</span>
<span id="cb126-9"><a href="#cb126-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">colors =</span> reds) <span class="sc">+</span></span>
<span id="cb126-10"><a href="#cb126-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb126-11"><a href="#cb126-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb126-12"><a href="#cb126-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(FACET_GENE), <span class="at">rows =</span> <span class="fu">vars</span>(<span class="fu">rev</span>(FACET_SAMPLE)), <span class="at">space =</span> <span class="st">"free"</span>, <span class="at">scales =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb126-13"><a href="#cb126-13" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_y_discrete(position = "right") +</span></span>
<span id="cb126-14"><a href="#cb126-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">position =</span> <span class="st">"top"</span>) <span class="sc">+</span></span>
<span id="cb126-15"><a href="#cb126-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb126-16"><a href="#cb126-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb126-17"><a href="#cb126-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">1</span>),</span>
<span id="cb126-18"><a href="#cb126-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-19"><a href="#cb126-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.line =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-20"><a href="#cb126-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.border =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-21"><a href="#cb126-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-22"><a href="#cb126-22" aria-hidden="true" tabindex="-1"></a>          <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb126-23"><a href="#cb126-23" aria-hidden="true" tabindex="-1"></a>          <span class="at">strip.text =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-55-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<blockquote class="blockquote">
<p>What kinds of things can we say about specific genes?</p>
</blockquote>
</section>
</section>
<section id="question-2" class="level1">
<h1>Question 2</h1>
<blockquote class="blockquote">
<p>Do we see a similar euphotic vs.&nbsp;sub-euphotic shift in the metabolic potential of taxa observed at coastal California and the NPSG?</p>
</blockquote>
<section id="untargeted-1" class="level2">
<h2 class="anchored" data-anchor-id="untargeted-1">Untargeted</h2>
<p>To approach question 2 with an untargeted approach, used <code>run_bydepth_deseq_taxa.R</code> script to generate R objects of each comparison.</p>
<p>To compare euphotic versus subeuphotic, decided to with all samples, CA only samples and then only NPSG samples. See code: <code>ca-light.R</code>, <code>all-light.R</code>, and <code>npsg-light.R</code>.</p>
<p>Run this script, which selects the differentially expressed genes and includes the functional annotation information: <code>euphotic_v_subeuphotic.R</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Import KEGG information</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a>taxfxn <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"TaxonomicAndFunctionalAnnotations.csv"</span>)</span>
<span id="cb127-3"><a href="#cb127-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(taxfxn)</span>
<span id="cb127-4"><a href="#cb127-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-5"><a href="#cb127-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Sequence IDs are all unique, because they correspond to taxa. Therefore, we need to include gene-level IDs so we can compare what is DE across the euphotic vs. subeuphotic regions.</span></span>
<span id="cb127-6"><a href="#cb127-6" aria-hidden="true" tabindex="-1"></a>CA_only <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"bytaxa_CA-LIGHT_DE.csv"</span>)</span>
<span id="cb127-7"><a href="#cb127-7" aria-hidden="true" tabindex="-1"></a>NPSG_only <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"bytaxa_NPSG-LIGHT_DE.csv"</span>)</span>
<span id="cb127-8"><a href="#cb127-8" aria-hidden="true" tabindex="-1"></a>all_only <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"bytaxa_ALL-LIGHT_DE.csv"</span>)</span>
<span id="cb127-9"><a href="#cb127-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-10"><a href="#cb127-10" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb127-11"><a href="#cb127-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(CA_only)</span>
<span id="cb127-12"><a href="#cb127-12" aria-hidden="true" tabindex="-1"></a><span class="co"># For this there are three tables. </span></span>
<span id="cb127-13"><a href="#cb127-13" aria-hidden="true" tabindex="-1"></a>CA_tax_bydepth <span class="ot">&lt;-</span> CA_only <span class="sc">%&gt;%</span> </span>
<span id="cb127-14"><a href="#cb127-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SequenceID, log2FoldChange, pvalue, REGULATION, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-15"><a href="#cb127-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span>
<span id="cb127-16"><a href="#cb127-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-17"><a href="#cb127-17" aria-hidden="true" tabindex="-1"></a><span class="co"># head(CA_tax_bydepth)</span></span>
<span id="cb127-18"><a href="#cb127-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-19"><a href="#cb127-19" aria-hidden="true" tabindex="-1"></a>NPSG_tax_bydepth <span class="ot">&lt;-</span> NPSG_only <span class="sc">%&gt;%</span> </span>
<span id="cb127-20"><a href="#cb127-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SequenceID, log2FoldChange, pvalue, REGULATION, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-21"><a href="#cb127-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span>
<span id="cb127-22"><a href="#cb127-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-23"><a href="#cb127-23" aria-hidden="true" tabindex="-1"></a>ALL_tax_bydepth <span class="ot">&lt;-</span> all_only <span class="sc">%&gt;%</span> </span>
<span id="cb127-24"><a href="#cb127-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SequenceID, log2FoldChange, pvalue, REGULATION, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-25"><a href="#cb127-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span>
<span id="cb127-26"><a href="#cb127-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-27"><a href="#cb127-27" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb127-28"><a href="#cb127-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-29"><a href="#cb127-29" aria-hidden="true" tabindex="-1"></a>alltax_compareDE <span class="ot">&lt;-</span> ALL_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb127-30"><a href="#cb127-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-31"><a href="#cb127-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb127-32"><a href="#cb127-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-33"><a href="#cb127-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb127-34"><a href="#cb127-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb127-35"><a href="#cb127-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TAXA, <span class="at">values_from =</span> n)</span>
<span id="cb127-36"><a href="#cb127-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-37"><a href="#cb127-37" aria-hidden="true" tabindex="-1"></a>NPSGtax_compareDE <span class="ot">&lt;-</span> NPSG_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb127-38"><a href="#cb127-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-39"><a href="#cb127-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb127-40"><a href="#cb127-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-41"><a href="#cb127-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb127-42"><a href="#cb127-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb127-43"><a href="#cb127-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TAXA, <span class="at">values_from =</span> n)</span>
<span id="cb127-44"><a href="#cb127-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-45"><a href="#cb127-45" aria-hidden="true" tabindex="-1"></a>CAtax_compareDE <span class="ot">&lt;-</span> CA_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb127-46"><a href="#cb127-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb127-47"><a href="#cb127-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb127-48"><a href="#cb127-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb127-49"><a href="#cb127-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb127-50"><a href="#cb127-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb127-51"><a href="#cb127-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> TAXA, <span class="at">values_from =</span> n)</span>
<span id="cb127-52"><a href="#cb127-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb127-53"><a href="#cb127-53" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(alltax_compareDE, CAtax_compareDE, NPSGtax_compareDE, ALL_tax_bydepth, NPSG_tax_bydepth, CA_tax_bydepth, <span class="at">file =</span> <span class="st">"taxa-untarget_bydepth-q2.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Locally, now can work with results.</p>
<section id="local--untargeted" class="level4">
<h4 class="anchored" data-anchor-id="local--untargeted">Local -untargeted</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/taxa-untarget_bydepth-q2.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  alltax_compareDE
  CAtax_compareDE
  NPSGtax_compareDE
  ALL_tax_bydepth
  NPSG_tax_bydepth
  CA_tax_bydepth</code></pre>
</div>
</div>
</section>
<section id="shared-vs-unique" class="level3">
<h3 class="anchored" data-anchor-id="shared-vs-unique">Shared vs unique</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggupset)</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a>ALL_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENEID_ko_pfam, REGULATION) <span class="sc">%&gt;%</span> </span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">TAXA =</span> <span class="fu">list</span>(TAXA)) <span class="sc">%&gt;%</span> </span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> TAXA)) <span class="sc">+</span></span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> REGULATION)) <span class="sc">+</span></span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#74add1"</span>, <span class="st">"#fdae61"</span>)) <span class="sc">+</span></span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_upset</span>(<span class="at">order_by =</span> <span class="st">"freq"</span>) <span class="sc">+</span></span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>,</span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"All euphotic vs subeuphotic comparison"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>NPSG_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENEID_ko_pfam, REGULATION) <span class="sc">%&gt;%</span> </span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">TAXA =</span> <span class="fu">list</span>(TAXA)) <span class="sc">%&gt;%</span> </span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> TAXA)) <span class="sc">+</span></span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> REGULATION)) <span class="sc">+</span></span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#74add1"</span>, <span class="st">"#fdae61"</span>)) <span class="sc">+</span></span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_upset</span>(<span class="at">order_by =</span> <span class="st">"freq"</span>) <span class="sc">+</span></span>
<span id="cb131-13"><a href="#cb131-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb131-14"><a href="#cb131-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb131-15"><a href="#cb131-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>,</span>
<span id="cb131-16"><a href="#cb131-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"NPSG comparison"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-58-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>CA_tax_bydepth <span class="sc">%&gt;%</span> </span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(GENEID_ko_pfam, PFAMs, KEGG_ko, <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(REGULATION, TAXA, log2FoldChange, GENEID_ko_pfam) <span class="sc">%&gt;%</span> </span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(REGULATION, GENEID_ko_pfam, TAXA) <span class="sc">%&gt;%</span></span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n =</span> dplyr<span class="sc">::</span><span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n <span class="sc">&gt;</span> 1L) <span class="sc">%&gt;%</span> </span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(GENEID_ko_pfam, REGULATION) <span class="sc">%&gt;%</span> </span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">TAXA =</span> <span class="fu">list</span>(TAXA)) <span class="sc">%&gt;%</span> </span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> TAXA)) <span class="sc">+</span></span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">fill =</span> REGULATION)) <span class="sc">+</span></span>
<span id="cb132-11"><a href="#cb132-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#74add1"</span>, <span class="st">"#fdae61"</span>)) <span class="sc">+</span></span>
<span id="cb132-12"><a href="#cb132-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_upset</span>(<span class="at">order_by =</span> <span class="st">"freq"</span>) <span class="sc">+</span></span>
<span id="cb132-13"><a href="#cb132-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb132-14"><a href="#cb132-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb132-15"><a href="#cb132-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>,</span>
<span id="cb132-16"><a href="#cb132-16" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Coastal California comparison"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-58-3.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="what-transcripts-are-shared-across-all-taxa-vs.-unique-1" class="level4">
<h4 class="anchored" data-anchor-id="what-transcripts-are-shared-across-all-taxa-vs.-unique-1">What transcripts are shared across all taxa vs.&nbsp;unique?</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>tax_indiv <span class="ot">&lt;-</span> alltax_compareDE <span class="sc">%&gt;%</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CATEGORY =</span> <span class="fu">case_when</span>(</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>                     (<span class="fu">is.na</span>(Ciliate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Rhizaria) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Dinoflagellate only"</span>,</span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Rhizaria) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Ciliate only"</span>,</span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Ciliate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Rhizaria)) <span class="sc">~</span> <span class="st">"Haptophyte only"</span>,</span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>                   (<span class="fu">is.na</span>(Dinoflagellate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Ciliate) <span class="sc">&amp;</span> <span class="fu">is.na</span>(Haptophyte)) <span class="sc">~</span> <span class="st">"Rhizaria only"</span>,</span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>                   <span class="cn">TRUE</span> <span class="sc">~</span> <span class="cn">NA</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(CATEGORY))</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb133-10"><a href="#cb133-10" aria-hidden="true" tabindex="-1"></a>shared_euphotic <span class="ot">&lt;-</span> alltax_compareDE <span class="sc">%&gt;%</span> </span>
<span id="cb133-11"><a href="#cb133-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb133-12"><a href="#cb133-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY =</span> <span class="st">"Shared across all taxa"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb133-13"><a href="#cb133-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(tax_indiv) <span class="sc">%&gt;%</span> </span>
<span id="cb133-14"><a href="#cb133-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(GENEID_ko_pfam, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">"PFAMs"</span>, <span class="st">"KEGG_ko"</span>), <span class="at">sep =</span> <span class="st">";"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb133-15"><a href="#cb133-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KEGG =</span> <span class="fu">str_remove</span>(KEGG_ko, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb133-16"><a href="#cb133-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(key_geneid) <span class="sc">%&gt;%</span> </span>
<span id="cb133-17"><a href="#cb133-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">COUNT =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(KEGG)`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., key_geneid): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 16589 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(shared_euphotic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 23
  REGULATION   PFAMs KEGG_ko Dinoflagellate Ciliate Haptophyte Rhizaria CATEGORY
  &lt;chr&gt;        &lt;chr&gt; &lt;chr&gt;            &lt;int&gt;   &lt;int&gt;      &lt;int&gt;    &lt;int&gt; &lt;chr&gt;   
1 upregulated… Actin ko:K05…            100      23        220       12 Shared …
2 upregulated… Actin ko:K05…            100      23        220       12 Shared …
3 upregulated… Actin ko:K05…            100      23        220       12 Shared …
4 upregulated… Actin ko:K05…            100      23        220       12 Shared …
5 upregulated… Actin ko:K05…            100      23        220       12 Shared …
6 upregulated… Actin ko:K05…            100      23        220       12 Shared …
# ℹ 15 more variables: KEGG &lt;chr&gt;, KeggOrthology_A &lt;chr&gt;,
#   KeggOrthology_B &lt;chr&gt;, KeggOrthology_C &lt;chr&gt;,
#   KeggOrthology_Gene_description &lt;chr&gt;, KeggOrthology_EC &lt;chr&gt;,
#   Category01 &lt;chr&gt;, Category02 &lt;chr&gt;, FullName &lt;chr&gt;, GeneID &lt;chr&gt;,
#   Gene_identification &lt;chr&gt;, PFAM &lt;chr&gt;, Descriptions &lt;chr&gt;, REF &lt;chr&gt;,
#   COUNT &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(alltax_compareDE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(shared_euphotic<span class="sc">$</span>KeggOrthology_A)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Environmental Information Processing"
[2] "Cellular Processes"                  
[3] NA                                    
[4] "Genetic Information Processing"      
[5] "Metabolism"                          </code></pre>
</div>
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(shared_euphotic<span class="sc">$</span>Category01)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Phagotrophy"                NA                          
 [3] "Energy Acquisition"         "N and P metabolism"        
 [5] "Glyoxylate cycle"           "Nutrient processing"       
 [7] ""                           "Carbon fixation"           
 [9] "Gluconeogenesis-glycolysis" "Photosynthesis"            </code></pre>
</div>
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>paths_ORDER <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Photosynthesis"</span>, <span class="st">"Energy Acquisition"</span>, <span class="st">"Carbon fixation"</span>, <span class="st">"Nutrient processing"</span>, <span class="st">"Gluconeogenesis-glycolysis"</span>, <span class="st">"N and P metabolism"</span>, <span class="st">"Glyoxylate cycle"</span>, <span class="st">"Phagotrophy"</span>, </span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Environmental Information Processing"</span>, <span class="st">"Cellular Processes"</span>,</span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Genetic Information Processing"</span>, <span class="st">"Metabolism"</span>,</span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"Other"</span>)</span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a>paths_COLOR <span class="ot">&lt;-</span><span class="fu">c</span> (<span class="st">"#f7fcb9"</span>, <span class="st">"#fdae6b"</span>, <span class="st">"#fc9272"</span>, <span class="st">"#7fcdbb"</span>, <span class="st">"#d9f0a3"</span>, <span class="st">"#78c679"</span>, <span class="st">"#bfd3e6"</span>, <span class="st">"#fde0dd"</span>,</span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#fd8d3c"</span>, <span class="st">"#225ea8"</span>, </span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#cb181d"</span>, <span class="st">"#41ab5d"</span>,</span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>                 <span class="st">"#969696"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb144"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb144-1"><a href="#cb144-1" aria-hidden="true" tabindex="-1"></a>shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb144-2"><a href="#cb144-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Category01 =</span> <span class="fu">str_replace_na</span>(Category01, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb144-3"><a href="#cb144-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(CATEGORY, REGULATION, Category01) <span class="sc">%&gt;%</span> </span>
<span id="cb144-4"><a href="#cb144-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">Category01_SUM =</span> <span class="fu">sum</span>(COUNT)) <span class="sc">%&gt;%</span> </span>
<span id="cb144-5"><a href="#cb144-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY_TYPE =</span> <span class="st">"KEGG_cat01"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb144-6"><a href="#cb144-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(CATEGORY, REGULATION, CATEGORY_TYPE, <span class="at">TYPE_NAME =</span> Category01, <span class="at">SUM =</span> Category01_SUM) <span class="sc">%&gt;%</span></span>
<span id="cb144-7"><a href="#cb144-7" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb144-8"><a href="#cb144-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(</span>
<span id="cb144-9"><a href="#cb144-9" aria-hidden="true" tabindex="-1"></a>    <span class="co">#</span></span>
<span id="cb144-10"><a href="#cb144-10" aria-hidden="true" tabindex="-1"></a>    shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb144-11"><a href="#cb144-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">KeggOrthology_A =</span> <span class="fu">str_replace_na</span>(KeggOrthology_A, <span class="st">"Other"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb144-12"><a href="#cb144-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(CATEGORY, REGULATION, KeggOrthology_A) <span class="sc">%&gt;%</span> </span>
<span id="cb144-13"><a href="#cb144-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">KeggOrthology_A_SUM =</span> <span class="fu">sum</span>(COUNT)) <span class="sc">%&gt;%</span> </span>
<span id="cb144-14"><a href="#cb144-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_column</span>(<span class="at">CATEGORY_TYPE =</span> <span class="st">"KEGG_ORTHO_A"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb144-15"><a href="#cb144-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(CATEGORY, REGULATION, CATEGORY_TYPE, <span class="at">TYPE_NAME =</span> KeggOrthology_A, <span class="at">SUM =</span> KeggOrthology_A_SUM)</span>
<span id="cb144-16"><a href="#cb144-16" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb144-17"><a href="#cb144-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb144-18"><a href="#cb144-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">CATEGORY_TYPE_ORDER =</span> <span class="fu">factor</span>(TYPE_NAME, <span class="at">levels =</span> paths_ORDER)) <span class="sc">%&gt;%</span> </span>
<span id="cb144-19"><a href="#cb144-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> CATEGORY_TYPE, <span class="at">fill =</span> CATEGORY_TYPE_ORDER, <span class="at">y =</span> SUM)) <span class="sc">+</span></span>
<span id="cb144-20"><a href="#cb144-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">stroke =</span> <span class="fl">0.4</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(CATEGORY, REGULATION)) <span class="sc">+</span></span>
<span id="cb144-21"><a href="#cb144-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb144-22"><a href="#cb144-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb144-23"><a href="#cb144-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">h =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>),</span>
<span id="cb144-24"><a href="#cb144-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>),</span>
<span id="cb144-25"><a href="#cb144-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb144-26"><a href="#cb144-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> paths_COLOR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'CATEGORY', 'REGULATION'. You can override
using the `.groups` argument.
`summarise()` has grouped output by 'CATEGORY', 'REGULATION'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_bar(stat = "identity", position = "fill", color = "black", :
Ignoring unknown parameters: `stroke`</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-61-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(shared_euphotic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 23
  REGULATION   PFAMs KEGG_ko Dinoflagellate Ciliate Haptophyte Rhizaria CATEGORY
  &lt;chr&gt;        &lt;chr&gt; &lt;chr&gt;            &lt;int&gt;   &lt;int&gt;      &lt;int&gt;    &lt;int&gt; &lt;chr&gt;   
1 upregulated… Actin ko:K05…            100      23        220       12 Shared …
2 upregulated… Actin ko:K05…            100      23        220       12 Shared …
3 upregulated… Actin ko:K05…            100      23        220       12 Shared …
4 upregulated… Actin ko:K05…            100      23        220       12 Shared …
5 upregulated… Actin ko:K05…            100      23        220       12 Shared …
6 upregulated… Actin ko:K05…            100      23        220       12 Shared …
# ℹ 15 more variables: KEGG &lt;chr&gt;, KeggOrthology_A &lt;chr&gt;,
#   KeggOrthology_B &lt;chr&gt;, KeggOrthology_C &lt;chr&gt;,
#   KeggOrthology_Gene_description &lt;chr&gt;, KeggOrthology_EC &lt;chr&gt;,
#   Category01 &lt;chr&gt;, Category02 &lt;chr&gt;, FullName &lt;chr&gt;, GeneID &lt;chr&gt;,
#   Gene_identification &lt;chr&gt;, PFAM &lt;chr&gt;, Descriptions &lt;chr&gt;, REF &lt;chr&gt;,
#   COUNT &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(shared_euphotic)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "REGULATION"                     "PFAMs"                         
 [3] "KEGG_ko"                        "Dinoflagellate"                
 [5] "Ciliate"                        "Haptophyte"                    
 [7] "Rhizaria"                       "CATEGORY"                      
 [9] "KEGG"                           "KeggOrthology_A"               
[11] "KeggOrthology_B"                "KeggOrthology_C"               
[13] "KeggOrthology_Gene_description" "KeggOrthology_EC"              
[15] "Category01"                     "Category02"                    
[17] "FullName"                       "GeneID"                        
[19] "Gene_identification"            "PFAM"                          
[21] "Descriptions"                   "REF"                           
[23] "COUNT"                         </code></pre>
</div>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"Phagotrophy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb151-3"><a href="#cb151-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Gene_identification, <span class="at">y =</span> CATEGORY)) <span class="sc">+</span></span>
<span id="cb151-4"><a href="#cb151-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> COUNT)) <span class="sc">+</span></span>
<span id="cb151-5"><a href="#cb151-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb151-6"><a href="#cb151-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb152"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb152-1"><a href="#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>((shared_euphotic <span class="sc">%&gt;%</span> </span>
<span id="cb152-2"><a href="#cb152-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"Phagotrophy"</span>)), <span class="st">"output-data/phagotrophy-shared_v_unique-gene.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="targeted-1" class="level2">
<h2 class="anchored" data-anchor-id="targeted-1">Targeted</h2>
<p>Shifting gears to focus on phagotrophy-related transcripts. There are a lot of them, and there is another category that separates different phagotrophy pathways.</p>
<blockquote class="blockquote">
<p>genes that were present in at least 2 samples for the taxonomic group? - Normalize to individual taxa.</p>
</blockquote>
<p>First, isolate a list of KEGG IDs that are found in at least 2 samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(counts_curated_annot)</span></span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> counts_curated_annot <span class="sc">%&gt;%</span> </span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"Phagotrophy"</span> <span class="sc">&amp;</span> Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean."</span>), KEGG)</span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tmp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  mean.Catalina mean.July_1000m mean.July_150m mean.July_5m mean.July_DCM
1      0.000000               0              0    27.954439             0
2      1.296684               0              0    84.190243             0
3      0.000000               0              0     9.060444             0
4      0.000000               0              0     9.060444             0
5      0.915887               0              0    32.037620             0
6      0.915887               0              0    32.037620             0
  mean.March_1000m mean.March_150m mean.March_5m mean.March_DCM mean.PortofLA
1                0               0             0              0             0
2                0               0             0              0             0
3                0               0             0              0             0
4                0               0             0              0             0
5                0               0             0              0             0
6                0               0             0              0             0
  mean.SPOT_150m mean.SPOT_890m mean.SPOT_surface   KEGG
1              0              0         0.0000000 K00645
2              0              0         2.3958786 K05692
3              0              0         0.1008968 K08057
4              0              0         0.1008968 K08057
5              0              0         0.0000000 K01365
6              0              0         0.0000000 K01365</code></pre>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tmp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 821183     14</code></pre>
</div>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a>tmp2 <span class="ot">&lt;-</span> tmp[<span class="fu">apply</span>(tmp <span class="sc">!=</span> <span class="dv">0</span>, <span class="dv">1</span>, sum) <span class="sc">&gt;=</span><span class="dv">2</span>,]</span>
<span id="cb157-2"><a href="#cb157-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tmp)[<span class="dv">1</span>] <span class="sc">-</span> <span class="fu">dim</span>(tmp2)[<span class="dv">1</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 18953</code></pre>
</div>
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="co"># (dim(tmp2)[1]) / (dim(tmp)[1])</span></span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>kegg_keep_hets <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(tmp2<span class="sc">$</span>KEGG))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Removed about 18k transcripts, keeping 98% of the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>het_TPM_scale <span class="ot">&lt;-</span> counts_curated_annot <span class="sc">%&gt;%</span></span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter by category and domain</span></span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">==</span> <span class="st">"Phagotrophy"</span> <span class="sc">&amp;</span> Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb160-4"><a href="#cb160-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter by frequency in samples</span></span>
<span id="cb160-5"><a href="#cb160-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(KEGG <span class="sc">%in%</span> kegg_keep_hets) <span class="sc">%&gt;%</span> </span>
<span id="cb160-6"><a href="#cb160-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean."</span>), Supergroup<span class="sc">:</span>Species, Category02, GeneID, KEGG, Gene_identification, PFAMs) <span class="sc">%&gt;%</span></span>
<span id="cb160-7"><a href="#cb160-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TAXA =</span> <span class="fu">case_when</span>(</span>
<span id="cb160-8"><a href="#cb160-8" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Dinoflagellata"</span> <span class="sc">~</span> <span class="st">"Dinoflagellate"</span>,</span>
<span id="cb160-9"><a href="#cb160-9" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Ciliophora"</span> <span class="sc">~</span> <span class="st">"Ciliate"</span>,</span>
<span id="cb160-10"><a href="#cb160-10" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Haptophyta"</span> <span class="sc">~</span> <span class="st">"Haptophyte"</span>,</span>
<span id="cb160-11"><a href="#cb160-11" aria-hidden="true" tabindex="-1"></a>    Phylum <span class="sc">==</span> <span class="st">"Chlorophyta"</span> <span class="sc">~</span> <span class="st">"Chlorophyte"</span>,</span>
<span id="cb160-12"><a href="#cb160-12" aria-hidden="true" tabindex="-1"></a>    Class <span class="sc">==</span> <span class="st">"Bacillariophyta"</span> <span class="sc">~</span> <span class="st">"Diatom"</span>,</span>
<span id="cb160-13"><a href="#cb160-13" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Rhizaria"</span> <span class="sc">~</span> <span class="st">"Rhizaria"</span>,</span>
<span id="cb160-14"><a href="#cb160-14" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Else"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb160-15"><a href="#cb160-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TAXA <span class="sc">!=</span> <span class="st">"Else"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb160-16"><a href="#cb160-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"mean."</span>), <span class="at">names_to =</span> <span class="st">"SAMPLE_tmp"</span>, <span class="at">values_to =</span> <span class="st">"TPM"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb160-17"><a href="#cb160-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">LOG_TPM =</span> <span class="fu">log</span>(TPM))  <span class="sc">%&gt;%</span> </span>
<span id="cb160-18"><a href="#cb160-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE_tmp, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb160-19"><a href="#cb160-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE_ORDER =</span> <span class="fu">factor</span>(SAMPLE, <span class="at">levels =</span> sample_order, <span class="at">labels =</span> sample_label))</span>
<span id="cb160-20"><a href="#cb160-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb160-21"><a href="#cb160-21" aria-hidden="true" tabindex="-1"></a><span class="co"># head(het_TPM_scale)</span></span>
<span id="cb160-22"><a href="#cb160-22" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(het_TPM_scale$SAMPLE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a>sample_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Catalina"</span>, <span class="st">"PortofLA"</span>, <span class="st">"SPOT_surface"</span>, <span class="st">"SPOT_150m"</span>, <span class="st">"SPOT_890m"</span>,</span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>  <span class="st">"July_5m"</span>, <span class="st">"March_5m"</span>, <span class="st">"July_DCM"</span>,<span class="st">"March_DCM"</span>, <span class="st">"July_150m"</span>, <span class="st">"March_150m"</span>, <span class="st">"July_1000m"</span>, <span class="st">"March_1000m"</span>, <span class="st">"Mean"</span>)</span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a>sample_label <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Catalina"</span>, <span class="st">"Port of LA"</span>, <span class="st">"SPOT surface"</span>, <span class="st">"SPOT 150m"</span>, <span class="st">"SPOT 890m"</span>,</span>
<span id="cb161-4"><a href="#cb161-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ALOHA surface (July)"</span>, <span class="st">"ALOHA surface (March)"</span>, <span class="st">"ALOHA DCM (July)"</span>,<span class="st">"ALOHA DCM (March)"</span>, <span class="st">"ALOHA 150m (July)"</span>, <span class="st">"ALOHA 150m (March)"</span>, <span class="st">"ALOHA 1000m (July)"</span>, <span class="st">"ALOHA 1000m (March)"</span>, <span class="st">"Mean"</span>)</span>
<span id="cb161-5"><a href="#cb161-5" aria-hidden="true" tabindex="-1"></a>yl_red <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#ffffcc"</span>, <span class="st">"#fed976"</span>, <span class="st">"#feb24c"</span>, <span class="st">"#fc4e2a"</span>, <span class="st">"#e31a1c"</span>, <span class="st">"#b10026"</span>)</span>
<span id="cb161-6"><a href="#cb161-6" aria-hidden="true" tabindex="-1"></a>reds <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"#f7f7f7"</span>,<span class="st">"#fee5d9"</span>, <span class="st">"#fcae91"</span>, <span class="st">"#fb6a4a"</span>, <span class="st">"#a50f15"</span>)</span>
<span id="cb161-7"><a href="#cb161-7" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df<span class="sc">$</span>SAMPLE_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(n_p_TPM_df<span class="sc">$</span>SAMPLE, <span class="at">levels =</span> <span class="fu">rev</span>(sample_order), <span class="at">labels =</span> <span class="fu">rev</span>(sample_label))</span>
<span id="cb161-8"><a href="#cb161-8" aria-hidden="true" tabindex="-1"></a>n_p_TPM_df<span class="sc">$</span>GeneID_ORDER <span class="ot">&lt;-</span> <span class="fu">factor</span>(n_p_TPM_df<span class="sc">$</span>GeneID, <span class="at">levels =</span> geneid_order_nuts)</span>
<span id="cb161-9"><a href="#cb161-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb161-10"><a href="#cb161-10" aria-hidden="true" tabindex="-1"></a>phyto <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Diatom"</span>, <span class="st">"Chlorophyte"</span>, <span class="st">"Haptophyte"</span>, <span class="st">"Dinoflagellate"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here, we’ve narrowed things down to key phagotrophy-related genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb162"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb162-1"><a href="#cb162-1" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(het_TPM_scale$LOG_TPM)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library(ggdist)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb164"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(het_TPM_scale)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 18
  Supergroup Phylum     Class Order Family Genus Species Category02 GeneID KEGG 
  &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt;
1 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
2 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
3 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
4 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
5 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
6 Alveolata  Dinoflage… Dino… Sues… Suess… Uncl… Unclas… Fatty aci… fabD   K006…
# ℹ 8 more variables: Gene_identification &lt;chr&gt;, PFAMs &lt;chr&gt;, TAXA &lt;chr&gt;,
#   SAMPLE_tmp &lt;chr&gt;, TPM &lt;dbl&gt;, LOG_TPM &lt;dbl&gt;, SAMPLE &lt;chr&gt;,
#   SAMPLE_ORDER &lt;fct&gt;</code></pre>
</div>
</div>
<p>Try a birds eye view look at how different the phagotrophy signature may be between taxa.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb166"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb166-1"><a href="#cb166-1" aria-hidden="true" tabindex="-1"></a>het_TPM_scale <span class="sc">%&gt;%</span> </span>
<span id="cb166-2"><a href="#cb166-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb166-3"><a href="#cb166-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GeneID, <span class="at">y =</span> SAMPLE_ORDER)) <span class="sc">+</span></span>
<span id="cb166-4"><a href="#cb166-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> LOG_TPM)) <span class="sc">+</span></span>
<span id="cb166-5"><a href="#cb166-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb166-6"><a href="#cb166-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(TAXA), <span class="at">rows =</span> <span class="fu">vars</span>(Category02), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb166-7"><a href="#cb166-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb166-8"><a href="#cb166-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb166-9"><a href="#cb166-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-69-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a>het_TPM_scale <span class="sc">%&gt;%</span> </span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(TPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb167-3"><a href="#cb167-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GeneID, <span class="at">y =</span> SAMPLE_ORDER)) <span class="sc">+</span></span>
<span id="cb167-4"><a href="#cb167-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> LOG_TPM), <span class="at">color =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb167-5"><a href="#cb167-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># scale_fill_binned(type = "gradient") +</span></span>
<span id="cb167-6"><a href="#cb167-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_steps</span>(</span>
<span id="cb167-7"><a href="#cb167-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">low =</span> <span class="st">"#ece2f0"</span>,</span>
<span id="cb167-8"><a href="#cb167-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">high =</span> <span class="st">"#016c59"</span>,</span>
<span id="cb167-9"><a href="#cb167-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">space =</span> <span class="st">"Lab"</span>,</span>
<span id="cb167-10"><a href="#cb167-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">na.value =</span> <span class="st">"grey50"</span>,</span>
<span id="cb167-11"><a href="#cb167-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">guide =</span> <span class="st">"coloursteps"</span>,</span>
<span id="cb167-12"><a href="#cb167-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">aesthetics =</span> <span class="st">"fill"</span></span>
<span id="cb167-13"><a href="#cb167-13" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb167-14"><a href="#cb167-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb167-15"><a href="#cb167-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(TAXA), <span class="at">rows =</span> <span class="fu">vars</span>(Category02), <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">space =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb167-16"><a href="#cb167-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb167-17"><a href="#cb167-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.placement =</span> <span class="st">"outside"</span>,</span>
<span id="cb167-18"><a href="#cb167-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">0</span>),</span>
<span id="cb167-19"><a href="#cb167-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust =</span> <span class="dv">0</span>),</span>
<span id="cb167-20"><a href="#cb167-20" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-70-1.png" class="img-fluid" width="1440"></p>
</div>
</div>
</section>
</section>
<section id="supplementary" class="level1">
<h1>Supplementary</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="co"># months &lt;- read.csv("summary-months.csv")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="session-info" class="level1">
<h1>Session info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.5.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggupset_0.3.0      patchwork_1.1.2    compositions_2.0-6 data.table_1.14.8 
 [5] lubridate_1.9.2    forcats_1.0.0      stringr_1.5.0      dplyr_1.1.2       
 [9] purrr_1.0.1        readr_2.1.4        tidyr_1.3.0        tibble_3.2.1      
[13] ggplot2_3.4.2      tidyverse_2.0.0   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10       DEoptimR_1.0-13   compiler_4.2.2    pillar_1.9.0     
 [5] tools_4.2.2       digest_0.6.31     timechange_0.2.0  jsonlite_1.8.5   
 [9] evaluate_0.21     lifecycle_1.0.3   gtable_0.3.3      pkgconfig_2.0.3  
[13] rlang_1.1.1       cli_3.6.1         rstudioapi_0.14   yaml_2.3.7       
[17] xfun_0.39         fastmap_1.1.1     withr_2.5.0       knitr_1.43       
[21] generics_0.1.3    vctrs_0.6.2       htmlwidgets_1.6.2 hms_1.1.3        
[25] grid_4.2.2        tidyselect_1.2.0  robustbase_0.95-1 glue_1.6.2       
[29] R6_2.5.1          fansi_1.0.4       bayesm_3.1-5      rmarkdown_2.22   
[33] tensorA_0.36.2    farver_2.1.1      tzdb_0.4.0        magrittr_2.0.3   
[37] MASS_7.3-58.1     scales_1.2.1      htmltools_0.5.5   colorspace_2.1-0 
[41] labeling_0.4.2    utf8_1.2.3        stringi_1.7.12    munsell_0.5.0    </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>