<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Analysis of metatranscriptome data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">Analysis of metatranscriptome data</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./env-metadata.html" class="sidebar-item-text sidebar-link">Water column profiles</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./process-eukrhythmic-output.html" class="sidebar-item-text sidebar-link">eukrhythmic output</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metat-analysis.html" class="sidebar-item-text sidebar-link active">Data analysis</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up-working-r-environment" id="toc-set-up-working-r-environment" class="nav-link active" data-scroll-target="#set-up-working-r-environment">Set up working R environment</a></li>
  <li><a href="#import-metatranscriptome-r-data-objects" id="toc-import-metatranscriptome-r-data-objects" class="nav-link" data-scroll-target="#import-metatranscriptome-r-data-objects">Import metatranscriptome R data objects</a>
  <ul class="collapse">
  <li><a href="#input-all-tpm-results" id="toc-input-all-tpm-results" class="nav-link" data-scroll-target="#input-all-tpm-results">Input all TPM results</a></li>
  </ul></li>
  <li><a href="#taxonomic-composition-distribution" id="toc-taxonomic-composition-distribution" class="nav-link" data-scroll-target="#taxonomic-composition-distribution">Taxonomic composition &amp; distribution</a>
  <ul class="collapse">
  <li><a href="#taxonomic-level-curation-and-estimate-relative-abundance" id="toc-taxonomic-level-curation-and-estimate-relative-abundance" class="nav-link" data-scroll-target="#taxonomic-level-curation-and-estimate-relative-abundance">Taxonomic level curation and estimate relative abundance</a></li>
  <li><a href="#taxonomy-bar-plot" id="toc-taxonomy-bar-plot" class="nav-link" data-scroll-target="#taxonomy-bar-plot">Taxonomy bar plot</a></li>
  </ul></li>
  <li><a href="#whole-community-functional-profile" id="toc-whole-community-functional-profile" class="nav-link" data-scroll-target="#whole-community-functional-profile">Whole community functional profile</a>
  <ul class="collapse">
  <li><a href="#pcoa-with-whole-community-functional-profile" id="toc-pcoa-with-whole-community-functional-profile" class="nav-link" data-scroll-target="#pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</a>
  <ul class="collapse">
  <li><a href="#does-pcoa-change-when-only-taxonomic-information-is-input" id="toc-does-pcoa-change-when-only-taxonomic-information-is-input" class="nav-link" data-scroll-target="#does-pcoa-change-when-only-taxonomic-information-is-input">Does PCoA change when only taxonomic information is input?</a></li>
  </ul></li>
  <li><a href="#define-each-site-by-module" id="toc-define-each-site-by-module" class="nav-link" data-scroll-target="#define-each-site-by-module">Define each site by module</a></li>
  </ul></li>
  <li><a href="#transcriptional-profiles-for-taxonomic-groups" id="toc-transcriptional-profiles-for-taxonomic-groups" class="nav-link" data-scroll-target="#transcriptional-profiles-for-taxonomic-groups">Transcriptional profiles for taxonomic groups</a>
  <ul class="collapse">
  <li><a href="#pcoa-with-curated-functional-profiles-and-known-taxonomic-ids." id="toc-pcoa-with-curated-functional-profiles-and-known-taxonomic-ids." class="nav-link" data-scroll-target="#pcoa-with-curated-functional-profiles-and-known-taxonomic-ids.">PCoA with curated functional profiles and known taxonomic IDs.</a>
  <ul class="collapse">
  <li><a href="#heterotrophy" id="toc-heterotrophy" class="nav-link" data-scroll-target="#heterotrophy">Heterotrophy</a></li>
  <li><a href="#n-p-metabolism" id="toc-n-p-metabolism" class="nav-link" data-scroll-target="#n-p-metabolism">N &amp; P metabolism</a></li>
  <li><a href="#determine-statistical-significance" id="toc-determine-statistical-significance" class="nav-link" data-scroll-target="#determine-statistical-significance">Determine statistical significance</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#functional-redundancy" id="toc-functional-redundancy" class="nav-link" data-scroll-target="#functional-redundancy">Functional redundancy</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">Analysis of metatranscriptome data</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="set-up-working-r-environment" class="level1">
<h1>Set up working R environment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(multidplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dbplyr)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RSQLite)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tictoc)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-metatranscriptome-r-data-objects" class="level1">
<h1>Import metatranscriptome R data objects</h1>
<section id="input-all-tpm-results" class="level2">
<h2 class="anchored" data-anchor-id="input-all-tpm-results">Input all TPM results</h2>
<p>Below R objects sourced by first isolating only the eukaryotic transcripts, averaging across replicates, and placing into long format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"/scratch/user/skhu/SPOT-ALOHA/Avg_tpm_ALL_acrossreps_annot.RData"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tpm_ALL_mean_reps_annot)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tpm_ALL_mean_reps_annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Use <code>multidplyr</code> to perform tidyverse functions in parallel.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># cluster</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tic</span>()</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>all_shard <span class="ot">&lt;-</span> tpm_ALL_mean_reps_annot <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLENAME) <span class="sc">%&gt;%</span> </span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">partition</span>(cluster)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">toc</span>() <span class="co">#95.245 sec elapsed</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Supergroup</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>supergroup_all <span class="ot">&lt;-</span> all_shard <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Domain, Supergroup, PACIFIC_REGION, SAMPLENAME, SAMPLE_ID_PARSED, DEPTH_CATEGORY) <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM_TPM_supergroup =</span> <span class="fu">sum</span>(MEAN_TPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">collect</span>()</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>supergroup_all <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ID_PARSED, <span class="at">y =</span> SUM_TPM_supergroup, <span class="at">fill =</span> Supergroup)) <span class="sc">+</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Phylum</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>phylum_all <span class="ot">&lt;-</span> all_shard <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Domain, Supergroup, Phylum, PACIFIC_REGION, SAMPLENAME, SAMPLE_ID_PARSED, DEPTH_CATEGORY) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">SUM_TPM_supergroup =</span> <span class="fu">sum</span>(MEAN_TPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">collect</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unite</span>(Supergroup_phylum, Supergroup, Phylum, <span class="at">sep =</span> <span class="st">" "</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>phylum_all <span class="sc">%&gt;%</span> </span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ID_PARSED, <span class="at">y =</span> SUM_TPM_supergroup, <span class="at">fill =</span> Supergroup_phylum)) <span class="sc">+</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="taxonomic-composition-distribution" class="level1">
<h1>Taxonomic composition &amp; distribution</h1>
<p>Evaluate taxonomic diversity and composition across all sites.</p>
<section id="taxonomic-level-curation-and-estimate-relative-abundance" class="level2">
<h2 class="anchored" data-anchor-id="taxonomic-level-curation-and-estimate-relative-abundance">Taxonomic level curation and estimate relative abundance</h2>
<p>Taxa curation 3) Average across replicates (or how should replicates be considered) 4) Sum to each sample</p>
</section>
<section id="taxonomy-bar-plot" class="level2">
<h2 class="anchored" data-anchor-id="taxonomy-bar-plot">Taxonomy bar plot</h2>
<ol start="5" type="1">
<li>bar plot of relative TPM? CPM? for major taxnomic groups Keep ALOHA July and April separate, ALOHA on the left side and on the right side, all CA sites. Also place “sun” emoji for samples that are within the euphotic zone</li>
</ol>
</section>
</section>
<section id="whole-community-functional-profile" class="level1">
<h1>Whole community functional profile</h1>
<p>Prepare data ahead of time. Use normalization to the whole community. Include functional modules that were curated for this work.</p>
<blockquote class="blockquote">
<p>What percentage of the data are represented when we only look at the curated data?</p>
</blockquote>
<p>Change input data to all numeric for downstream transformations, etc.</p>
<ol type="1">
<li><p>all data, change zeros to 1 (or to below the lowest value)</p></li>
<li><p>all data, remove NAs and zeros. Only represent proteins that appear in all samples</p></li>
<li><p>Subset to only those protein IDs of interest (get stats on what this is) Make sure to retain module information in definiation. 3.1) change zeroes to 1 (or lowest) 3.2) remove NAs</p></li>
</ol>
<section id="pcoa-with-whole-community-functional-profile" class="level2">
<h2 class="anchored" data-anchor-id="pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(tpm_all_euk_clr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>pca_lr <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">as.data.frame</span>(tpm_all_euk_clr))</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>variance_lr <span class="ot">&lt;-</span> (pca_lr<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)<span class="sc">/</span><span class="fu">sum</span>(pca_lr<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="do">## View bar plot</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(variance_lr, <span class="at">main =</span> <span class="st">"Log-Ratio PCA Screeplot"</span>, <span class="at">xlab =</span> <span class="st">"PC Axis"</span>, <span class="at">ylab =</span> <span class="st">"% Variance"</span>, </span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.names =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="does-pcoa-change-when-only-taxonomic-information-is-input" class="level3">
<h3 class="anchored" data-anchor-id="does-pcoa-change-when-only-taxonomic-information-is-input">Does PCoA change when only taxonomic information is input?</h3>
<p>(supplementary)</p>
</section>
</section>
<section id="define-each-site-by-module" class="level2">
<h2 class="anchored" data-anchor-id="define-each-site-by-module">Define each site by module</h2>
<p>(k-means?)</p>
</section>
</section>
<section id="transcriptional-profiles-for-taxonomic-groups" class="level1">
<h1>Transcriptional profiles for taxonomic groups</h1>
<p>(supplementary?) Normalized by individual taxonomic groups.</p>
<section id="pcoa-with-curated-functional-profiles-and-known-taxonomic-ids." class="level2">
<h2 class="anchored" data-anchor-id="pcoa-with-curated-functional-profiles-and-known-taxonomic-ids.">PCoA with curated functional profiles and known taxonomic IDs.</h2>
<p>(subset this way to increase confidence?)</p>
<section id="heterotrophy" class="level3">
<h3 class="anchored" data-anchor-id="heterotrophy">Heterotrophy</h3>
<p>Isolate and sum to pathways associated with heterotrophy. Gene level is likely too much. Determine for each</p>
<blockquote class="blockquote">
<p>How varied are the transcriptional signatures related to heterotrophy among protists?</p>
</blockquote>
<p>Use flag/half square heatmap, where top half shows the percent number of genes and bottom half shows the TPM summary.</p>
</section>
<section id="n-p-metabolism" class="level3">
<h3 class="anchored" data-anchor-id="n-p-metabolism">N &amp; P metabolism</h3>
<p>Isolate at individual gene level for N &amp; P pathways. Show TPM across sites and core taxa</p>
<blockquote class="blockquote">
<p>For ecologically-important phytoplankton at ALOHA and coastal CA, what is the relative importance of nutrient profiles? nutrient utilization?</p>
</blockquote>
<p>Plot heat map for each taxa. Show all N and P relevant genes on the x axis and y axis should have all samples.</p>
</section>
<section id="determine-statistical-significance" class="level3">
<h3 class="anchored" data-anchor-id="determine-statistical-significance">Determine statistical significance</h3>
<p>Estimate statistical significance for N&amp;P genes, pairwise comparisons. Then add a column for each gene ID that lists the number of samples that are statistically different.</p>
</section>
</section>
</section>
<section id="functional-redundancy" class="level1">
<h1>Functional redundancy</h1>
<p>Input data is all normalized. Then perform pairwise Jaccard dissimilarity with taxa and then with KEGG/pfam IDs - plot as x/y</p>
</section>
<section id="session-info" class="level1">
<h1>Session info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>