<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Analysis of metatranscriptome data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./metat-analysis.html">Data analysis</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation docked overflow-auto">
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Home</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./env-metadata.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Water column profiles</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./process-eukrhythmic-output.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">eukrhythmic output</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metat-analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Data analysis</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#set-up-working-r-environment" id="toc-set-up-working-r-environment" class="nav-link active" data-scroll-target="#set-up-working-r-environment">Set up working R environment</a></li>
  <li><a href="#import-metatranscriptome-r-data-objects" id="toc-import-metatranscriptome-r-data-objects" class="nav-link" data-scroll-target="#import-metatranscriptome-r-data-objects">Import metatranscriptome R data objects</a>
  <ul class="collapse">
  <li><a href="#input-all-tpm-results" id="toc-input-all-tpm-results" class="nav-link" data-scroll-target="#input-all-tpm-results">Input all TPM results</a></li>
  <li><a href="#taxonomic-composition-distribution" id="toc-taxonomic-composition-distribution" class="nav-link" data-scroll-target="#taxonomic-composition-distribution">Taxonomic composition &amp; distribution</a>
  <ul class="collapse">
  <li><a href="#plot-domain-level" id="toc-plot-domain-level" class="nav-link" data-scroll-target="#plot-domain-level">Plot domain level</a></li>
  <li><a href="#plot-to-supergroup" id="toc-plot-to-supergroup" class="nav-link" data-scroll-target="#plot-to-supergroup">Plot to supergroup</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#whole-community-functional-profile" id="toc-whole-community-functional-profile" class="nav-link" data-scroll-target="#whole-community-functional-profile">Whole community functional profile</a>
  <ul class="collapse">
  <li><a href="#pcoa-with-whole-community-functional-profile" id="toc-pcoa-with-whole-community-functional-profile" class="nav-link" data-scroll-target="#pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</a></li>
  </ul></li>
  <li><a href="#taxonomic-breakdown" id="toc-taxonomic-breakdown" class="nav-link" data-scroll-target="#taxonomic-breakdown">Taxonomic breakdown</a></li>
  <li><a href="#differentially-expressed-genes" id="toc-differentially-expressed-genes" class="nav-link" data-scroll-target="#differentially-expressed-genes">Differentially expressed genes</a>
  <ul class="collapse">
  <li><a href="#light-vs.-region-in-the-pac-ocean" id="toc-light-vs.-region-in-the-pac-ocean" class="nav-link" data-scroll-target="#light-vs.-region-in-the-pac-ocean">Light vs.&nbsp;Region in the Pac Ocean</a></li>
  <li><a href="#ca-and-npsg-only" id="toc-ca-and-npsg-only" class="nav-link" data-scroll-target="#ca-and-npsg-only">CA and NPSG only</a></li>
  <li><a href="#euphotic-and-subeuphotic-only" id="toc-euphotic-and-subeuphotic-only" class="nav-link" data-scroll-target="#euphotic-and-subeuphotic-only">Euphotic and subeuphotic only</a>
  <ul class="collapse">
  <li><a href="#bring-locally." id="toc-bring-locally." class="nav-link" data-scroll-target="#bring-locally.">Bring locally.</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#closer-look-at-functional-profiles" id="toc-closer-look-at-functional-profiles" class="nav-link" data-scroll-target="#closer-look-at-functional-profiles">Closer look at functional profiles</a>
  <ul class="collapse">
  <li><a href="#heterotrophy" id="toc-heterotrophy" class="nav-link" data-scroll-target="#heterotrophy">Heterotrophy</a></li>
  <li><a href="#n-p-metabolism" id="toc-n-p-metabolism" class="nav-link" data-scroll-target="#n-p-metabolism">N &amp; P metabolism</a></li>
  <li><a href="#determine-statistical-significance" id="toc-determine-statistical-significance" class="nav-link" data-scroll-target="#determine-statistical-significance">Determine statistical significance</a></li>
  </ul></li>
  <li><a href="#functional-redundancy" id="toc-functional-redundancy" class="nav-link" data-scroll-target="#functional-redundancy">Functional redundancy</a></li>
  <li><a href="#session-info" id="toc-session-info" class="nav-link" data-scroll-target="#session-info">Session info</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analysis of metatranscriptome data</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="set-up-working-r-environment" class="level1">
<h1>Set up working R environment</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'data.table'

The following objects are masked from 'package:lubridate':

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from 'package:dplyr':

    between, first, last

The following object is masked from 'package:purrr':

    transpose</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(compositions)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to compositions, a package for compositional data analysis.
Find an intro with "? compositions"


Attaching package: 'compositions'

The following objects are masked from 'package:stats':

    anova, cor, cov, dist, var

The following object is masked from 'package:graphics':

    segments

The following objects are masked from 'package:base':

    %*%, norm, scale, scale.default</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="import-metatranscriptome-r-data-objects" class="level1">
<h1>Import metatranscriptome R data objects</h1>
<section id="input-all-tpm-results" class="level2">
<h2 class="anchored" data-anchor-id="input-all-tpm-results">Input all TPM results</h2>
<p>Below R objects sourced by first isolating only the eukaryotic transcripts, averaging across replicates, and placing into long format.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"Avg_scaled_tpm_08222023.RData"</span>, <span class="at">verbose =</span> T)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>taxfxn <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"TaxonomicAndFunctionalAnnotations.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Import metadata as well.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># metadata_tmp &lt;- read.delim("input-data/complete-sample-list.txt", header = TRUE, sep = ",")</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sample_metadata &lt;- read.csv("input-data/sample_table.csv") %&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># unite(SAMPLE, Sample, Replicate, sep = "_", remove = FALSE) %&gt;% </span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select(RUN = SRR, everything()) %&gt;% </span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># right_join(metadata_tmp)</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(sample_metadata, file = "input-data/sample-metadata.csv")</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>sample_metadata_1 <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"input-data/sample-metadata.csv"</span>)</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>sample_metadata <span class="ot">&lt;-</span> sample_metadata_1 <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE_MATCH =</span> <span class="fu">sub</span>(<span class="st">"_[^_]+$"</span>, <span class="st">""</span>, SAMPLE))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxonomic-composition-distribution" class="level2">
<h2 class="anchored" data-anchor-id="taxonomic-composition-distribution">Taxonomic composition &amp; distribution</h2>
<p>Evaluate taxonomic diversity and composition across all sites.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(taxfxn<span class="sc">$</span>SequenceID))</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 14840771</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">rownames</span>(mean_counts_df)))</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 14737135</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>annotated_mean <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(mean_counts_df <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">mutate</span>(<span class="at">SequenceID =</span> <span class="fu">rownames</span>(mean_counts_df)),</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                                 taxfxn,</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">by =</span> <span class="st">"SequenceID"</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 14737135-14737693</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="do">## 558 were duplicated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Remove the duplicated annotations and make a base taxonomic dataframe for downstream work.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>tax_plot <span class="ot">&lt;-</span> annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Domain<span class="sc">:</span>Species, <span class="fu">starts_with</span>(<span class="st">"mean"</span>), SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># dim(tax_plot) #14737624\</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/tax_plot_08222023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | echo: true</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | eval: false</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>tax_only <span class="ot">&lt;-</span> tax_plot <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"mean"</span>), <span class="at">names_to =</span> <span class="st">"SAMPLE"</span>, <span class="at">values_to =</span> <span class="st">"scaledTPM"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">SUM_scaledTPM =</span> <span class="fu">sum</span>(scaledTPM)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/tax_only_08242023.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  tax_only</code></pre>
</div>
</div>
<section id="plot-domain-level" class="level3">
<h3 class="anchored" data-anchor-id="plot-domain-level">Plot domain level</h3>
<p>Creating plot to see total number and relative abundance of non-eukaryotic hits in the data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>(tax_only <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> DOMAIN)) <span class="sc">+</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"stack"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))) <span class="sc">/</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>(tax_only <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> DOMAIN)) <span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="480"></p>
</div>
</div>
<p>Stats on what % of the data are eukaryotic vs.&nbsp;not.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>SUM_TOTAL_TPM <span class="ot">&lt;-</span> <span class="fu">sum</span>(tax_only<span class="sc">$</span>SUM_scaledTPM)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>tax_only <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SUM_scaledTPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span>  <span class="co"># Remove zeroes</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(DOMAIN) <span class="sc">%&gt;%</span> </span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">Sum_of_TPM =</span> <span class="fu">sum</span>(SUM_scaledTPM),</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>              <span class="at">Perc_of_total =</span> <span class="dv">100</span><span class="sc">*</span>(Sum_of_TPM<span class="sc">/</span>SUM_TOTAL_TPM),</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">.groups =</span> <span class="st">"rowwise"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 3
# Rowwise:  DOMAIN
  DOMAIN       Sum_of_TPM Perc_of_total
  &lt;chr&gt;             &lt;dbl&gt;         &lt;dbl&gt;
1 Archaea         305556.        0.159 
2 Bacteria       5438767.        2.84  
3 Eukaryota    185858173.       96.9   
4 Unclassified    174159.        0.0908</code></pre>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Count = count(DOMAIN))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plot-to-supergroup" class="level3">
<h3 class="anchored" data-anchor-id="plot-to-supergroup">Plot to supergroup</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tax_only)</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>tax_only <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SUPERGROUP =</span> <span class="fu">str_squish</span>(Supergroup),</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">PHYLUM =</span> <span class="fu">str_squish</span>(Phylum),</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DOMAIN <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE, Supergroup) <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE, <span class="at">y =</span> SUM, <span class="at">fill =</span> Supergroup)) <span class="sc">+</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE'. You can override using the
`.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Tally composition of the supergroup.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tax_only)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "SAMPLE"        "Domain"        "Supergroup"    "Phylum"       
 [5] "Class"         "Order"         "Family"        "Genus"        
 [9] "Species"       "SUM_scaledTPM"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># unique(tax_only$Phylum)</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>tax_names <span class="ot">&lt;-</span> tax_only <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">DOMAIN =</span> <span class="fu">str_squish</span>(Domain),</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">SUPERGROUP =</span> <span class="fu">str_squish</span>(Supergroup),</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">PHYLUM =</span> <span class="fu">str_squish</span>(Phylum),</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">SAMPLE =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(DOMAIN <span class="sc">==</span> <span class="st">"Eukaryota"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(DOMAIN, SUPERGROUP, Class<span class="sc">:</span>Species) <span class="sc">%&gt;%</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Adding missing grouping variables: `SAMPLE`, `Domain`, `Supergroup`, `Phylum`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_names<span class="sc">$</span>Phylum)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] " Apicomplexa"      " Ciliophora"       " Dinoflagellata"  
 [4] " Perkinsea"        "Unclassified"      " Conosa"          
 [7] " Discosea"         " Lobosa"           " Chlorophyta"     
[10] " Glaucophyta"      " Rhodophyta"       " Discoba"         
[13] " Cryptophyta"      " Hacrobia_X"       " Haptophyta"      
[16] " Choanoflagellida" " Fungi"            " Cercozoa"        
[19] " Foraminifera"     " Ochrophyta"       " Opalozoa"        
[22] " Sagenista"        " Stramenopiles_X" </code></pre>
</div>
</div>
<p>Now, reformat taxonomic information to visualize.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(tax_only<span class="sc">$</span>SAMPLE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "mean.Catalina"     "mean.July_1000m"   "mean.July_150m"   
 [4] "mean.July_5m"      "mean.July_DCM"     "mean.March_1000m" 
 [7] "mean.March_150m"   "mean.March_5m"     "mean.March_DCM"   
[10] "mean.PortofLA"     "mean.SPOT_150m"    "mean.SPOT_890m"   
[13] "mean.SPOT_surface"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colnames(tax_only)</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="co"># head(sample_metadata)</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>tax_formatted <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax_only) <span class="sc">%&gt;%</span></span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove preceding spaces in taxonomic names</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(Domain, Supergroup, Phylum, Class, Order, Family, Genus, Species), <span class="sc">~</span> <span class="fu">as.character</span>(<span class="fu">str_squish</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> SUM_scaledTPM <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SAMPLE_MATCH =</span> <span class="fu">str_remove</span>(SAMPLE, <span class="st">"mean."</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="st">"SAMPLE_MATCH"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Taxonomic group curation</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">case_when</span>(</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>    Supergroup <span class="sc">==</span> <span class="st">"Alveolata"</span> <span class="sc">~</span> <span class="fu">paste</span>(Supergroup, Phylum, <span class="at">sep =</span> <span class="st">"-"</span>),</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> Supergroup</span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  ))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(., sample_metadata, by = "SAMPLE_MATCH"): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 13 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tax_formatted)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Assign levels for taxonomic groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>tax_order_color<span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">"#612741"</span>,<span class="st">"#b74a70"</span>,<span class="st">"#b7757c"</span>,<span class="st">"#eecfbf"</span>,<span class="st">"#92462f"</span>,<span class="st">"#bb603c"</span>,<span class="st">"#dfa837"</span>,<span class="st">"#ccc050"</span>,<span class="st">"#33431e"</span>,<span class="st">"#93b778"</span>,<span class="st">"#61ac86"</span>,<span class="st">"#657abb"</span>,<span class="st">"#1c1949"</span>,<span class="st">"#8a8d84"</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>supergroup_order <span class="ot">&lt;-</span> (<span class="fu">as.character</span>(<span class="fu">unique</span>(tax_formatted<span class="sc">$</span>SUPERGROUP)))</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tax_order_color) <span class="ot">&lt;-</span> (supergroup_order)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="figure-1" class="level4">
<h4 class="anchored" data-anchor-id="figure-1">Figure 1</h4>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>tax_formatted <span class="sc">%&gt;%</span> </span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUPERGROUP =</span> <span class="fu">factor</span>(tax_formatted<span class="sc">$</span>SUPERGROUP, <span class="at">levels =</span> supergroup_order)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SAMPLE_ID_PARSED, PACIFIC_REGION, LIGHT, DEPTH, SUPERGROUP) <span class="sc">%&gt;%</span> </span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">SUM =</span> <span class="fu">sum</span>(SUM_scaledTPM)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> SAMPLE_ID_PARSED, <span class="at">y =</span> SUM, <span class="at">fill =</span> SUPERGROUP)) <span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"fill"</span>, <span class="at">color =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(PACIFIC_REGION), <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">space =</span> <span class="st">"free"</span>) <span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> tax_order_color) <span class="sc">+</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"TPM Relative Abundance"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'SAMPLE_ID_PARSED', 'PACIFIC_REGION',
'LIGHT', 'DEPTH'. You can override using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
</section>
</section>
<section id="whole-community-functional-profile" class="level1">
<h1>Whole community functional profile</h1>
<p>Prepare data ahead of time. Use normalization to the whole community. Include functional modules that were curated for this work.</p>
<p>Import normalized over center data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load("normed_center_df_08222023.RData", verbose = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This represents a toy dataset to troubleshoot code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/toy_1000.Rdata"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  annotated_mean_10000
  ctr_norm_10000</code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change to original names, so code below can be written</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>annotated_mean <span class="ot">&lt;-</span> annotated_mean_10000</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>df_ctr_norm <span class="ot">&lt;-</span> ctr_norm_10000</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>How many transcripts are annotated?</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>             <span class="fu">count</span>(<span class="at">output =</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">name =</span> <span class="st">"KEGGs"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"PFAMs"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> GOs <span class="sc">!=</span> <span class="st">"-"</span>, </span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"GOs"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>             <span class="fu">left_join</span>(annotated_mean <span class="sc">%&gt;%</span> </span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">count</span>(<span class="at">output =</span> (GOs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>), </span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">name =</span> <span class="st">"All"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(output <span class="sc">==</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>output) <span class="sc">%&gt;%</span> </span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">everything</span>(), <span class="at">values_to =</span> <span class="st">"Number annotated"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Percent_annotated =</span> <span class="dv">100</span><span class="sc">*</span>(<span class="st">`</span><span class="at">Number annotated</span><span class="st">`</span><span class="sc">/</span><span class="fu">length</span>(annotated_mean<span class="sc">$</span>SequenceID)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<table class="table">
<thead>
<tr class="header">
<th>DB</th>
<th>Number of seqs annotated</th>
<th>Percent annotated</th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>KEGGs</td>
<td>9970784</td>
<td>67.7</td>
<td></td>
</tr>
<tr class="even">
<td>PFAMs</td>
<td>13595566</td>
<td>92.3</td>
<td></td>
</tr>
<tr class="odd">
<td>GOs</td>
<td>5442486</td>
<td>36.9</td>
<td></td>
</tr>
<tr class="even">
<td>All</td>
<td>4963542</td>
<td>33.7</td>
<td></td>
</tr>
</tbody>
</table>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>summary_annotated <span class="ot">&lt;-</span> annotated_mean <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">SUM =</span> <span class="fu">rowSums</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric)))) <span class="sc">%&gt;%</span></span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(SUM, GOs, PFAMs, KEGG_ko, Domain, Supergroup)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(summary_annotated, file = "summary_of_annotated_totalTPM.csv")</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co"># sum(summary_annotated$SUM) # TPM amount that is annotated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="pcoa-with-whole-community-functional-profile" class="level2">
<h2 class="anchored" data-anchor-id="pcoa-with-whole-community-functional-profile">PCoA with whole community functional profile</h2>
<p>Import PCA objects from running R on the HPC.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"input-data/pca_objects.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Loading objects:
  pca_df
  variance_lr</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pca_lr &lt;- prcomp(as.data.frame(t(df_ctr_norm))) # done via HPC</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="co"># variance_lr &lt;- (pca_lr$sdev^2)/sum(pca_lr$sdev^2)</span></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># ## View bar plot</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="co"># svg("screeplot.svg", h = 6, w = 9)</span></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(variance_lr, <span class="at">main =</span> <span class="st">"Log-Ratio PCA Screeplot"</span>, <span class="at">xlab =</span> <span class="st">"PC Axis"</span>, <span class="at">ylab =</span> <span class="st">"% Variance"</span>, <span class="at">cex.names =</span> <span class="fl">1.5</span>, <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dev.off()</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pca_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>            SAMPLE_MATCH       PC1       PC2        PC3
PortofLA_4    PortofLA_4 -704.4176 2145.6806 1696.13738
PortofLA_3    PortofLA_3 -578.4998 2446.1131 2032.43291
PortofLA_2    PortofLA_2 -425.8412 2750.0947 2354.43634
PortofLA_1    PortofLA_1 -295.0105 2854.7883 2500.92913
Catalina_24  Catalina_24 2282.0410 -649.0612   18.82071
Catalina_23  Catalina_23 2007.2327 -560.7567   12.59361</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>pc1 <span class="ot">&lt;-</span> <span class="fu">round</span>(variance_lr[<span class="dv">1</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>pc2 <span class="ot">&lt;-</span> <span class="fu">round</span>(variance_lr[<span class="dv">2</span>] <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample_metadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  X        RUN         SAMN                 SAMPLE     Sample   Replicate
1 1 SRR6048900 SAMN07647713 July_1000m_Rep1andRep2 July_1000m Rep1andRep2
2 2 SRR6048899 SAMN07647714  July_150m_Rep1andRep2  July_150m Rep1andRep2
3 3 SRR6048898 SAMN07647715           July_5m_Rep1    July_5m        Rep1
4 4 SRR6048897 SAMN07647716           July_5m_Rep2    July_5m        Rep2
5 5 SRR6048896 SAMN07647717          July_DCM_Rep1   July_DCM        Rep1
6 6 SRR6048895 SAMN07647718          July_DCM_Rep2   July_DCM        Rep2
        BASES  BIOPROJECT    BIOSAMPLE       DATE EXPERIMENT
1 22179808750 PRJNA406025 SAMN07647713 2015-07-30 SRX3195831
2 22251034000 PRJNA406025 SAMN07647714 2015-07-30 SRX3195832
3 36858334750 PRJNA406025 SAMN07647715 2015-07-30 SRX3195833
4 28427020000 PRJNA406025 SAMN07647716 2015-07-30 SRX3195834
5 33698315000 PRJNA406025 SAMN07647717 2015-07-30 SRX3195835
6 30540287750 PRJNA406025 SAMN07647718 2015-07-30 SRX3195836
                 LAT_LON           LIBRARY           SAMPLENAME DEPTH
1 24.54987 N 156.33178 W        July_1000m NPSG_ALOHA_1000_July  1000
2 24.54987 N 156.33178 W         July_150m  NPSG_ALOHA_150_July   150
3 24.54987 N 156.33178 W July_surface_Rep1    NPSG_ALOHA_5_July     5
4 24.54987 N 156.33178 W July_surface_Rep2    NPSG_ALOHA_5_July     5
5 24.54987 N 156.33178 W     July_DCM_Rep1  NPSG_ALOHA_119_July   119
6 24.54987 N 156.33178 W     July_DCM_Rep2  NPSG_ALOHA_119_July   119
    SAMPLE_ID_PARSED DEPTH_CATEGORY REGION PACIFIC_REGION MONTH        LIGHT
1  July ALOHA 1000 m         1000 m  ALOHA           NPSG  July Sub-euphotic
2   July ALOHA 150 m          150 m  ALOHA           NPSG  July     Euphotic
3 July ALOHA surface        surface  ALOHA           NPSG  July     Euphotic
4 July ALOHA surface        surface  ALOHA           NPSG  July     Euphotic
5     July ALOHA DCM            DCM  ALOHA           NPSG  July     Euphotic
6     July ALOHA DCM            DCM  ALOHA           NPSG  July     Euphotic
  SAMPLE_MATCH
1   July_1000m
2    July_150m
3      July_5m
4      July_5m
5     July_DCM
6     July_DCM</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># svg("Rplot_metat_pca_wholecommunity.svg", width = 7, height = 7)</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>pca_df <span class="sc">%&gt;%</span> </span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(sample_metadata, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"SAMPLE_MATCH"</span> <span class="ot">=</span> <span class="st">"SAMPLE"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2)) <span class="sc">+</span></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">stroke =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="fu">aes</span>(<span class="at">shape =</span> REGION, <span class="at">fill =</span> DEPTH_CATEGORY, <span class="at">color =</span> LIGHT)) <span class="sc">+</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">22</span>, <span class="dv">23</span>, <span class="dv">24</span>)) <span class="sc">+</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#6a51a3"</span>, <span class="st">"#f768a1"</span>, <span class="st">"#8c96c6"</span>, <span class="st">"#fd8d3c"</span>, <span class="st">"#238b45"</span>)) <span class="sc">+</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#cccccc"</span>, <span class="st">"#252525"</span>)) <span class="sc">+</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_linedraw</span>() <span class="sc">+</span> <span class="fu">coord_fixed</span>(<span class="at">ratio =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">guides</span>(<span class="at">fill =</span> <span class="fu">guide_legend</span>(<span class="at">override.aes =</span> <span class="fu">list</span>(<span class="at">shape =</span> <span class="fu">c</span>(<span class="dv">22</span>)))) <span class="sc">+</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.title =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> <span class="dv">13</span>)) <span class="sc">+</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span>  <span class="fu">paste</span>(pc1, <span class="st">"%"</span>), <span class="at">y =</span> <span class="fu">paste</span>(pc2, <span class="st">"%"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="metat-analysis_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="taxonomic-breakdown" class="level1">
<h1>Taxonomic breakdown</h1>
<p>By depth, look at taxonomic breakdown at higher resolution. Use the center normalized data - separating this specifically by sub-euphotic and euphotic to pair with the DESeq results.</p>
<p>This represents a toy dataset to troubleshoot code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load("input-data/toy_1000.Rdata", verbose = TRUE)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Change to original names, so code below can be written</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="co"># annotated_mean &lt;- annotated_mean_10000</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co"># df_ctr_norm &lt;- ctr_norm_10000</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Aug 25 **This the above toy data set to work with the ctr normalized data. Get the taxa sorted - what is the taxonomic resolution associated with euphotic vs.&nbsp;subeuphotic.</p>
</blockquote>
</section>
<section id="differentially-expressed-genes" class="level1">
<h1>Differentially expressed genes</h1>
<p>Resources: 1. http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#the-deseqdataset</p>
<p>Pre-filtering / pre-processing data: Remove counts that are less than 10, and make sure the minimal number of samples that a count appears is 3.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="light-vs.-region-in-the-pac-ocean" class="level3">
<h3 class="anchored" data-anchor-id="light-vs.-region-in-the-pac-ocean">Light vs.&nbsp;Region in the Pac Ocean</h3>
<p>Using the whole dataset, these are the comparisons.</p>
<p>light: upregulated in the euphotic zone vs subeuphotic pac: upregulated in NPSG vs CA lightpac - combo!?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"light-pacregion-deseq.RData"</span>)</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ds_tpm_light</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ds_tpm_pac</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ds_tpm_lightpac</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_light) <span class="sc">&gt;=</span> <span class="dv">100</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_light[keep,]</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered<span class="sc">$</span>LIGHT <span class="ot">&lt;-</span> <span class="fu">factor</span>(ds_tpm_light_filtered<span class="sc">$</span>LIGHT, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Euphotic"</span>, <span class="st">"Sub-euphotic"</span>))</span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-12"><a href="#cb55-12" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb55-13"><a href="#cb55-13" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb55-14"><a href="#cb55-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-15"><a href="#cb55-15" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb55-16"><a href="#cb55-16" aria-hidden="true" tabindex="-1"></a>res_light <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare with the Pacific Ocean</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_pac) <span class="sc">&gt;=</span> <span class="dv">100</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered <span class="ot">&lt;-</span> ds_tpm_pac[keep,]</span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CA"</span>, <span class="st">"NPSG"</span>))</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(dds_obj_filtered) </span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>res_pacific <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare with Pacific ocean and light</span></span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_lightpac) <span class="sc">&gt;=</span> <span class="dv">100</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered <span class="ot">&lt;-</span> ds_tpm_lightpac[keep,]</span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered<span class="sc">$</span>LIGHT <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds_obj_filtered<span class="sc">$</span>LIGHT, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Euphotic"</span>, <span class="st">"Sub-euphotic"</span>))</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(dds_obj_filtered)</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>res_lightpac <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(res_light, res_pacific, res_lightpac, <span class="at">file =</span> <span class="st">"DE_results_light.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="ca-and-npsg-only" class="level3">
<h3 class="anchored" data-anchor-id="ca-and-npsg-only">CA and NPSG only</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/npsg-deseq.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="at">file =</span> <span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/ca-deseq.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds_tpm_npsg_light</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="co"># ds_tpm_ca_light</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>npsg_light: upreg in euphotic on npsg ca_light : upreg in euphotic in CA *use the month NPSG one for supplement -it should support that July and March are being used as Biological replicates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_npsg_light) <span class="sc">&gt;=</span> <span class="dv">50</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_npsg_light[keep,]</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered<span class="sc">$</span>LIGHT <span class="ot">&lt;-</span> <span class="fu">factor</span>(ds_tpm_light_filtered<span class="sc">$</span>LIGHT, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Euphotic"</span>, <span class="st">"Sub-euphotic"</span>))</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb60-8"><a href="#cb60-8" aria-hidden="true" tabindex="-1"></a>res_npsg_light <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check NPSG month - here - place in supplement.</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_npsg_month) <span class="sc">&gt;=</span> <span class="dv">50</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_npsg_month[keep,]</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered<span class="sc">$</span>MONTH <span class="ot">&lt;-</span> <span class="fu">factor</span>(ds_tpm_light_filtered<span class="sc">$</span>MONTH, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"July"</span>, <span class="st">"March"</span>))</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>res_npsg_month <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="co"># CA only</span></span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_ca_light) <span class="sc">&gt;=</span> <span class="dv">50</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_ca_light[keep,]</span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered<span class="sc">$</span>LIGHT <span class="ot">&lt;-</span> </span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="fu">factor</span>(ds_tpm_light_filtered<span class="sc">$</span>LIGHT, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Euphotic"</span>, <span class="st">"Sub-euphotic"</span>))</span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>res_ca_light <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(res_npsg_month, res_npsg_light, res_ca_light, <span class="at">file =</span> <span class="st">"DE_results_regional.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="euphotic-and-subeuphotic-only" class="level2">
<h2 class="anchored" data-anchor-id="euphotic-and-subeuphotic-only">Euphotic and subeuphotic only</h2>
<p>subeuphotic:in the subeuphotic, upreg in npsg or CA euphotic: in the euphotic only, upreg in NPSG or CA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/euphotic-subeuphotic-deseq.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Euphotic</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_euphotic) <span class="sc">&gt;=</span> <span class="dv">50</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_euphotic[keep,]</span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CA"</span>, <span class="st">"NPSG"</span>))</span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb65-11"><a href="#cb65-11" aria-hidden="true" tabindex="-1"></a>res_euphotic_reg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subeuphotic</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>groupsize <span class="ot">&lt;-</span> <span class="dv">2</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_subeuphotic) <span class="sc">&gt;=</span> <span class="dv">50</span>) <span class="sc">&gt;=</span> groupsize</span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light_filtered <span class="ot">&lt;-</span> ds_tpm_subeuphotic[keep,]</span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION <span class="ot">&lt;-</span> <span class="fu">factor</span>(dds_obj_filtered<span class="sc">$</span>PACIFIC_REGION, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CA"</span>, <span class="st">"NPSG"</span>))</span>
<span id="cb66-7"><a href="#cb66-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-8"><a href="#cb66-8" aria-hidden="true" tabindex="-1"></a><span class="co"># this step takes time</span></span>
<span id="cb66-9"><a href="#cb66-9" aria-hidden="true" tabindex="-1"></a>dds_light <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_light_filtered) </span>
<span id="cb66-10"><a href="#cb66-10" aria-hidden="true" tabindex="-1"></a><span class="co"># report results</span></span>
<span id="cb66-11"><a href="#cb66-11" aria-hidden="true" tabindex="-1"></a>res_subeuphotic_reg <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_light)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(res_subeuphotic_reg, res_euphotic_reg, <span class="at">file =</span> <span class="st">"DE_results_region_light.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Aug 25 ** this is all running on poseidon!</p>
</blockquote>
<section id="bring-locally." class="level3">
<h3 class="anchored" data-anchor-id="bring-locally.">Bring locally.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load("input-data/DE_results_light.RData", verbose = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get stats on these outputs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(res_light)</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="co"># summary(res_pacific)</span></span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a><span class="co"># hist(res_pacific$log2FoldChange)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plotMA(res_light)</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="co"># plotMA(res_lightpac)</span></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plotMA(res_light_05)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="closer-look-at-functional-profiles" class="level1">
<h1>Closer look at functional profiles</h1>
<p>KEGG IDs that were manually curated were pulled out and used in the TXIMPORT subset functions. Here we are importing the results to be used for visualizations and stats. For the normalized center results, we can use this for heat maps, etc.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load("input-data/Avg_scaled_tpm_curated_08252023.RData", verbose = TRUE)</span></span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="co"># load("input-data/normed_center_df_curated_08252023.RData", verbose = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Aug 25 - see above outpus from poseidon. these are going to give me options for heatmaps and more! Look at heterotrophic and N&amp;P metabolism for NPSG vs CA.</p>
</blockquote>
<section id="heterotrophy" class="level3">
<h3 class="anchored" data-anchor-id="heterotrophy">Heterotrophy</h3>
<p>Isolate and sum to pathways associated with heterotrophy. Gene level is likely too much. Determine for each</p>
<blockquote class="blockquote">
<p>How varied are the transcriptional signatures related to heterotrophy among protists?</p>
</blockquote>
<p>Use flag/half square heatmap, where top half shows the percent number of genes and bottom half shows the TPM summary.</p>
</section>
<section id="n-p-metabolism" class="level3">
<h3 class="anchored" data-anchor-id="n-p-metabolism">N &amp; P metabolism</h3>
<p>Isolate at individual gene level for N &amp; P pathways. Show TPM across sites and core taxa</p>
<blockquote class="blockquote">
<p>For ecologically-important phytoplankton at ALOHA and coastal CA, what is the relative importance of nutrient profiles? nutrient utilization?</p>
</blockquote>
<p>Plot heat map for each taxa. Show all N and P relevant genes on the x axis and y axis should have all samples.</p>
</section>
<section id="determine-statistical-significance" class="level3">
<h3 class="anchored" data-anchor-id="determine-statistical-significance">Determine statistical significance</h3>
<p>Estimate statistical significance for N&amp;P genes, pairwise comparisons. Then add a column for each gene ID that lists the number of samples that are statistically different.</p>
</section>
</section>
<section id="functional-redundancy" class="level1">
<h1>Functional redundancy</h1>
<p>Input data is all normalized. Then perform pairwise Jaccard dissimilarity with taxa and then with KEGG/pfam IDs - plot as x/y</p>
</section>
<section id="session-info" class="level1">
<h1>Session info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] DESeq2_1.38.3               SummarizedExperiment_1.28.0
 [3] Biobase_2.58.0              MatrixGenerics_1.10.0      
 [5] matrixStats_1.0.0           GenomicRanges_1.50.2       
 [7] GenomeInfoDb_1.34.9         IRanges_2.32.0             
 [9] S4Vectors_0.36.2            BiocGenerics_0.44.0        
[11] patchwork_1.1.2             compositions_2.0-6         
[13] data.table_1.14.8           lubridate_1.9.2            
[15] forcats_1.0.0               stringr_1.5.0              
[17] dplyr_1.1.2                 purrr_1.0.1                
[19] readr_2.1.4                 tidyr_1.3.0                
[21] tibble_3.2.1                ggplot2_3.4.2              
[23] tidyverse_2.0.0            

loaded via a namespace (and not attached):
 [1] httr_1.4.6             bit64_4.0.5            jsonlite_1.8.5        
 [4] tensorA_0.36.2         blob_1.2.4             GenomeInfoDbData_1.2.9
 [7] yaml_2.3.7             robustbase_0.95-1      RSQLite_2.3.1         
[10] pillar_1.9.0           lattice_0.20-45        glue_1.6.2            
[13] digest_0.6.31          RColorBrewer_1.1-3     XVector_0.38.0        
[16] colorspace_2.1-0       htmltools_0.5.5        Matrix_1.5-1          
[19] XML_3.99-0.14          pkgconfig_2.0.3        zlibbioc_1.44.0       
[22] xtable_1.8-4           scales_1.2.1           tzdb_0.4.0            
[25] BiocParallel_1.32.6    timechange_0.2.0       annotate_1.76.0       
[28] KEGGREST_1.38.0        generics_0.1.3         farver_2.1.1          
[31] cachem_1.0.8           withr_2.5.0            cli_3.6.1             
[34] crayon_1.5.2           magrittr_2.0.3         memoise_2.0.1         
[37] evaluate_0.21          fansi_1.0.4            MASS_7.3-58.1         
[40] tools_4.2.2            hms_1.1.3              lifecycle_1.0.3       
[43] locfit_1.5-9.7         munsell_0.5.0          DelayedArray_0.24.0   
[46] Biostrings_2.66.0      AnnotationDbi_1.60.2   compiler_4.2.2        
[49] rlang_1.1.1            grid_4.2.2             RCurl_1.98-1.12       
[52] rstudioapi_0.14        htmlwidgets_1.6.2      bitops_1.0-7          
[55] labeling_0.4.2         rmarkdown_2.22         gtable_0.3.3          
[58] codetools_0.2-18       DBI_1.1.3              R6_2.5.1              
[61] bayesm_3.1-5           knitr_1.43             bit_4.0.5             
[64] fastmap_1.1.1          utf8_1.2.3             stringi_1.7.12        
[67] parallel_4.2.2         Rcpp_1.0.10            png_0.1-8             
[70] vctrs_0.6.2            geneplotter_1.76.0     DEoptimR_1.0-13       
[73] tidyselect_1.2.0       xfun_0.39             </code></pre>
</div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>