<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.251">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>eukrhythmic compilation</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title">eukrhythmic compilation</h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation docked overflow-auto">
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Home</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./env-metadata.html" class="sidebar-item-text sidebar-link">Water column profiles</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./process-eukrhythmic-output.html" class="sidebar-item-text sidebar-link active">eukrhythmic output</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./metat-analysis.html" class="sidebar-item-text sidebar-link">Data analysis</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#working-with-eukrhythmic-output-files" id="toc-working-with-eukrhythmic-output-files" class="nav-link active" data-scroll-target="#working-with-eukrhythmic-output-files">Working with eukrhythmic output files</a>
  <ul class="collapse">
  <li><a href="#compilation-scripts" id="toc-compilation-scripts" class="nav-link" data-scroll-target="#compilation-scripts">Compilation scripts</a></li>
  </ul></li>
  <li><a href="#obtain-library-normalized-count-data" id="toc-obtain-library-normalized-count-data" class="nav-link" data-scroll-target="#obtain-library-normalized-count-data">Obtain library normalized count data</a>
  <ul class="collapse">
  <li><a href="#set-up-r-environment" id="toc-set-up-r-environment" class="nav-link" data-scroll-target="#set-up-r-environment">Set up R environment</a></li>
  </ul></li>
  <li><a href="#deseq2-analysis" id="toc-deseq2-analysis" class="nav-link" data-scroll-target="#deseq2-analysis">DESeq2 analysis</a>
  <ul class="collapse">
  <li><a href="#by-sample-name" id="toc-by-sample-name" class="nav-link" data-scroll-target="#by-sample-name">By sample name</a></li>
  <li><a href="#extract-count-table" id="toc-extract-count-table" class="nav-link" data-scroll-target="#extract-count-table">Extract count table</a></li>
  <li><a href="#extract-centered-data" id="toc-extract-centered-data" class="nav-link" data-scroll-target="#extract-centered-data">Extract centered data</a></li>
  </ul></li>
  <li><a href="#deseq2-reanalysis-for-de-genes" id="toc-deseq2-reanalysis-for-de-genes" class="nav-link" data-scroll-target="#deseq2-reanalysis-for-de-genes">DEseq2 reanalysis for DE genes</a>
  <ul class="collapse">
  <li><a href="#create-subset-parameters" id="toc-create-subset-parameters" class="nav-link" data-scroll-target="#create-subset-parameters">Create subset parameters</a>
  <ul class="collapse">
  <li><a href="#subset-samples-as-parameters" id="toc-subset-samples-as-parameters" class="nav-link" data-scroll-target="#subset-samples-as-parameters">Subset samples as parameters</a></li>
  <li><a href="#subset-genes-taxa-as-parameters" id="toc-subset-genes-taxa-as-parameters" class="nav-link" data-scroll-target="#subset-genes-taxa-as-parameters">Subset genes &amp; taxa as parameters</a></li>
  <li><a href="#txi-function" id="toc-txi-function" class="nav-link" data-scroll-target="#txi-function">Txi function</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#whole-community-deseq-script" id="toc-whole-community-deseq-script" class="nav-link" data-scroll-target="#whole-community-deseq-script">Whole community DESeq script</a>
  <ul class="collapse">
  <li><a href="#all-samples" id="toc-all-samples" class="nav-link" data-scroll-target="#all-samples">All samples</a></li>
  <li><a href="#by-region" id="toc-by-region" class="nav-link" data-scroll-target="#by-region">By region</a></li>
  <li><a href="#by-light-availability" id="toc-by-light-availability" class="nav-link" data-scroll-target="#by-light-availability">By light availability</a></li>
  </ul></li>
  <li><a href="#by-taxonomic-groups---deseq" id="toc-by-taxonomic-groups---deseq" class="nav-link" data-scroll-target="#by-taxonomic-groups---deseq">By taxonomic groups - DESeq</a>
  <ul class="collapse">
  <li><a href="#question-1" id="toc-question-1" class="nav-link" data-scroll-target="#question-1">Question 1</a></li>
  <li><a href="#question-2" id="toc-question-2" class="nav-link" data-scroll-target="#question-2">Question 2</a></li>
  </ul></li>
  <li><a href="#curated-genes-only" id="toc-curated-genes-only" class="nav-link" data-scroll-target="#curated-genes-only">Curated genes only</a></li>
  <li><a href="#july-versus-march-for-npsg" id="toc-july-versus-march-for-npsg" class="nav-link" data-scroll-target="#july-versus-march-for-npsg">July versus March for NPSG</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block">eukrhythmic compilation</h1>
</div>



<div class="quarto-title-meta">

    
    
  </div>
  

</header>

<section id="working-with-eukrhythmic-output-files" class="level1">
<h1>Working with eukrhythmic output files</h1>
<p><strong>List of output directories</strong>:</p>
<p><code>final-files/</code></p>
<ul>
<li><code>00-nucleotide_assembly/</code>
<ul>
<li><code>MAD.filtered.nospace.fasta</code>: Full length sequences of merged assembly groups (MAD) that have been de-duplicated</li>
</ul></li>
<li><code>01-predicted_proteins/</code>
<ul>
<li><code>MAD.fasta.transdecoder.cds</code>: Identified coding regions of transcripts from the Merged assembly groups (MAD)</li>
<li><code>MAD.fasta.transdecoder.pep</code>: Translated peptide sequences from the Merged assembly groups (MAD)</li>
</ul></li>
<li><code>02-annotation_table/</code>
<ul>
<li><code>TaxonomicAndFunctionalAnnotations.csv</code>: Taxonomic levels, GOs, PFAMs, and KEGG KOs - assigned by short seq ID and the full sequence ID.</li>
</ul></li>
<li><code>03-abundance_tables/</code>
<ul>
<li><code>SeqID_Dict_NoSpaceNames.csv</code>: key to link long sequence read IDs to ShortSeq IDs that will link to taxonomic and functional annotations</li>
</ul></li>
</ul>
<p><code>intermediate-files/</code></p>
<ul>
<li><code>04-compare</code>
<ul>
<li><code>14-MAD-mapping-filtered</code>: location of individual salmon outputs for all samples (quant.sf)</li>
</ul></li>
</ul>
<section id="compilation-scripts" class="level2">
<h2 class="anchored" data-anchor-id="compilation-scripts">Compilation scripts</h2>
<p><code>collate-results.py</code>: <em>description of code</em></p>
<p>~Arianna - how do you want to handle this?~</p>
<blockquote class="blockquote">
<p>Notes for Sarah &amp; Arianna: hopefully we can include a description of taking outputs from eukrhythmic and putting them in specific directories, etc. that can go stright into the run_tximport.R below.</p>
</blockquote>
</section>
</section>
<section id="obtain-library-normalized-count-data" class="level1">
<h1>Obtain library normalized count data</h1>
<p>R script to run this: <code>scripts/run_tximport.R</code>.</p>
<p>Our goal is to take eukrhthymic estimated counts (from salmon) to get transcript-level abundance estimates that will account for gene length. The outcome from this is that we get average transcript length across samples. For this we want to import all data into DESeq using tximport as an interface.</p>
<ul>
<li>tximport
<ul>
<li><em>Why tximport</em>? tximport allows us to correct for the changes in gene length across our samples. This way we can use transcript-level estimates for our metatranscriptomic analysis. Also, tximport knows the output from salmon, so we can easily put the salmon output into tximport. Then it can link directly to edgeR or DESeq2.</li>
</ul></li>
<li>DESeq
<ul>
<li>DESeq: Love, M.I., Huber, W., Anders, S. (2014) Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology, 15:550. 10.1186/s13059-014-0550-8</li>
<li>Explaination using DESeq: https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html</li>
<li>https://bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf</li>
</ul></li>
</ul>
<section id="set-up-r-environment" class="level2">
<h2 class="anchored" data-anchor-id="set-up-r-environment">Set up R environment</h2>
<p>Library required to execute code in R</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Installations</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     install.packages("BiocManager")</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("DESeq2")</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># BiocManager::install("tximport")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load necessary libraries &amp; get number of available threads for data.table.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># | message: false</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># | warning: false</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>num_threads <span class="ot">&lt;-</span> <span class="fu">getDTthreads</span>()</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Conflicts between tidyverse and data.table can be ignored</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Output files from<code>run_tximport.R</code> provide an RData file that can be used for the DESeq input file (<code>tximport_metaT-ALOHA-CA.Rdata</code>).</p>
</section>
</section>
<section id="deseq2-analysis" class="level1">
<h1>DESeq2 analysis</h1>
<p>Import as DESeq object from tximport.</p>
<p>DESeq will automatically find a control in your levels or list for the design. There is a way for DESeq to ignore this need for a control.</p>
<p>SEE SHARON GRIM’S CODE</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"tximport_metaT-ALOHA-CA.Rdata"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># colData : Rows of colData correspond to columns of countData</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># design ~ x == serves the purpose of considering biological replicates</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>Loading objects: txi sample_merged</p>
</blockquote>
<section id="by-sample-name" class="level2">
<h2 class="anchored" data-anchor-id="by-sample-name">By sample name</h2>
<p>For taxonomic bar plot and overall cluster analysis. (March and July are separate at ALOHA)</p>
<p>See how to set up subsetting? https://github.com/WHOIGit/super-waffle/blob/main/01_seqpain_deseq.R</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>ds_tpm_samplename <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi,</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> sample_merged,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> SAMPLENAME)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><em>optional</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_samplename)) <span class="sc">&gt;=</span> <span class="dv">10</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> ds_tpm_samplename[keep,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><code>dds</code> includes &gt; 10. Now dim is 6972094, it was originally, 14737135. Keeping about 47% of data.</p>
</section>
<section id="extract-count-table" class="level2">
<h2 class="anchored" data-anchor-id="extract-count-table">Extract count table</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>counts_scaled <span class="ot">&lt;-</span> <span class="fu">makeCountsFromAbundance</span>(</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi<span class="sc">$</span>counts),</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi<span class="sc">$</span>abundance),</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi<span class="sc">$</span>length),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">countsFromAbundance =</span> <span class="st">"scaledTPM"</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>counts_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts_scaled)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Rename so replicates have the same name for counts</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>names_orig <span class="ot">&lt;-</span> <span class="fu">colnames</span>(counts_df)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>names_new <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_[^_]+$"</span>, <span class="st">""</span>, names_orig)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(counts_df) <span class="ot">&lt;-</span> names_new</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Mean across columns that have the same name - which are replicates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>mean_counts_df <span class="ot">&lt;-</span> counts_df <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(<span class="fu">as.list</span>(.) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Filter</span>(is.numeric, .) <span class="sc">%&gt;%</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(<span class="fu">names</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(as.data.frame) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(rowMeans) <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setNames</span>(<span class="fu">paste0</span>(<span class="st">"mean."</span>, <span class="fu">names</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mean_counts_df, sample_merged, <span class="at">file =</span> <span class="st">"Avg_scaled_tpm_08222023.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-centered-data" class="level2">
<h2 class="anchored" data-anchor-id="extract-centered-data">Extract centered data</h2>
<p>See “Count data transformations” in https://introtogenomics.readthedocs.io/en/latest/2021.11.11.DeseqTutorial.html</p>
<p>Use variance Stabilizing Transformation (<code>vst</code>) command to estimate dispersion.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># output is a DESeqtransform object</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>vsd_all <span class="ot">&lt;-</span> <span class="fu">vst</span>(ds_tpm_samplename)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># make a transformed count matrix</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># vsd_blind &lt;- vst(ds_tpm_samplename, blind = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Example output: &gt; using ‘avgTxLength’ from assays(dds), correcting for library size</p>
<p>Look at output with assay.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(vsd_all), <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Extract as a data frame, where values are center normalized - based on avgTxLength.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>df_ctr_norm <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(vsd_all))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(df_ctr_norm, <span class="at">file =</span> <span class="st">"normed_center_df_08222023.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="deseq2-reanalysis-for-de-genes" class="level1">
<h1>DEseq2 reanalysis for DE genes</h1>
<p>Repeat DESeq2 design to extracted differentially expressed genes for specific patterns below. Subset is required ahead of time - we can do this with txi directly.</p>
<p>There are a few different ways we want to perform the untargeted analysis. First we will do a ‘whole community’ series of analyses to look at overall amount of differentially expressed genes. Below we will set up parameters associated with the queries we’ve planned.</p>
<p>See script <code>scripts/run_get_params.R</code> This is stored on GRACE.</p>
<section id="create-subset-parameters" class="level2">
<h2 class="anchored" data-anchor-id="create-subset-parameters">Create subset parameters</h2>
<section id="subset-samples-as-parameters" class="level3">
<h3 class="anchored" data-anchor-id="subset-samples-as-parameters">Subset samples as parameters</h3>
<p>Use the metadata data frame to create sample lists for the comparisons we want to do.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"input-data/sample-list-revised.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For the whole community work, we want to compare CA vs NPSG and euphotic vs.&nbsp;subeuphotic.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>npsg_only <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PACIFIC_REGION <span class="sc">==</span> <span class="st">"NPSG"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare euphotic and subeuphotic</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Use this to compare across months too</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>ca_only <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(PACIFIC_REGION <span class="sc">!=</span> <span class="st">"NPSG"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare euphotic and subeuphotic</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>euphotic <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(LIGHT <span class="sc">==</span> <span class="st">"Euphotic"</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare NPSG to CA euphotic zone</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>subeuphotic <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(LIGHT <span class="sc">!=</span> <span class="st">"Euphotic"</span>)<span class="sc">%&gt;%</span> </span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare NPSG to CA subeuphotic zone</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="subset-genes-taxa-as-parameters" class="level3">
<h3 class="anchored" data-anchor-id="subset-genes-taxa-as-parameters">Subset genes &amp; taxa as parameters</h3>
<p>Need to make a list of the <code>SequenceID</code> from the taxonomy and functional annotations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>taxfxn <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"TaxonomicAndFunctionalAnnotations.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="genes" class="level4">
<h4 class="anchored" data-anchor-id="genes">Genes</h4>
<p>Determine character lists from gene list as subsets for our later queries.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get list of annotations where a GO, PFAM, and KEGG ID were assigned.</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>genes_fxn_all <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, GOs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]]) <span class="co">#this line outputs the selected vector from the pipe</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Removing unclassified and unannotated sequences</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>genes_tax_fxn_all <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, GOs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">&amp;</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>         <span class="sc">&amp;</span> Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> Supergroup <span class="sc">!=</span> <span class="st">"Unclassified"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>From KEGG database reformatting make <code>key_geneid</code>. Isolate genes that are curated.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../KEGG_DB/combined_kegg.csv"</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(kegg<span class="sc">$</span>KO_number)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>curated_kegg <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../KEGG_DB/reformat-kegg-pfam-skh.csv"</span>)</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>key_geneid <span class="ot">&lt;-</span> curated_kegg <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>X) <span class="sc">%&gt;%</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">right_join</span>(kegg <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="at">KEGG =</span> KO_number, <span class="fu">everything</span>(), <span class="sc">-</span>X)) <span class="sc">%&gt;%</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"KeggOrthology_"</span>), Category01, Category02, FullName, GeneID, Gene_identification, KEGG, PFAM, Descriptions, <span class="at">REF =</span> REFs)</span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a><span class="co"># write.csv(key_geneid, file = "keygene_id.csv")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>kegg_keep <span class="ot">&lt;-</span> <span class="fu">as.character</span>(key_geneid <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category01 <span class="sc">!=</span> <span class="st">""</span> <span class="sc">&amp;</span> <span class="sc">!</span>(<span class="fu">is.na</span>(Category01))) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(KEGG) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  .[[<span class="st">"KEGG"</span>]])</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># length(kegg_keep)</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>genes_kegg_curated <span class="ot">&lt;-</span> <span class="fu">as.character</span>(taxfxn <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">KEGG_mod =</span> <span class="fu">str_remove_all</span>(KEGG_ko, <span class="st">"ko:"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(KEGG_mod <span class="sc">%in%</span> kegg_keep) <span class="sc">%&gt;%</span> </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="taxa" class="level4">
<h4 class="anchored" data-anchor-id="taxa">Taxa</h4>
<p>Isolate transcript IDs (<em>SequenceID</em>) of individual taxa. For each taxa subset, we are also only selecting transcript IDs with PFAM or KEGG IDs.</p>
<p>For queries, we want to isolate:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>euks_only <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Domain <span class="sc">==</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>diatom <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Class <span class="sc">==</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]]) </span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Dinoflagellata pPhylum</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>dinos <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Phylum <span class="sc">==</span> <span class="st">"Dinoflagellata"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]]) </span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Haptophyta in phylum</span></span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>hapto <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Phylum <span class="sc">==</span> <span class="st">"Haptophyta"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Chlorophyta</span></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>chloro <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Phylum <span class="sc">==</span> <span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a><span class="co">#Ciliophora</span></span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>ciliate <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Phylum <span class="sc">==</span> <span class="st">"Ciliophora"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Rhizaria at supergroup</span></span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>rhizaria <span class="ot">&lt;-</span> <span class="fu">as.character</span>(</span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(taxfxn, Supergroup <span class="sc">==</span> <span class="st">"Rhizaria"</span> <span class="sc">&amp;</span> (PFAMs <span class="sc">!=</span> <span class="st">"-"</span> <span class="sc">|</span> KEGG_ko <span class="sc">!=</span> <span class="st">"-"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-38"><a href="#cb20-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(SequenceID) <span class="sc">%&gt;%</span> </span>
<span id="cb20-39"><a href="#cb20-39" aria-hidden="true" tabindex="-1"></a>    .[[<span class="st">"SequenceID"</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Save objects for subseting.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># These are data frames</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  all_samples, npsg_only, ca_only, euphotic, subeuphotic,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># these are lists (character lists)</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>     genes_fxn_all, genes_tax_fxn_all, key_geneid, genes_kegg_curated, </span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>     euks_only, diatom, dinos, hapto, chloro, ciliate, rhizaria,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">file =</span> <span class="st">"sample-gene-lists-forTXI.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="txi-function" class="level3">
<h3 class="anchored" data-anchor-id="txi-function">Txi function</h3>
<p>Code for this function is from: https://rdrr.io/github/hcnh174/hlsgr/man/subsetTxi.html Code: https://rdrr.io/github/hcnh174/hlsgr/src/R/rnaseq.R</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset txi directly</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>subsetTxi <span class="ot">&lt;-</span> <span class="cf">function</span>(txi, samples, <span class="at">include_genes=</span><span class="fu">rownames</span>(txi<span class="sc">$</span>counts))</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(txi<span class="sc">$</span>counts)[<span class="fu">rownames</span>(txi<span class="sc">$</span>counts) <span class="sc">%in%</span> include_genes]</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  txi<span class="sc">$</span>abundance <span class="ot">&lt;-</span> txi<span class="sc">$</span>abundance[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  txi<span class="sc">$</span>counts <span class="ot">&lt;-</span> txi<span class="sc">$</span>counts[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  txi<span class="sc">$</span>length <span class="ot">&lt;-</span> txi<span class="sc">$</span>length[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(txi)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Example usage </span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co"># tmp &lt;- sample(taxfxn$SequenceID,10,replace = FALSE)</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co"># pola &lt;- data.frame(sample = c("PortofLA_1", "PortofLA_2"))</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co"># txi_pola &lt;- subsetTxi(txi, pola, tmp)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="whole-community-deseq-script" class="level1">
<h1>Whole community DESeq script</h1>
<p>Compare across light and pacific region, first isolate only eukaryotes and annotated transcripts.</p>
<section id="all-samples" class="level2">
<h2 class="anchored" data-anchor-id="all-samples">All samples</h2>
<p>See script <code>scripts/run_txi_wholecomm.R</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset eukaryotes only, and keep all samples.</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>txi_euk_annot <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, all_samples, euks_only)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>ds_tpm_light <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_euk_annot,</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> sample_merged,</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> LIGHT)</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>ds_tpm_pac <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_euk_annot,</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> sample_merged,</span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> PACIFIC_REGION)</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>ds_tpm_lightpac <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_euk_annot,</span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> sample_merged,</span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> PACIFIC_REGION <span class="sc">+</span> LIGHT)</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ds_tpm_light, ds_tpm_pac, ds_tpm_lightpac, <span class="at">file =</span> <span class="st">"light-pacregion-deseq.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="by-region" class="level2">
<h2 class="anchored" data-anchor-id="by-region">By region</h2>
<p>See script <code>scripts/run_txi_region.r</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset txi</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>txi_npsg <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, npsg_only, euks_only)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Reset sample_merged</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="do">## Subset</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(npsg_only<span class="sc">$</span>sample))</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Set names</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_npsg<span class="sc">$</span>counts)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare Euphotic vs. sub-euphotic samples in the NPSG</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>ds_tpm_npsg_light <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_npsg,</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> LIGHT)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare July vs. March in the NPSG</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>ds_tpm_npsg_month <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_npsg,</span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> MONTH)</span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ds_tpm_npsg_light, ds_tpm_npsg_month, <span class="at">file =</span> <span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/npsg-deseq.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>txi_ca <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, ca_only, euks_only)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(ca_only<span class="sc">$</span>sample))</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_ca<span class="sc">$</span>counts)</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare euphotic vs. subeuphotic in coastal California</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Includes Port of LA and Catalina</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>ds_tpm_ca_light <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_ca,</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> LIGHT)</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ds_tpm_ca_light, <span class="at">file =</span> <span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/ca-deseq.RData"</span>)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="do">##</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(txi_npsg);<span class="fu">rm</span>(txi_ca) <span class="co">#Save room</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="by-light-availability" class="level2">
<h2 class="anchored" data-anchor-id="by-light-availability">By light availability</h2>
<p>See script <code>scripts/run_txi_light.r</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset txi</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>txi_euph <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, euphotic, euks_only)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(euphotic<span class="sc">$</span>sample))</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_euph<span class="sc">$</span>counts)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare CA versus NPSG within euphotic samples</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>ds_tpm_euphotic <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_euph,</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> PACIFIC_REGION)</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>txi_subeuph <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, subeuphotic, euks_only)</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(subeuphotic<span class="sc">$</span>sample))</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_subeuph<span class="sc">$</span>counts)</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare CA versus NPSG at subeuphotic depths</span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>ds_tpm_subeuphotic <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_subeuph,</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> PACIFIC_REGION)</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ds_tpm_subeuphotic, ds_tpm_euphotic, <span class="at">file =</span> <span class="st">"/vortexfs1/scratch/sarahhu/txi-objs-metaT/euphotic-subeuphotic-deseq.RData"</span>)</span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a><span class="co"># save(txi_euph, txi_subeuph, file = "/vortexfs1/scratch/sarahhu/txi-objs-metaT/light_txi.RData")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="by-taxonomic-groups---deseq" class="level1">
<h1>By taxonomic groups - DESeq</h1>
<p>Compare and contrast within regions. For part of our analysis we want to subset only the euphotic zone samples. Specifically targeting the difference between the SPOT station surface compared to: NPSG surface, NPSG DCM, NPSG 150m, Port of LA, and Catalina.</p>
<section id="question-1" class="level2">
<h2 class="anchored" data-anchor-id="question-1">Question 1</h2>
<p>For the first of our two core questions, we want to know what genes are significantly differentially expressed among the euphotic zone samples for the dominant protistan phytoplankton. We are specifically focusing on diatoms, chlorophytes, archaeplastidia (chlorophya), dinoflagellates, and haptophytes for this.</p>
<blockquote class="blockquote">
<p>How does nutrient utilization among euphotic zone phytoplankton vary between coastal California and the NPSG?</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to subset txi by individual taxa and perform DESeq</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>deseq_region_bytax <span class="ot">&lt;-</span> <span class="cf">function</span>(sample_set, gene_set){</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># First incorporate the txi subset fxn</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  subsetTxi <span class="ot">&lt;-</span> <span class="cf">function</span>(txi, samples, <span class="at">include_genes=</span><span class="fu">rownames</span>(txi<span class="sc">$</span>counts))</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>    genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(txi<span class="sc">$</span>counts)[<span class="fu">rownames</span>(txi<span class="sc">$</span>counts) <span class="sc">%in%</span> include_genes]</span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>    txi<span class="sc">$</span>abundance <span class="ot">&lt;-</span> txi<span class="sc">$</span>abundance[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>    txi<span class="sc">$</span>counts <span class="ot">&lt;-</span> txi<span class="sc">$</span>counts[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>    txi<span class="sc">$</span>length <span class="ot">&lt;-</span> txi<span class="sc">$</span>length[genes, samples<span class="sc">$</span>sample]</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">return</span>(txi)</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Run subset and sample merge re-set</span></span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  txi_output <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, sample_set, gene_set)</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># This script will keep replacing tmp_sample_merged</span></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(sample_set<span class="sc">$</span>sample))</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_output<span class="sc">$</span>counts)</span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Import as DESeq object - use PACIFIC_REGION in the design</span></span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># DESeq</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>  ds_tpm_output <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_output,</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> PACIFIC_REGION)</span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># return(ds_tpm_output)</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Further process DESeq</span></span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>  groupsize <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="co"># Transcript to consider, must be in at least 3 samples</span></span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>  keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">counts</span>(ds_tpm_output) <span class="sc">&gt;=</span> <span class="dv">10</span>) <span class="sc">&gt;=</span> groupsize <span class="co"># And have &gt;= to 10 counts</span></span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>  ds_tpm_output_filtered <span class="ot">&lt;-</span> ds_tpm_output[keep,]</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>  <span class="do">###</span></span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtering stats:</span></span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">Started with "</span>, <span class="fu">dim</span>(ds_tpm_output)[<span class="dv">1</span>], <span class="st">"observations. Filtering by 2 samples and 10 counts resulted in,"</span>, <span class="fu">dim</span>(ds_tpm_output_filtered)[<span class="dv">1</span>], <span class="st">", which is"</span>, (<span class="dv">100</span><span class="sc">*</span>(<span class="fu">dim</span>(ds_tpm_output_filtered)[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">dim</span>(ds_tpm_output)[<span class="dv">1</span>])), <span class="st">"% of the data.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>  <span class="do">###</span></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>  <span class="do">## Positive log fold change == up regulared in CA, compared to NPSG</span></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>  ds_tpm_output_filtered<span class="sc">$</span>PACIFIC_REGION <span class="ot">&lt;-</span> <span class="fu">factor</span>(ds_tpm_output_filtered<span class="sc">$</span>PACIFIC_REGION, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"NPSG"</span>, <span class="st">"CA"</span>))</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>  de_tax_output <span class="ot">&lt;-</span> DESeq2<span class="sc">::</span><span class="fu">DESeq</span>(ds_tpm_output_filtered)</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">resultsNames</span>(de_tax_output)</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summary</span>(de_tax_output)</span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(de_tax_output)</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">Completed.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Apply to each taxa</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="co"># From the euphotic samples, what are DE genes among haptophytes between CA and NPSG?</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">Diatoms start.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>de_diatom <span class="ot">&lt;-</span> <span class="fu">deseq_region_bytax</span>(euphotic, diatom)</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">\Haptophytes start.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="co"># From the euphotic samples, what are DE genes among haptophytes between CA and NPSG?</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>de_hapto <span class="ot">&lt;-</span> <span class="fu">deseq_region_bytax</span>(euphotic, hapto)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">\Dinoflagellates start.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="co"># From the euphotic samples, what are DE genes among Dinoflagellates between CA and NPSG?</span></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>de_dinos <span class="ot">&lt;-</span> <span class="fu">deseq_region_bytax</span>(euphotic, dinos)</span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n</span><span class="st">\Chlorophytes start.</span><span class="sc">\n\n</span><span class="st">"</span>)</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a><span class="co"># From the euphotic samples, what are DE genes among chlorophytes between CA and NPSG?</span></span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>de_chloro <span class="ot">&lt;-</span> <span class="fu">deseq_region_bytax</span>(euphotic, chloro)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(de_diatom, de_hapto, de_dinos, de_chloro, <span class="at">file =</span> <span class="st">"/scratch/group/hu-lab/pacocean-metaT/Robjs/euphotic_by_taxa.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="question-2" class="level2">
<h2 class="anchored" data-anchor-id="question-2">Question 2</h2>
<p>We can repeat the above approach, but select different taxa.</p>
<blockquote class="blockquote">
<p>Do we see a similar euphotic vs.&nbsp;sub-euphotic shift in the metabolic potential of taxa observed at coastal California and the NPSG?</p>
</blockquote>
<p>~ repeat code above with euphotic versus euphotic ~</p>
</section>
</section>
<section id="curated-genes-only" class="level1">
<h1>Curated genes only</h1>
<p>Here, we isolated annotated transcripts that we curated ourselves.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>all_samples <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="at">sample =</span> SAMPLE)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>txi_curated <span class="ot">&lt;-</span> <span class="fu">subsetTxi</span>(txi, all_samples, genes_kegg_curated)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>counts_scaled_curated <span class="ot">&lt;-</span> <span class="fu">makeCountsFromAbundance</span>(</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi_curated<span class="sc">$</span>counts),</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi_curated<span class="sc">$</span>abundance),</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>(txi_curated<span class="sc">$</span>length),</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">countsFromAbundance =</span> <span class="st">"scaledTPM"</span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>counts_df_curated <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(counts_scaled_curated)</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>names_orig <span class="ot">&lt;-</span> <span class="fu">colnames</span>(counts_df_curated)</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>names_new <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"_[^_]+$"</span>, <span class="st">""</span>, names_orig)</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(counts_df_curated) <span class="ot">&lt;-</span> names_new</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>mean_counts_df_curated <span class="ot">&lt;-</span> counts_df_curated <span class="sc">%&gt;%</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cbind</span>(<span class="fu">as.list</span>(.) <span class="sc">%&gt;%</span></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">Filter</span>(is.numeric, .) <span class="sc">%&gt;%</span></span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">split</span>(<span class="fu">names</span>(.)) <span class="sc">%&gt;%</span></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(as.data.frame) <span class="sc">%&gt;%</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lapply</span>(rowMeans) <span class="sc">%&gt;%</span></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">setNames</span>(<span class="fu">paste0</span>(<span class="st">"mean."</span>, <span class="fu">names</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"mean"</span>))</span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>counts_curated <span class="ot">&lt;-</span> mean_counts_df_curated <span class="sc">%&gt;%</span> </span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SequenceID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(counts_curated, sample_merged, <span class="at">file =</span> <span class="st">"Avg_scaled_tpm_curated_08252023.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>tmp_sample_merged <span class="ot">&lt;-</span> sample_merged <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Sample_rep <span class="sc">%in%</span> <span class="fu">as.character</span>(all_samples<span class="sc">$</span>sample))</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> tmp_sample_merged<span class="sc">$</span>Sample_rep</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_sample_merged) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(txi_curated<span class="sc">$</span>counts)</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>ds_tpm_curated <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromTximport</span>(txi_curated,</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> tmp_sample_merged,</span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span> <span class="sc">+</span> SAMPLENAME)</span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>vsd_all <span class="ot">&lt;-</span> <span class="fu">vst</span>(ds_tpm_curated)</span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>df_ctr_norm_curated <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(vsd_all))</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a>ctr_norm_curated <span class="ot">&lt;-</span> df_ctr_norm_curated <span class="sc">%&gt;%</span> </span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SequenceID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(taxfxn)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(ctr_norm_curated, ds_tpm_curated, <span class="at">file =</span> <span class="st">"normed_center_df_curated_08252023.RData"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="july-versus-march-for-npsg" class="level1">
<h1>July versus March for NPSG</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"/scratch/group/hu-lab/pacocean-metaT/Robjs/npsg-deseq.RData"</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Compare July vs. March in the NPSG</span></span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ds_tpm_npsg_month &lt;- DESeqDataSetFromTximport(txi_npsg,</span></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                                               colData = tmp_sample_merged,</span></span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                                               design = ~0 + MONTH)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>How many genes are differentially expressed between July and March?</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Positive log fold change == up regulated in July, compared to March</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>ds_tpm_npsg_month<span class="sc">$</span>MONTH <span class="ot">&lt;-</span> <span class="fu">factor</span>(ds_tpm_npsg_month<span class="sc">$</span>MONTH, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"March"</span>, <span class="st">"July"</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>de_month <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ds_tpm_npsg_month)</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>results_month <span class="ot">&lt;-</span> <span class="fu">results</span>(de_month, <span class="at">alpha=</span><span class="fl">0.05</span>)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plotMA(results_month)</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">summary</span>(results_month)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(out, <span class="at">file =</span> <span class="st">"summary-months.csv"</span>)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot log fold change</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="fu">svg</span>(<span class="st">"rplot-month.csv"</span>, <span class="at">width =</span> <span class="dv">9</span>, <span class="at">height =</span> <span class="dv">8</span>)</span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(results_month) <span class="sc">%&gt;%</span> </span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGULATION =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in July"</span>,</span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in March"</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">SIGNIFICANT =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a>    pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Significantly"</span>,</span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Not significantly"</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> baseMean, <span class="at">y =</span> log2FoldChange, <span class="at">color =</span> SIGNIFICANT)) <span class="sc">+</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_log10</span>() <span class="sc">+</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#878787"</span>, <span class="st">"#d73027"</span>)) <span class="sc">+</span></span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">mcols</span>(results_month)<span class="sc">$</span>description[<span class="dv">2</span>])</span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Get table</span></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a>month_transcripts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(results_month) <span class="sc">%&gt;%</span> </span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">REGULATION =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in July"</span>,</span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a>    log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"upregulated in March"</span></span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a>  ),</span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">SIGNIFICANT =</span> <span class="fu">case_when</span>(</span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a>    pvalue <span class="sc">&lt;=</span> <span class="fl">0.05</span> <span class="sc">~</span> <span class="st">"Significantly"</span>,</span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Not significantly"</span></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(SIGNIFICANT <span class="sc">==</span> <span class="st">"Significantly"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"SequenceID"</span>)</span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(month_transcripts, <span class="at">file =</span> <span class="st">"/scratch/group/hu-lab/pacocean-metaT/month-transcripts.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>